{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# variant2vec\n",
    "\n",
    "Learn an emedding of variants annotations (HGVS, BIC etc...), Hugo gene symbols, and classifications (pathogenic, benign...) from the text of pubmed articles.\n",
    "\n",
    "Goal to see if something like [this](https://www.nature.com/articles/s41586-019-1335-8) paper can be applied to determine characteristics of a variant, or at lease identify the gene associated with the variant from the paper where multiple variants may be mentioned with no canonical gene or transcript qualifier."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "import gzip\n",
    "import glob\n",
    "\n",
    "crawl_path = \"/public/groups/brcaexchange/literature-search/crawl\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ingest\n",
    "Load the text from the pubMunch [pubStore](https://github.com/maximilianh/pubMunch/blob/master/lib/pubStore.py) generated in a BRCA Exchange literature search crawl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15079 articles loaded from the articles sqlite database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Trends in genetics : TIG</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>8507085</td>\n",
       "      <td>2001</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>S18</td>\n",
       "      <td>11585672</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/s0168-9525(01)02451-9</td>\n",
       "      <td>https://linkinghub.elsevier.com/retrieve/pii/S...</td>\n",
       "      <td>2018-11-14T16:29:09+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>5258</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Breast cancer research and treatment</td>\n",
       "      <td>0167-6806</td>\n",
       "      <td>1573-7217</td>\n",
       "      <td>8111104</td>\n",
       "      <td>2010</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>575</td>\n",
       "      <td>19688261</td>\n",
       "      <td></td>\n",
       "      <td>10.1007/s10549-009-0501-3</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:16+0000</td>\n",
       "      <td>5048</td>\n",
       "      <td>100825</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5012228710</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Science (New York, N.Y.)</td>\n",
       "      <td>0036-8075</td>\n",
       "      <td>1095-9203</td>\n",
       "      <td>0404511</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>5588</td>\n",
       "      <td>1837</td>\n",
       "      <td>12228710</td>\n",
       "      <td></td>\n",
       "      <td>10.1126/science.297.5588.1837</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:40+0000</td>\n",
       "      <td>1436851129</td>\n",
       "      <td>148950</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5029369605</td>\n",
       "      <td>PMID29369605</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Genetika</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0047354</td>\n",
       "      <td>2016</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1215</td>\n",
       "      <td>29369605</td>\n",
       "      <td></td>\n",
       "      <td>10.1134/s102279541609012x</td>\n",
       "      <td>https://link.springer.com/article/10.1134%2FS1...</td>\n",
       "      <td>2018-11-14T16:30:43+0000</td>\n",
       "      <td>4458079378</td>\n",
       "      <td>22264</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5022084640</td>\n",
       "      <td>PMID22084640</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Therapeutic advances in medical oncology</td>\n",
       "      <td>1758-8340</td>\n",
       "      <td>1758-8359</td>\n",
       "      <td>101510808</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>257</td>\n",
       "      <td>22084640</td>\n",
       "      <td>3210467</td>\n",
       "      <td>10.1177/1758834011417039</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:30:58+0000</td>\n",
       "      <td>6079002966</td>\n",
       "      <td>93229</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5011585672  PMID11585672         download            \n",
       "1  5019688261  PMID19688261         download            \n",
       "2  5012228710  PMID12228710         download            \n",
       "3  5029369605  PMID29369605         download            \n",
       "4  5022084640  PMID22084640         download            \n",
       "\n",
       "                                    journal  printIssn      eIssn  \\\n",
       "0                  Trends in genetics : TIG  0168-9525  0168-9525   \n",
       "1      Breast cancer research and treatment  0167-6806  1573-7217   \n",
       "2                  Science (New York, N.Y.)  0036-8075  1095-9203   \n",
       "3                                  Genetika  0016-6758  0016-6758   \n",
       "4  Therapeutic advances in medical oncology  1758-8340  1758-8359   \n",
       "\n",
       "  journalUniqueId  year  ... issue  page      pmid    pmcId  \\\n",
       "0         8507085  2001  ...    10   S18  11585672            \n",
       "1         8111104  2010  ...     3   575  19688261            \n",
       "2         0404511  2002  ...  5588  1837  12228710            \n",
       "3         0047354  2016  ...    10  1215  29369605            \n",
       "4       101510808  2011  ...     6   257  22084640  3210467   \n",
       "\n",
       "                             doi  \\\n",
       "0  10.1016/s0168-9525(01)02451-9   \n",
       "1      10.1007/s10549-009-0501-3   \n",
       "2  10.1126/science.297.5588.1837   \n",
       "3      10.1134/s102279541609012x   \n",
       "4       10.1177/1758834011417039   \n",
       "\n",
       "                                         fulltextUrl  \\\n",
       "0  https://linkinghub.elsevier.com/retrieve/pii/S...   \n",
       "1                                                      \n",
       "2                                                      \n",
       "3  https://link.springer.com/article/10.1134%2FS1...   \n",
       "4                                                      \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2018-11-14T16:29:09+0000          78    5258  0_00000  \n",
       "1  2018-11-14T16:29:16+0000        5048  100825  0_00000  \n",
       "2  2018-11-14T16:29:40+0000  1436851129  148950  0_00000  \n",
       "3  2018-11-14T16:30:43+0000  4458079378   22264  0_00000  \n",
       "4  2018-11-14T16:30:58+0000  6079002966   93229  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(f\"file:{crawl_path}/text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "print(\"{} articles loaded from the articles sqlite database\".format(articles.shape[0]))\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28938 article text blocks consuming 2.43 megabytes\n",
      "CPU times: user 28.1 s, sys: 3.32 s, total: 31.4 s\n",
      "Wall time: 31.4 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "articles = pd.DataFrame()\n",
    "for path in glob.glob(f\"{crawl_path}/text/*.files.gz\"):\n",
    "    articles = articles.append(pd.read_csv(path, sep=\"\\t\", encoding=\"utf-8\"))\n",
    "    \n",
    "print(\"{} article text blocks consuming {:.2f} megabytes\".format(\n",
    "    articles.shape[0], articles.memory_usage(index=True).sum() / 2**20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#fileId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>articleId</th>\n",
       "      <th>url</th>\n",
       "      <th>desc</th>\n",
       "      <th>fileType</th>\n",
       "      <th>time</th>\n",
       "      <th>mimeType</th>\n",
       "      <th>locFname</th>\n",
       "      <th>content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672001</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td>5011585672</td>\n",
       "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
       "      <td>supplemental file (.pdf)</td>\n",
       "      <td>main.pdf</td>\n",
       "      <td>2018-11-20T17:13:51+0000</td>\n",
       "      <td>application/pdf</td>\n",
       "      <td>/crawl/download/files/11585672.main.pdf</td>\n",
       "      <td>Review\u0007\u0007|\u0007\u0007A TRENDS Guide to Mouse Models of H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261001</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td>5019688261</td>\n",
       "      <td>https://link.springer.com/article/10.1007/s105...</td>\n",
       "      <td>supplemental file (.html)</td>\n",
       "      <td>main.html</td>\n",
       "      <td>2018-11-20T17:13:51+0000</td>\n",
       "      <td>text/html</td>\n",
       "      <td>/crawl/download/files/19688261.main.html</td>\n",
       "      <td>Breast Cancer Research and Treatment\u0007\u0007June 201...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5019688261002</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td>5019688261</td>\n",
       "      <td>https://link.springer.com/content/pdf/10.1007%...</td>\n",
       "      <td>supplemental file (.pdf)</td>\n",
       "      <td>main.pdf</td>\n",
       "      <td>2018-11-20T17:13:51+0000</td>\n",
       "      <td>application/pdf</td>\n",
       "      <td>/crawl/download/files/19688261.main.pdf</td>\n",
       "      <td>PRECLINICAL STUDY\u0007An integrative genomic and t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5012228710001</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td>5012228710</td>\n",
       "      <td>http://science.sciencemag.org/content/297/5588...</td>\n",
       "      <td>supplemental file (.html)</td>\n",
       "      <td>main.html</td>\n",
       "      <td>2018-11-20T17:13:51+0000</td>\n",
       "      <td>text/html</td>\n",
       "      <td>/crawl/download/files/12228710.main.html</td>\n",
       "      <td>Skip to main content\u0007\u0007Science Logo\u0007\u0007  * Home\u0007 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5012228710002</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td>5012228710</td>\n",
       "      <td>http://science.sciencemag.org/content/sci/297/...</td>\n",
       "      <td>supplemental file (.pdf)</td>\n",
       "      <td>main.pdf</td>\n",
       "      <td>2018-11-20T17:13:51+0000</td>\n",
       "      <td>application/pdf</td>\n",
       "      <td>/crawl/download/files/12228710.main.pdf</td>\n",
       "      <td>RESEARCH ARTICLES\u0007\u0007References and Notes\u0007\u00071. S....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         #fileId    externalId   articleId  \\\n",
       "0  5011585672001  PMID11585672  5011585672   \n",
       "1  5019688261001  PMID19688261  5019688261   \n",
       "2  5019688261002  PMID19688261  5019688261   \n",
       "3  5012228710001  PMID12228710  5012228710   \n",
       "4  5012228710002  PMID12228710  5012228710   \n",
       "\n",
       "                                                 url  \\\n",
       "0  https://api.elsevier.com/content/article/pii/S...   \n",
       "1  https://link.springer.com/article/10.1007/s105...   \n",
       "2  https://link.springer.com/content/pdf/10.1007%...   \n",
       "3  http://science.sciencemag.org/content/297/5588...   \n",
       "4  http://science.sciencemag.org/content/sci/297/...   \n",
       "\n",
       "                        desc   fileType                      time  \\\n",
       "0   supplemental file (.pdf)   main.pdf  2018-11-20T17:13:51+0000   \n",
       "1  supplemental file (.html)  main.html  2018-11-20T17:13:51+0000   \n",
       "2   supplemental file (.pdf)   main.pdf  2018-11-20T17:13:51+0000   \n",
       "3  supplemental file (.html)  main.html  2018-11-20T17:13:51+0000   \n",
       "4   supplemental file (.pdf)   main.pdf  2018-11-20T17:13:51+0000   \n",
       "\n",
       "          mimeType                                  locFname  \\\n",
       "0  application/pdf   /crawl/download/files/11585672.main.pdf   \n",
       "1        text/html  /crawl/download/files/19688261.main.html   \n",
       "2  application/pdf   /crawl/download/files/19688261.main.pdf   \n",
       "3        text/html  /crawl/download/files/12228710.main.html   \n",
       "4  application/pdf   /crawl/download/files/12228710.main.pdf   \n",
       "\n",
       "                                             content  \n",
       "0  Review\u0007\u0007|\u0007\u0007A TRENDS Guide to Mouse Models of H...  \n",
       "1  Breast Cancer Research and Treatment\u0007\u0007June 201...  \n",
       "2  PRECLINICAL STUDY\u0007An integrative genomic and t...  \n",
       "3  Skip to main content\u0007\u0007Science Logo\u0007\u0007  * Home\u0007 ...  \n",
       "4  RESEARCH ARTICLES\u0007\u0007References and Notes\u0007\u00071. S....  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 338 variants\n",
      "CPU times: user 170 ms, sys: 49 ms, total: 219 ms\n",
      "Wall time: 216 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "variants = set()\n",
    "pattern = re.compile(\"c\\.\\d+[atcgATCG]+>[atcgATCG]+\")\n",
    "for article in articles.iloc[0:1000].itertuples():\n",
    "    variants = variants.union(set(re.findall(pattern, article.content)))\n",
    "print(\"Found {} variants\".format(len(variants)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean\n",
    "https://machinelearningmastery.com/clean-text-machine-learning-python/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: You are using pip version 19.1.1, however version 19.2.3 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n",
      "\u001b[33mWARNING: You are using pip version 19.1.1, however version 19.2.3 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to /tf/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!pip3 install --user --upgrade --quiet unidecode\n",
    "import unidecode\n",
    "\n",
    "!pip3 install --user --upgrade --quiet nltk\n",
    "import nltk\n",
    "import nltk.tokenize\n",
    "import nltk.stem.porter\n",
    "# nltk.download(\"punkt\")\n",
    "nltk.download(\"stopwords\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['cancer', 'breast', 'introduct', 'genet', 'test', 'famili', 'multipl', 'case', 'breast', 'and/or', 'ovarian', 'cancer', 'often', 'target', 'youngest', 'affect', 'woman', 'index', 'case', 'famili', 'clinic', 'genet', 'test', 'context', 'larg', 'limit', 'brca1', 'brca2', 'gene', 'unless', 'addit', 'indic', 'present', 'recent', 'women', 'breast', 'cancer', 'test', 'uninform', 'identifi', 'clearli', 'pathogen', 'mutat', 'either', 'gene.1', 'mani', 'put', 'breast', 'cancer', 'suscept', 'gene', 'identifi', 'vari', 'level', 'evid', 'associ', 'breast', 'cancer', 'today', 'diagnost', 'test', 'facil', 'includ', 'larg', 'number', 'gene', 'singl', 'panel', 'test', 'use', 'massiv', 'parallel', 'next', 'gener', 'sequenc', 'consider', 'reduc', 'cost', 'howev', 'gene', 'panel', 'test', 'pose', 'consider', 'challeng', 'clinic', 'genet', 'servic', 'mani', 'gene', 'valid', 'breast', 'cancer', 'suscept', 'gene', 'even', 'risk', 'associ', 'differ', 'type', 'mutat', 'poorli', 'defined.2', 'test', 'use', 'estim', 'reduct', 'risk', 'mutat', 'carrier', 'known', 'estim', 'risk', 'cancer', 'non-mut', 'carrier', 'famili', 'gene', 'current', 'includ', 'commerci', 'breast', 'cancer', 'suscept', 'gene', 'panel', 'addit', 'brca1', 'brca2', 'vari', 'laboratori', 'compani', 'rang', 'reason', 'well-characteris', 'breast', 'cancer', 'predisposit', 'gene', 'tp53', 'palb2', 'atm', 'chek2', 'cdh1', 'pten', 'stk11', 'one', 'recur', 'mutat', 'nbn3', 'gene', 'limit', 'evid', 'mutat', 'confer', 'elev', 'breast', 'cancer', 'risk', 'eg', 'bard1', 'brip1', 'mre11', 'rad50', 'rad51c', 'rad51d', 'cdkn2a', 'xrcc2', '.2', 'gene', 'select', 'panel', 'appear', 'driven', 'commerci', 'interest', 'rather', 'strength', 'evid', 'role', 'breast', 'cancer', 'suscept', 'vital', 'verifi', 'breast', 'cancer', 'suscept', 'gene', 'magnitud', 'associ', 'risk', 'order', 'make', 'best', 'use', 'data', 'gene', 'panel', 'test', 'clinic', 'genet', 'servic', 'carrier', 'pathogen', 'mutat', 'breast', 'cancer', 'suscept', 'gene', 'brca1', 'brca2', 'intervent', 'includ', 'mastectomi', 'bilater', 'salpingo-oophorectomi', 'medic', 'use', 'reduc', 'risk', 'howev', 'surgic', 'option', 'invas', 'irrevers', 'risk', 'reduct', 'medic', 'may', 'unfavour', 'side', 'effect', 'clinician', 'must', 'robust', 'data', 'proven', 'util', 'use', 'genet', 'test', 'involv', 'new', 'suit', 'breast', 'cancer', 'suscept', 'gene', 'basi', 'risk', 'assess', 'risk', 'manag', 'clinician', 'also', 'need', 'understand', 'valu', 'neg', 'predict', 'test', 'famili', 'member', 'caus', 'gene', 'mutat', 'famili', 'establish', 'sequenc', '17', 'known', 'put', 'breast', 'cancer', 'suscept', 'gene', 'plu', 'brca1', 'brca2', '684', 'women', 'affect', 'breast/ovarian', 'cancer', 'strong', 'famili', 'histori', 'breast', 'cancer', 'protein-trunc', 'put', 'deleteri', 'missens', 'mutat', 'genotyp', 'avail', 'famili', 'member', 'assess', 'co-segreg', 'variant', 'diseas', 'famili', 'order', 'estim', 'breast', 'cancer', 'risk', 'associ', 'mutat', 'method', 'patient', 'cohort', 'select', '684', \"'non-brca1/2\", 'famili', 'kathleen', 'cuningham', 'foundat', 'consortium', 'research', 'famili', 'breast', 'cancer', 'kconfab', 'non-brca1/2', 'famili', 'recruit', 'kconfab', 'famili', 'cancer', 'clinic', 'fulfil', 'one', 'follow', 'criteria1', 'four', 'case', 'breast', 'ovarian', 'cancer', 'one', 'side', 'famili', 'least', 'two', 'live', 'affect', 'four', 'live', 'unaffect', 'famili', 'member', 'ii', 'three', 'case', 'breast', 'ovarian', 'cancer', 'one', 'side', 'famili', 'includ', 'least', 'one', 'high-risk', 'featur', 'male', 'breast', 'cancer', 'bilater', 'breast', 'cancer', 'breast', 'plu', 'ovarian', 'cancer', 'woman', 'breast', 'cancer', 'age', '40', 'least', 'two', 'live', 'affect', 'four', 'live', 'unaffect', 'famili', 'member', 'iii', 'potenti', 'high-risk', 'individu', 'fresh', 'tumour', 'avail', 'research', 'least', 'two', 'live', 'affect', 'four', 'live', 'unaffect', 'famili', 'member', 'select', '684', 'famili', 'kconfab', 'base', 'follow', 'criteria', 'known', 'pathogen', 'mutat', 'brca1', 'brca2', 'famili', 'member', 'time', 'select', 'ii', 'known', 'protein-trunc', 'mutat', 'palb2', 'atm', 'v2424g', 'mutat', 'famili', 'brca1/2', 'test', 'perform', 'prior', 'select', 'addit', 'prioritis', 'famili', 'select', 'basi', 'age', 'diagnosi', 'individu', 'sequenc', 'ii', 'includ', 'famili', 'case', 'pancreat', 'cancer', 'n=94', 'iii', 'famili', 'previous', 'sensit', 'complet', 'test', 'brca1', 'brca2', 'iv', 'avail', 'least', 'two', 'germlin', 'dna', 'sampl', 'relat', 'famili', 'member', 'rank', 'base', 'probabl', 'calcul', 'boadicea', 'carri', 'brca1/2', 'mutat', 'famili', 'select', 'youngest', 'breast', 'cancer', 'case', 'germlin', 'dna', 'avail', 'index', 'case', 'sequenc', 'target', 'sequenc', 'select', 'put', 'mutat', 'perform', 'custom-design', 'target', 'sequenc', 'cover', 'code', 'exon', 'brca1', 'brca2', 'tp53', 'palb2', 'atm', 'chek2', 'cdh1', 'pten', 'stk11', 'bard1', 'brip1', 'mre11', 'nbn', 'rad50', 'rad51c', 'rad51d', 'cdkn2a', 'cdk4', 'xrcc2', 'target', 'captur', 'use', 'agil', 'target', 'enrich', 'kit', 'axeq', 'technolog', 'accord', 'manufactur', \"'s\", 'instruct', 'sequenc', '100', 'bp', 'paired-end', 'read', 'illumina', 'hiseq', '2000', 'platform', 'high-qual', 'sequenc', 'data', 'averag', 'depth', '354.4-fold', 'rang', '24.61-1505.14', 'per', 'exon', 'obtain', 'sequenc', 'align', 'human', 'refer', 'genom', 'grch37', 'use', 'bwa.4', 'read', 'mark', 'pcr', 'optic', 'duplic', 'remov', 'consider', 'picard', 'http', '//picard.sourceforge.net', 'variant', 'call', 'target', 'exon', 'also', 'off-target', 'read', '25', 'nucleotid', 'intron', 'captur', 'potenti', 'splice-sit', 'mutat', 'single-nucleotid', 'variant', 'small', 'indel', 'call', 'gatk.5', 'call', 'elimin', 'possibl', 'fals', 'posit', 'variant', 'appli', 'strict', 'qualiti', 'control', 'filter', 'exclud', 'variant', '30', 'read', 'variant', 'refer', 'altern', 'read', 'ratio', '0.2', '5.0', 'use', 'annovar', 'http', '//annovar.openbioinformatics.org', 'annot', 'function', 'consequ', 'variant', 'well', 'retriev', 'allel', 'frequenc', 'public', 'databas', 'dbsnp138', '1000', 'genom', 'project', 'complet', 'genom', 'nhlbi-esp', '6500', 'exom', '.6', 'potenti', 'function', 'import', 'non-', 'synonym', 'variant', 'predict', 'follow', 'tool', 'sift,7', 'polyphen2,8', 'lrt,9', 'mutationtaster10', 'cadd.11', 'non-synonym', 'variant', 'describ', 'least', 'four', 'algorithm', 'damag', 'probabl', 'damag', 'cadd', 'score', '15', 'regard', 'put', 'deleteri', 'select', 'segreg', 'analysi', 'protein-trunc', 'mutat', 'within', 'last', 'exon', 'remov', 'candid', 'list', 'sinc', 'predict', 'activ', 'nonsense-medi', 'mrna', 'decay', 'impact', 'splice', 'predict', 'intron', 'variant', 'use', 'stand', 'alon', 'perl', 'script', 'maxent', 'http', '//genes.mit.edu/burgelab/maxent/xmaxentscan_scoreseq.html', 'famili', 'sequenc', 'segreg', 'analysi', 'segreg', 'analysi', 'put', 'pathogen', 'mutat', 'carri', 'either', 'sanger', 'sequenc', 'use', 'standard', 'sequenc', 'procedur', 'bigdy', 'termin', 'v.3.1', 'cycl', 'sequenc', 'kit', 'pe', 'appli', 'biosystem', 'pe', 'appli', 'biosystem', 'fluidigm', '96.96', 'dynam', 'array', 'integr', 'fluidic', 'circuit', 'biomark', 'hd', 'system', 'fluidigm', 'corpor', 'accord', 'manufactur', \"'s\", 'instruct', 'primer', 'sequenc', 'provid', 'upon', 'request', 'order', 'assess', 'associ', 'sequenc', 'variant', 'breast', 'cancer', 'perform', 'two', 'analys', 'base', 'maximis', 'likelihood', 'observ', 'pedigre', 'genotyp', 'condit', 'pedigre', 'phenotyp', 'genotyp', 'index', 'case', 'primari', 'analysi', 'calcul', 'penetr', 'breast', 'cancer', 'carrier', 'suscept', 'gene', 'assum', 'constant', 'rel', 'risk', 'age', 'second', 'analysi', 'assum', 'rel', 'risk', 'constant', 'multipl', 'antoni', 'et', 'al12', 'estim', 'brca2', 'calcul', 'cumul', 'penetr', 'trial', 'valu', 'multipli', 'allow', 'similar', 'pattern', 'age-specif', 'effect', 'brca2', 'requir', 'estim', 'singl', 'paramet', 'model', 'fit', 'maximum', 'likelihood', 'theori', 'use', 'modifi', 'version', 'linkag', 'genet', 'analysi', 'package.13', 'non-carri', 'assum', 'popul', 'risk', 'specif', 'australia', 'incid', 'rate', 'taken', 'cancer', 'registri', 'data', 'obtain', 'cancer', 'incid', 'five', 'contin', 'viii', 'iarc', 'lyon', 'risk', 'ratio', 'rr', 'age-specif', 'breast', 'cancer', 'incid', 'rate', 'carrier', 'divid', 'relev', 'popul', 'rate', 'estim', 'model', 'first', 'consid', 'report', 'case', 'includ', 'affect', 'restrict', 'analys', 'cancer', 'case', 'confirm', 'patholog', 'medic', 'record', 'death', 'certif', 'report', 'breast', 'cancer', 'unknown', 'age', 'diagnosi', 'exclud', 'analys', 'cancer', 'breast', 'includ', 'ovarian', 'cancer', 'treat', 'unaffect', 'age', 'cancer', 'diagnosi', 'perform', 'two', 'sensit', 'analys', 'specif', 'includ', 'ovarian', 'cancer', 'affect', 'weight', 'variant', 'predict', 'deleteri', 'likelihood', 'ratio', 'score', 'base', 'weight', 'function', '10', 'differ', 'silico', 'analys', 'well', 'minor', 'allel', 'frequenc', 'variant', 'weight', 'normalis', 'sum', 'observ', 'number', 'variant', 'gene', 'histopatholog', 'uropathologist', 'dc', 'review', 'one', 'repres', 'e-stain', 'section', 'formalin-fix', 'paraffin-embed', 'tissu', 'three', 'case', 'prostat', 'adenocarcinoma', 'result', '684', 'index', 'case', 'test', 'identifi', '11', 'pathogen', 'frameshift', 'canon', 'splice', 'site', 'brca1', 'mutat', '13', 'individu', '10', 'uniqu', 'brca2', 'mutat', '11', 'individu', 'none', '24', 'carrier', 'previou', 'brca1/2', 'test', 'remain', '660', 'non-brca1/2', 'index', 'case', 'found', '31', 'put', 'deleteri', 'mutat', 'protein-trunc', 'non-synonym', 'missens', 'known', 'breast', 'cancer', 'suscept', 'gene', '45', 'index', 'case', 'atm', 'n=8', 'includ', 'three', 'missens', 'cdh1', 'n=1', 'missens', 'chek2', 'n=14', 'famili', 'truncat', 'includ', 'six', 'uniqu', 'mutat', 'palb2', 'n=14', 'famili', 'eight', 'uniqu', 'mutat', 'includ', 'two', 'missens', 'pten', 'n=1', 'missens', 'stk11', 'n=1', 'probabl', 'splice', 'tp53', 'n=6', 'missens', 'tabl', 'also', 'identifi', '22', 'potenti', 'deleteri', 'mutat', '31', 'index', 'case', 'bard1', 'n=4', 'includ', 'two', 'missens', 'brip1', 'n=16', 'includ', 'seven', 'missens', 'cdk4', 'n=1', 'missens', 'mre11', 'n=4', 'missens', 'nbn', 'n=1', 'frameshift', 'rad50', 'n=1', 'missens', 'rad51c', 'n=1', 'missens', 'rad51d', 'n=3', 'includ', 'two', 'missens', 'none', 'cdkn2a', 'xrcc2', 'one', 'intron', 'variant', 'c.375-8c', 'stk11', 'includ', 'increas', 'maxent', 'score', '5.68', 'compar', 'canon', 'sequenc', 'consid', 'put', 'splice-sit', 'mutation.14', 'minimum', 'coverag', 'exon', 'sequenc', '25-fold', 'mean', '354-fold', 'make', 'unlik', 'miss', 'pathogen', 'mutat', '76', 'individu', 'carri', 'put', 'pathogen', 'mutat', 'gene', 'brca1', 'brca2', 'addit', '558', 'famili', 'member', 'avail', 'germlin', 'dna', 'segreg', 'analysi', '76', 'mutat', 'valid', 'index', 'case', 'view', 'tabl', 'view', 'inlin', 'view', 'popup', 'tabl', 'put', 'mutat', 'found', 'gene', 'brca1', 'brca2', 'primari', 'analysi', 'assum', 'constant', 'rel', 'risk', 'age', 'four', 'gene', 'cdh1', 'chek2', 'palb2', 'tp53', 'show', 'evid', 'significantli', 'increas', 'risk', 'breast', 'cancer', 'lower', 'ci', '1.0', 'analysi', 'includ', 'case', 'regardless', 'valid', 'report', 'diagnos', 'tabl', 'section', 'three', 'case', 'adenocarcinoma', 'prostat', 'occur', 'carrier', 'p.met246val', 'tp53', 'famili', '72', 'review', 'three', 'case', 'show', 'high-grad', 'carcinoma', 'gleason', 'score', '9.', 'two', 'case', 'associ', 'intraduct', 'carcinoma', 'prostat', 'idc-p', 'third', 'case', 'compon', 'larg', 'cribriform', 'carcinoma', 'although', 'idc-p', 'demonstr', 'section', 'examin', 'view', 'tabl', 'view', 'inlin', 'view', 'popup', 'tabl', 'risk', 'ratio', 'gene', 'examin', 'evid', 'associ', 'atm', 'breast', 'cancer', 'risk', 'rr=2.51', '95', 'ci', '0.95', '8.26', 'includ', 'cancer', 'case', 'rr=2.67', '95', 'ci', '0.82', '10.56', 'restrict', 'analysi', 'case', 'diagnosi', 'confirm', 'medic', 'record', 'cancer', 'registri', 'match', 'associ', 'signific', 'brip1', 'rr=0.47', '95', 'ci', '0.15', '1.18', 'clinic', 'meaning', 'increas', 'risk', 'twofold', 'could', 'exclud', 'given', 'upper', 'ci', 'limit', '1.18', 'inform', 'analys', 'use', 'case', 'consid', 'case', 'confirm', 'diagnosi', 'hr', '1.56', 'exclud', 'gene', 'set', 'associ', 'ovarian', 'cancer', 'risk.15-18', 'howev', 'includ', 'ovarian', 'cancer', 'analys', 'substanti', 'chang', 'result', 'larg', 'due', 'small', 'number', 'verifi', 'unverifi', 'ovarian', 'case', 'pedigre', 'also', 'prohibit', 'estim', 'ovarian-specif', 'risk', 'particular', 'includ', 'ovarian', 'cancer', 'affect', 'chang', 'estim', 'either', 'brip1', 'rad51c', 'data', 'shown', 'weight', 'probabl', 'pathogen', 'variant', 'use', 'composit', 'score', 'perch', 'suit', 'program', 'http', '//www.fengbjlab.org/perch', 'also', 'larg', 'effect', 'result', 'atm', 'exampl', 'weight', 'rr', '2.55', '95', 'ci', '0.95', '8.30', 'compar', 'unweight', 'rr', '2.51', '95', 'ci', '0.95', '8.26', 'discuss', 'sever', 'recent', 'public', 'describ', 'panel', 'test', 'known', 'put', 'breast', 'cancer', 'gene', 'early-onset', 'multiple-cas', 'breast', 'cancer', 'families,19-23', 'unselect', 'seri', 'triple-neg', 'breast', 'cancer', 'cases.24', 'howev', 'none', 'studi', 'access', 'multipl', 'dna', 'sampl', 'carrier', 'famili', 'segreg', 'analysi', 'therefor', 'unabl', 'estim', 'risk', 'associ', 'mutat', 'put', 'breast', 'cancer', 'suscept', 'gene', 'limit', 'conclus', 'frequenc', 'mutat', 'impact', 'find', 'clinic', 'care', 'test', '17', 'put', 'breast', 'cancer', 'suscept', 'gene', '684', 'index', 'case', 'strong', 'famili', 'histori', 'breast', 'cancer', '660', 'carri', 'brca1', 'brca2', 'mutat', 'make', 'one', 'largest', 'studi', 'gene', 'breast', 'cancer', 'famili', 'date', 'furthermor', 'abl', 'perform', 'segreg', 'analysi', '76', 'put', 'deleteri', 'mutat', 'found', 'larg', 'number', 'blood', 'sampl', 'mean', '7.3', 'per', 'famili', 'collect', 'kconfab', 'found', 'mutat', 'australasian', 'famili', 'chek2', 'palb2', 'tp53', 'confer', 'signific', 'risk', 'breast', 'cancer', 'also', 'found', 'support', 'moder', 'risk', 'breast', 'cancer', 'associ', 'mutat', 'atm', 'addit', 'found', 'singl', 'famili', 'p.asp777asn', 'cdh1', 'mutat', 'rr=13.62', '95', 'ci', '1.12', '24.0', 'suggest', 'cdh1', 'breast', 'cancer', 'suscept', 'gene', 'figur', 'found', 'support', 'evid', 'bard1', 'brip1', 'cdkn2a', 'cdk4', 'mre11', 'nbn', 'rad50', 'rad51c', 'rad51d', 'stk11', 'xrcc2', 'moderate-risk', 'high-risk', 'breast', 'cancer', 'suscept', 'gene', 'howev', 'note', 'gene', 'evalu', 'except', 'brip1', 'upper', 'limit', '95', 'ci', 'exclud', 'possibl', 'gene', 'confer', 'least', 'moder', 'risk', 'breast', 'cancer', 'two', '13', 'famili', 'identifi', 'missens', 'mutat', 'brip1', 'also', 'pathogen', 'mutat', 'brca2', 'download', 'figur', 'open', 'new', 'tab', 'download', 'powerpoint', 'download', 'figur', 'open', 'new', 'tab', 'download', 'powerpoint', 'figur', 'pedigre', 'famili', 'mutat', 'tp53', 'cdh1', 'nbm', 'rad51c', 'repres', 'pedigre', 'famili', 'palb2', 'p.trp1038*', 'neg', 'non-carri', 'po', 'mutat', 'carrier', 'recent', 'review', 'literature2', 'conclud', 'protein-trunc', 'mutat', 'brca1', 'brca2', 'probabl', 'palb2', 'perhap', 'also', 'pten', 'confer', 'high', 'risk', 'defin', 'fourfold', 'breast', 'cancer', 'small', 'minor', 'missens', 'variant', 'brca1', 'brca2', 'associ', 'high', 'risk', 'breast', 'cancer', 'although', 'point', 'estim', 'risk', 'associ', 'palb2', 'variant', 'less', 'threefold', '95', 'confid', 'limit', 'ninefold', 'estim', 'exclud', 'possibl', 'high-risk', 'breast', 'cancer', 'suscept', 'gene', 'although', 'risk', 'estim', 'consist', 'high', 'estim', 'previous', 'report', 'australian', 'families,25', '95', 'confid', 'limit', 'analys', 'overlap', 'similarli', '95', 'confid', 'limit', 'overlap', 'publish', 'meta-analysi', '3-9.4', '.26', 'addit', 'missens', 'protein-trunc', 'variant', 'tp53', 'associ', 'substanti', 'increas', 'risk', 'breast', 'cancer', 'protein-trunc', 'mutat', 'chek2', 'well', 'protein-trunc', 'missens', 'mutat', 'atm', 'consid', 'moder', 'risk', '2-4-fold', 'breast', 'cancer', 'suscept', 'gene', 'three', 'gene', 'stk11,27', 'cdh128', ',29', 'nbn3', 'clear', 'evid', 'associ', 'least', 'mutat', 'risk', 'estim', 'imprecis', 'describ', 'high-risk', 'moderate-risk', 'breast', 'cancer', 'suscept', 'gene', 'cdh1', 'stk11', 'find', 'mutat', 'famili', 'hereditari', 'diffus', 'gastric', 'cancer', 'peutz-jegh', 'syndrom', 'pj', 'respect', 'provid', 'syndrome-associ', 'mutat', 'risk', 'breast', 'cancer', 'well', 'establish', 'predict', 'genet', 'test', 'usual', 'proce', 'within', 'famili', 'howev', 'mutat', 'identifi', 'outsid', 'clinic', 'context', 'stk11', 'mutat', 'famili', 'breast', 'cancer', 'clinic', 'sign', 'pj', 'breast', 'cancer', 'risk', 'associ', 'mutat', 'unclear', 'valu', 'predict', 'test', 'remain', 'uncertain', 'howev', 'note', 'stk11', 'mutat', 'report', 'predict', 'pathogen', 'silico', 'analysi', 'function', 'conform', 'would', 'requir', 'confirm', 'cell', 'line', 'avail', 'splice', 'assay', 'nbn', 'data', 'extrem', 'limit', 'data', 'appli', 'one', 'specif', 'mutat', 'within', 'one', 'population.30', 'gene', 'describ', 'breast', 'cancer', 'suscept', 'gene', 'bard1', 'brip1', 'cdkn2a', 'men1', 'mlh1', 'mre11', 'msh2', 'msh6', 'nb1', 'pms2', 'rad50', 'rad51c', 'rad51d', 'xrcc2', 'review', 'found', 'insuffici', 'evid', 'describ', 'such.2', 'combin', 'data', 'conclud', 'clinic', 'valu', 'mutat', 'test', '11', 'gene', 'breast', 'cancer', 'famili', 'evid', 'mutat', 'confer', 'twofold', 'greater', 'risk', 'breast', 'cancer', 'risk', 'estim', 'atm', 'card1', 'brip1', 'mre11', 'palb2', 'rad51d', 'tp53', 'could', 'underestim', 'due', 'mixtur', 'pathogen', 'non-pathogen', 'variant', 'function', 'assay', 'missens', 'variant', 'need', 'identifi', 'like', 'pathogen', 'howev', 'weight', 'result', 'probabl', 'pathogen', 'result', 'chang', 'significantli', 'compar', 'unadjust', 'probabl', 'due', 'rel', 'small', 'rang', 'posterior', 'probabl', 'pathogen', 'caus', 'select', 'variant', 'predict', 'deleteri', 'subset', 'silico', 'algorithm', 'use', 'comput', 'weight', 'number', 'variant', 'gene', 'larg', 'enough', 'analys', 'data', 'use', 'heterogen', 'model', 'assum', 'variant', 'mixtur', 'confer', 'increas', 'risk', 'true', 'pathogen', 'variant', 'figur', 'show', 'six', 'pedigre', 'tp53', 'mutat', 'one', 'cdh1', 'mutat', 'one', 'breast', 'cancer', 'case', 'lobular', 'morpholog', 'illustr', 'palb2-mut', 'pedigre', 'well', 'rad51c', 'nbn', 'mutat', 'segreg', 'breast', 'cancer-affect', 'statu', 'review', 'avail', 'case', 'breast', 'cancer', 'cdh1', 'mutat', 'carrier', 'show', 'carcinoma', 'mix', 'ductal', 'lobular', 'featur', 'reduc', 'aberr', 'stain', 'e-cadherin', 'lobular', 'region', 'two', 'six', 'tp53', 'famili', 'famili', '74', '76', 'fulfil', 'criteria', 'li', 'fraumeni', 'syndrom', 'lf', 'one', 'famili', '73', 'appear', 'deriv', 'de', 'novo', 'mutat', 'five', 'six', 'missens', 'mutat', 'exclud', 'p.glu388ala', 'found', 'tp53', 'previous', 'identifi', 'somat', 'germlin', 'mutat', 'iarc', 'database,31', 'p.glu388ala', 'famili', '71', 'p.gly262ser', 'famili', '75', 'regard', 'unclassifi', 'variant', 'clin', 'var', 'http', '//www.ncbi.nlm.nih.gov/clinvar/', 'interestingli', 'four', 'breast', 'cancer', 'case', 'patholog', 'data', 'avail', 'er+', 'pr+', 'and/or', 'her2+', 'consist', 'report', 'masciari', 'et', 'al32', 'breast', 'tumour', 'lf', 'hormon', 'receptor', 'and/or', 'her2', 'posit', 'one', 'famili', 'famili', '72', 'contain', 'four', 'case', 'adenocarcinoma', 'prostat', 'carrier', 'oblig', 'carrier', 'tp53', 'mutat', 'p.met246val', 'diagnos', 'age', '51', '53', '53', '66.', 'although', 'prostat', 'cancer', 'recognis', 'featur', 'lf', 'famili', 'remark', 'perform', 'patholog', 'review', 'section', 'review', 'three', 'case', 'prostat', 'adenocarcinoma', 'occur', 'men', 'p.met246val', 'tp53', 'mutat', 'case', 'show', 'high', 'grade', 'gleason', 'score', 'carcinoma', 'one', 'case', 'idc-p.', 'interestingli', 'three', 'tumour', 'resembl', 'prostat', 'adeonocarcinoma', 'men', 'germlin', 'mutat', 'brca2', 'associ', 'poor', 'surviv', 'rates,33', 'suggest', 'analysi', 'tp53', 'mutat', 'statu', 'well', 'histopatholog', 'surviv', 'analys', 'would', 'warrant', 'prostat', 'adenocarcinoma', 'identifi', 'three', 'famili', 'rare', 'evolutionarili', 'unlik', 'missens', 'mutat', 'fat', 'kinas', 'domain', 'atm', 'p.ala2274thr', 'p.asp2720hi', 'p.val2424gli', 'predict', 'confer', 'increas', 'risk', 'breast', 'cancer.34', 'singl', 'famili', 'identifi', 'mutat', 'pten', 'p.arg142trp', 'clinic', 'sign', 'cowden', 'syndrom', 'major', 'mutat', 'cowden', 'syndrom', 'protein', 'truncating35', ',36', 'p.arg142trp', 'describ', 'unclassifi', 'variant', 'missens', 'variant', 'found', 'known', 'breast', 'cancer', 'suscept', 'gene', 'palb2', 'also', 'consid', 'unclassifi', 'variant', 'evid', 'present', 'associ', 'elev', 'breast', 'cancer', 'risk', 'conclus', 'panel', 'test', '17', 'known', 'put', 'breast', 'cancer', 'suscept', 'gene', '660', 'non-brca1/2', 'multipl', 'case', 'breast', 'cancer', 'famili', 'found', '10', 'mutat', 'clinic', 'notifi', 'australia', '--', 'one', 'missens', 'p.val2424gli', 'atm', 'five', 'famili', 'p.trp1038*', 'truncat', 'mutat', 'palb2', 'four', 'missens', 'mutat', 'tp53', '--', 'demonstr', 'clinic', 'valu', 'find', 'present', 'australian', 'clinic', 'agre', 'notifi', 'famili', 'palb2', 'c.3113g', 'p.trp1038*', 'mutat', 'found', 'breast', 'cancer', 'risk', 'associ', 'particular', 'mutat', 'firmli', 'establish', 'known', 'approach', 'attribut', 'brca2', 'risk', 'attribut', 'truncat', 'palb2', 'mutat', 'outsid', 'specif', 'context', 'clear', 'famili', 'may', 'notifi', 'mutat', 'gener', 'approach', 'australia', 'conserv', 'thu', 'far', 'famili', 'tp53', 'mutat', 'featur', 'typic', 'lf', 'other', 'would', 'detect', 'famili', 'histori', 'alon', 'famili', 'find', 'pathogen', 'tp53', 'mutat', 'allow', 'chang', 'manag', 'young', 'unaffect', 'femal', 'mutat', 'carrier', 'may', 'elect', 'risk-reduc', 'bilater', 'mastectomi', 'bilater', 'mastectomi', 'oppos', 'breast', 'conserv', 'treatment', 'diagnos', 'breast', 'cancer', 'futur', 'tp53', 'carrier', 'may', 'also', 'wish', 'particip', 'cancer', 'screen', 'programm', 'avail', 'research', 'studi', 'also', 'allow', 'possibl', 'famili', 'plan', 'use', 'prenat', 'test', 'preimplant', 'genet', 'diagnosi', 'final', 'famili', 'member', 'carri', 'caus', 'tp53', 'mutat', 'reliev', 'high', 'risk', 'multiorgan', 'cancer', 'offspr', 'famili', 'palb2', 'c.3113g', 'mutat', 'carrier', 'risk', 'breast', 'cancer', 'approxim', 'relat', 'brca2', 'gene', 'mutat', 'mutat', 'carrier', 'take', 'advantag', 'addit', 'breast', 'cancer', 'screen', 'includ', 'mri', 'risk-reduc', 'medic', 'surgeri', 'similar', 'manag', 'appli', 'women', 'famili', 'high-risk', 'atm', 'variant', 'p.val2424gli', 'although', 'high', 'ovarian', 'cancer', 'risk', 'associ', 'brca1/2', 'mutat', 'either', 'palb2', 'atm', 'women', 'famili', 'reassur', 'ovarian', 'cancer', 'risk', 'although', 'c.1100delc', 'mutat', 'chek2', 'confer', 'moder', 'risk', 'breast', 'cancer', 'clinic', 'util', 'test', 'famili', 'member', 'variant', 'unknown', 'overal', 'addit', 'panel', 'test', 'non-brca1/2', 'famili', 'provid', 'addit', 'import', 'new', 'clinic', 'inform', 'breast/ovarian', 'cancer', 'famili', 'panel', 'test', 'becom', 'increasingli', 'popular', 'limit', 'valu', 'gain', 'broad', 'approach', 'data', 'avail', 'guid', 'manag', 'larg', 'major', 'famili', 'whose', 'genet', 'suscept', 'resolv', 'panel', 'test', 'also', 'potenti', 'famili', 'overinterpret', 'clinic', 'relev', 'variant', 'less-well-valid', 'put', 'predisposit', 'gene', 'includ', 'panel', 'consequ', 'may', 'tempt', 'radic', 'measur', 'reduc', 'breast', 'cancer', 'risk', 'altern', 'famili', 'member', 'abandon', 'effect', 'cancer', 'risk', 'manag', 'option', 'carri', 'variant', 'carri', 'index', 'case', 'although', 'shown', 'segreg', 'analysi', 'potenti', 'determin', 'breast', 'cancer', 'risk', 'associ', 'mutat', 'known', 'put', 'breast', 'cancer', 'suscept', 'gene', 'larg', 'case-control', 'studi', 'and/or', 'larg', 'family-bas', 'studi', 'need', 'defin', 'risk', 'accur', 'acknowledg', 'thank', 'evelin', 'niedermayr', 'kconfab', 'research', 'nurs', 'staff', 'head', 'staff', 'famili', 'cancer', 'clinic', 'clinic', 'follow', 'studi', 'contribut', 'resourc', 'mani', 'famili', 'contribut', 'kconfab', 'refer', '1.', 'mann', 'gj', '2.', 'thorn', '3.', 'ballein', 'rl', '4.', 'butow', 'pn', '5.', 'clark', 'cl', '6.', 'edkin', '7.', 'evan', 'gm', '8.', 'fereday', '9.', 'haan', '10.', 'gatta', '11.', 'gile', 'gg', '12.', 'goldblatt', '13.', 'hopper', 'jl', '14.', 'kirk', '15.', 'leari', 'ja', '16.', 'lindeman', '17.', 'niedermayr', '18.', 'phillip', 'ka', '19.', 'picken', '20.', 'pupo', 'gm', '21.', 'saunder', '22.', 'scott', 'cl', '23.', 'spurdl', 'ab', '24.', 'suther', '25.', 'tucker', '26.', 'chenevix-trench', 'kathleen', 'cuningham', 'consortium', 'research', 'famili', 'breast', 'c.', 'analysi', 'cancer', 'risk', 'brca1', 'brca2', 'mutat', 'preval', 'kconfab', 'famili', 'breast', 'cancer', 'resourc', 'breast', 'cancer', 're', '2006', 'r12', 'doi:10.1186/bcr1377', 'openurlcrossrefpubm', '1.', 'easton', 'df', '2.', 'pharoah', 'pd', '3.', 'antoni', 'ac', '4.', 'tischkowitz', '5.', 'tavtigian', 'sv', '6.', 'nathanson', 'kl', '7.', 'devile', '8.', 'meindl', '9.', 'couch', 'fj', '10.', 'southey', '11.', 'goldgar', 'de', '12.', 'evan', 'dg', '13.', 'chenevix-trench', '14.', 'rahman', '15.', 'robson', '16.', 'domchek', 'sm', '17.', 'foulk', 'wd', 'gene-panel', 'sequenc', 'predict', 'breast-canc', 'risk', 'engl', 'med', '2015', '372:2243-57.', 'doi:10.1056/nejmsr1501341', 'openurlcrossrefpubm', '1.', 'zhang', '2.', 'zeng', '3.', 'liu', '4.', 'wei', 'signific', 'associ', 'nijmegen', 'breakag', 'syndrom', '657del5', 'polymorph', 'breast', 'cancer', 'risk', 'tumour', 'biol', '2013', '34:2753-7.', 'doi:10.1007/s13277-013-0830-z', 'openurlcrossrefpubm', '1.', 'li', '2.', 'durbin', 'fast', 'accur', 'short', 'read', 'align', 'burrows-wheel', 'transform', 'bioinformat', '2009', '25:1754-60.', 'doi:10.1093/bioinformatics/btp324', 'openurlabstract/fre', 'full', 'text', '1.', 'mckenna', '2.', 'hanna', '3.', 'bank', '4.', 'sivachenko', '5.', 'cibulski', '6.', 'kernytski', '7.', 'garimella', '8.', 'altshul', '9.', 'gabriel', '10.', 'dali', '11.', 'depristo', 'genom', 'analysi', 'toolkit', 'mapreduc', 'framework', 'analyz', 'next-', 'gener', 'dna', 'sequenc', 'data', 'genom', 're', '2010', '20:1297-303.', 'doi:10.1101/gr.107524.110', 'openurlabstract/fre', 'full', 'text', '1.', 'wang', '2.', 'li', '3.', 'hakonarson', 'annovar', 'function', 'annot', 'genet', 'variant', 'high-throughput', 'sequenc', 'data', 'nucleic', 'acid', 're', '2010', '38', 'e164', 'doi:10.1093/nar/gkq603', 'openurlabstract/fre', 'full', 'text', '1.', 'kumar', '2.', 'henikoff', '3.', 'ng', 'pc', 'predict', 'effect', 'code', 'non-synonym', 'variant', 'protein', 'function', 'use', 'sift', 'algorithm', 'nat', 'protoc', '2009', '4:1073-81.', 'doi:10.1038/nprot.2009.86', 'openurlcrossrefpubmedweb', 'scienc', '1.', 'adzhubei', 'ia', '2.', 'schmidt', '3.', 'peshkin', '4.', 'ramenski', '5.', 'gerasimova', '6.', 'bork', '7.', 'kondrashov', '8.', 'sunyaev', 'sr', 'method', 'server', 'predict', 'damag', 'missens', 'mutat', 'nat', 'method', '2010', '7:248-9.', 'doi:10.1038/nmeth0410-248', 'openurlcrossrefpubmedweb', 'scienc', '1.', 'san', 'luca', 'fa', '2.', 'wang', '3.', 'scheet', '4.', 'peng', 'integr', 'annot', 'analysi', 'genet', 'variant', 'next-gener', 'sequenc', 'studi', 'variant', 'tool', 'bioinformat', '2012', '28:421-2.', 'doi:10.1093/bioinformatics/btr667', 'openurlabstract/fre', 'full', 'text', '10', '1.', 'schwarz', 'jm', '2.', 'rodelsperg', '3.', 'schuelk', '4.', 'seelow', 'mutationtast', 'evalu', 'disease-caus', 'potenti', 'sequenc', 'alter', 'nat', 'method', '2010', '7:575-6.', 'doi:10.1038/nmeth0810-575', 'openurlcrossrefpubmedweb', 'scienc', '11', '1.', 'kircher', '2.', 'witten', 'dm', '3.', 'jain', '4.', \"o'roak\", 'bj', '5.', 'cooper', 'gm', '6.', 'shendur', 'gener', 'framework', 'estim', 'rel', 'pathogen', 'human', 'genet', 'variant', 'nat', 'genet', '2014', '46:310-15.', 'doi:10.1038/ng.2892', 'openurlcrossrefpubm', '12', '1.', 'antoni', '2.', 'pharoah', 'pd', '3.', 'narod', '4.', 'risch', 'ha', '5.', 'eyfjord', 'je', '6.', 'hopper', 'jl', '7.', 'loman', '8.', 'olsson', '9.', 'johannsson', '10.', 'borg', '11.', 'pasini', '12.', 'radic', '13.', 'manoukian', '14.', 'eccl', 'dm', '15.', 'tang', '16.', 'olah', '17.', 'anton-culv', '18.', 'warner', '19.', 'lubinski', '20.', 'gronwald', '21.', 'gorski', '22.', 'tuliniu', '23.', 'thorlaciu', '24.', 'eerola', '25.', 'nevanlinna', '26.', 'syrjakoski', '27.', 'kallioniemi', 'op', '28.', 'thompson', '29.', 'evan', '30.', 'peto', '31.', 'lalloo', '32.', 'evan', 'dg', '33.', 'easton', 'df', 'averag', 'risk', 'breast', 'ovarian', 'cancer', 'associ', 'brca1', 'brca2', 'mutat', 'detect', 'case', 'seri', 'unselect', 'famili', 'histori', 'combin', 'analysi', '22', 'studi', 'hum', 'genet', '2003', '72:1117-30.', 'doi:10.1086/375033', 'openurlcrossrefpubmedweb', 'scienc', '13', '1.', 'lathrop', 'gm', '2.', 'lalouel', 'jm', '3.', 'julier', '4.', 'ott', 'strategi', 'multilocu', 'linkag', 'analysi', 'human', 'proc', 'natl', 'acad', 'sci', 'usa', '1984', '81:3443-6.', 'doi:10.1073/pnas.81.11.3443', 'openurlabstract/fre', 'full', 'text', '14', '1.', 'eng', '2.', 'coutinho', '3.', 'naha', '4.', 'yeo', '5.', 'tanouy', '6.', 'babaei', '7.', 'dork', '8.', 'burg', '9.', 'gatti', 'ra', 'nonclass', 'splice', 'mutat', 'code', 'noncod', 'region', 'atm', 'gene', 'maximum', 'entropi', 'estim', 'splice', 'junction', 'strength', 'hum', 'mutat', '2004', '23:67-76.', 'doi:10.1002/humu.10295', 'openurlcrossrefpubmedweb', 'scienc', '15', '1.', 'eccl', '2.', 'evan', 'dg', '3.', 'gore', '4.', 'renwick', '5.', 'seal', '6.', 'antoni', 'ac', '7.', 'rahman', '1.', 'loveday', '2.', 'turnbul', '3.', 'ruark', '4.', 'xicola', 'rm', '5.', 'ramsay', '6.', 'hugh', '7.', 'warren-perri', '8.', 'snape', 'breast', 'cancer', 'suscept', 'collabor', 'uk', 'eccl', 'evan', 'dg', 'gore', 'renwick', 'seal', 'antoni', 'ac', 'rahman', 'n.', 'germlin', 'rad51c', 'mutat', 'confer', 'suscept', 'ovarian', 'cancer', 'nat', 'genet', '2012', '44:475-6', 'author', 'repli', '6.', 'doi:10.1038/ng.2224', 'openurlcrossrefpubm', '16', '1.', 'eccl', '2.', 'evan', 'dg', '3.', 'renwick', '4.', 'seal', '5.', 'lord', 'cj', '6.', 'ashworth', '7.', 'reis-filho', 'js', '8.', 'antoni', 'ac', '9.', 'rahman', '1.', 'loveday', '2.', 'turnbul', '3.', 'ramsay', '4.', 'hugh', '5.', 'ruark', '6.', 'frankum', 'jr', '7.', 'bowden', '8.', 'kalmyrzaev', '9.', 'warren-perri', '10.', 'snape', '11.', 'adlard', 'jw', '12.', 'barwel', '13.', 'berg', '14.', 'bradi', 'af', '15.', 'brewer', '16.', 'brice', '17.', 'chapman', '18.', 'cook', '19.', 'davidson', '20.', 'donaldson', '21.', 'dougla', '22.', 'greenhalgh', '23.', 'henderson', '24.', 'izatt', '25.', 'kumar', '26.', 'lalloo', '27.', 'miedzybrodzka', '28.', 'morrison', 'pj', '29.', 'paterson', '30.', 'porteou', '31.', 'roger', 'mt', '32.', 'shanley', '33.', 'walker', 'breast', 'cancer', 'suscept', 'collabor', 'uk', 'eccl', 'evan', 'dg', 'renwick', 'seal', 'lord', 'cj', 'ashworth', 'reis-filho', 'js', 'antoni', 'ac', 'rahman', 'n.', 'germlin', 'mutat', 'rad51d', 'confer', 'suscept', 'ovarian', 'cancer', 'nat', 'genet', '2011', '43:879-82.', 'doi:10.1038/ng.893', 'openurlcrossrefpubm', '17', '1.', 'easton', 'df', '2.', 'stratton', 'mr', '3.', 'rahman', '1.', 'seal', '2.', 'thompson', '3.', 'renwick', '4.', 'elliott', '5.', 'kelli', '6.', 'barfoot', '7.', 'chagtai', '8.', 'jayatilak', '9.', 'ahm', '10.', 'spanova', '11.', 'north', '12.', 'mcguffog', '13.', 'evan', 'dg', '14.', 'eccl', 'breast', 'cancer', 'suscept', 'collabor', 'uk', 'easton', 'df', 'stratton', 'mr', 'rahman', 'n.', 'truncat', 'mutat', 'fanconi', 'anemia', 'gene', 'brip1', 'low-', 'penetr', 'breast', 'cancer', 'suscept', 'allel', 'nat', 'genet', '2006', '38:1239-41.', 'doi:10.1038/ng1902', 'openurlcrossrefpubmedweb', 'scienc', '18', '1.', 'rafnar', '2.', 'gudbjartsson', 'df', '3.', 'sulem', '4.', 'jonasdottir', '5.', 'sigurdsson', '6.', 'jonasdottir', '7.', 'besenbach', '8.', 'lundin', '9.', 'stacey', 'sn', '10.', 'gudmundsson', '11.', 'magnusson', 'ot', '12.', 'le', 'roux', '13.', 'orlygsdottir', '14.', 'helgadottir', 'ht', '15.', 'johannsdottir', '16.', 'gylfason', '17.', 'tryggvadottir', '18.', 'jonasson', 'jg', '19.', 'de', 'juan', '20.', 'ortega', '21.', 'ramon-caj', 'jm', '22.', 'garcia-prat', 'md', '23.', 'mayordomo', '24.', 'panadero', '25.', 'rivera', '26.', 'aben', 'kk', '27.', 'van', 'altena', '28.', 'massug', 'lf', '29.', 'aavikko', '30.', 'kujala', 'pm', '31.', 'staff', '32.', 'aaltonen', 'la', '33.', 'olafsdottir', '34.', 'bjornsson', '35.', 'kong', '36.', 'salvarsdottir', '37.', 'saemundsson', '38.', 'olafsson', '39.', 'benediktsdottir', 'kr', '40.', 'gulcher', '41.', 'masson', '42.', 'kiemeney', 'la', '43.', 'mayordomo', 'ji', '44.', 'thorsteinsdottir', '45.', 'stefansson', 'mutat', 'brip1', 'confer', 'high', 'risk', 'ovarian', 'cancer', 'nat', 'genet', '2011', '43:1104-7.', 'doi:10.1038/ng.955', 'openurlcrossrefpubm', '19', '1.', 'tung', '2.', 'battelli', '3.', 'allen', '4.', 'kaldat', '5.', 'bhatnagar', '6.', 'bowl', '7.', 'timm', '8.', 'garber', 'je', '9.', 'herold', '10.', 'ellisen', '11.', 'krejdovski', '12.', 'deleonardi', '13.', 'sedgwick', '14.', 'solti', '15.', 'roa', '16.', 'wenstrup', 'rj', '17.', 'hartman', 'ar', 'frequenc', 'mutat', 'individu', 'breast', 'cancer', 'refer', 'brca1', 'brca2', 'test', 'use', 'next-gener', 'sequenc', '25-gene', 'panel', 'cancer', '2015', '121:25-33.', 'doi:10.1002/cncr.29010', 'openurlcrossrefpubm', '20', '1.', 'kurian', 'aw', '2.', 'hare', 'ee', '3.', 'mill', '4.', 'kingham', 'ke', '5.', 'mcpherson', '6.', 'whittemor', '7.', 'mcguir', '8.', 'ladabaum', '9.', 'kobayashi', '10.', 'lincoln', 'se', '11.', 'cargil', '12.', 'ford', 'jm', 'clinic', 'evalu', 'multiple-gen', 'sequenc', 'panel', 'hereditari', 'cancer', 'risk', 'assess', 'clin', 'oncol', '2014', '32:2001-9.', 'doi:10.1200/jco.2013.53.6607', 'openurlabstract/fre', 'full', 'text', '21', '1.', 'maxwel', 'kn', '2.', 'wubbenhorst', '3.', \"d'andrea\", '4.', 'garman', '5.', 'long', 'jm', '6.', 'power', '7.', 'rathbun', '8.', 'stopfer', 'je', '9.', 'zhu', '10.', 'bradburi', 'ar', '11.', 'simon', 'ms', '12.', 'demichel', '13.', 'domchek', 'sm', '14.', 'nathanson', 'kl', 'preval', 'mutat', 'panel', 'breast', 'cancer', 'suscept', 'gene', 'brca1/2-neg', 'patient', 'early-onset', 'breast', 'cancer', 'genet', 'med', '2015', '17:630-8.', 'doi:10.1038/gim.2014.176', 'openurlcrossrefpubm', '22', '1.', 'castera', '2.', 'krieger', '3.', 'rousselin', '4.', 'legro', '5.', 'baumann', 'jj', '6.', 'bruet', '7.', 'brault', '8.', 'fouillet', '9.', 'goardon', '10.', 'letac', '11.', 'baert-desurmont', '12.', 'tinat', '13.', 'bera', '14.', 'dugast', '15.', 'berthet', '16.', 'polycarp', '17.', 'layet', '18.', 'hardouin', '19.', 'frebourg', '20.', 'vaur', 'next-gener', 'sequenc', 'diagnosi', 'hereditari', 'breast', 'ovarian', 'cancer', 'use', 'genom', 'captur', 'target', 'multipl', 'candid', 'gene', 'eur', 'hum', 'genet', '2014', '22:1305-13.', 'doi:10.1038/ejhg.2014.16', 'openurlcrossrefpubm', '23', '1.', 'minion', 'le', '2.', 'dolinski', 'js', '3.', 'chase', 'dm', '4.', 'dunlop', 'cl', '5.', 'chao', 'ec', '6.', 'monk', 'bj', 'hereditari', 'predisposit', 'ovarian', 'cancer', 'look', 'beyond', 'brca1/brca2', 'gynecol', 'oncol', '2015', '137:86-92.', 'doi:10.1016/j.ygyno.2015.01.537', 'openurlpubm', '24', '1.', 'couch', 'fj', '2.', 'hart', 'sn', '3.', 'sharma', '4.', 'toland', 'ae', '5.', 'wang', '6.', 'miron', '7.', 'olson', 'je', '8.', 'godwin', 'ak', '9.', 'pankratz', 'vs', '10.', 'olswold', '11.', 'slettedahl', '12.', 'hallberg', '13.', 'guidugli', '14.', 'davila', 'ji', '15.', 'beckmann', 'mw', '16.', 'janni', '17.', 'rack', '18.', 'ekici', 'ab', '19.', 'slamon', 'dj', '20.', 'konstantopoul', '21.', 'fostira', '22.', 'vratimo', '23.', 'fountzila', '24.', 'pelttari', 'lm', '25.', 'tapper', 'wj', '26.', 'durcan', '27.', 'cross', 'ss', '28.', 'pilarski', '29.', 'shapiro', 'cl', '30.', 'klemp', '31.', 'yao', '32.', 'garber', '33.', 'cox', '34.', 'brauch', '35.', 'ambroson', '36.', 'nevanlinna', '37.', 'yannoukako', '38.', 'slager', 'sl', '39.', 'vachon', 'cm', '40.', 'eccl', 'dm', '41.', 'fasch', 'pa', 'inherit', 'mutat', '17', 'breast', 'cancer', 'suscept', 'gene', 'among', 'larg', 'triple-neg', 'breast', 'cancer', 'cohort', 'unselect', 'famili', 'histori', 'breast', 'cancer', 'clin', 'oncol', '2015', '33:304-11.', 'doi:10.1200/jco.2014.57.1414', 'openurlabstract/fre', 'full', 'text', '25', '1.', 'southey', 'mc', '2.', 'teo', 'zl', '3.', 'dowti', 'jg', '4.', 'odefrey', 'fa', '5.', 'park', 'dj', '6.', 'tischkowitz', '7.', 'sabbaghian', '8.', 'apicella', '9.', 'byrn', 'gb', '10.', 'winship', '11.', 'baglietto', '12.', 'gile', 'gg', '13.', 'goldgar', 'de', '14.', 'foulk', 'wd', '15.', 'hopper', 'jl', 'kconfab', 'beast', 'cancer', 'famili', 'r.', 'palb2', 'mutat', 'associ', 'high', 'risk', 'breast', 'cancer', 'breast', 'cancer', 're', '2010', '12', 'r109', 'doi:10.1186/bcr2796', 'openurlcrossrefpubm', '26', '1.', 'antoni', 'ac', '2.', 'foulk', 'wd', '3.', 'tischkowitz', 'breast-canc', 'risk', 'famili', 'mutat', 'palb2', 'engl', 'med', '2014', '371:1651-2.', 'doi:10.1056/nejmc1410673', 'openurlpubm', '27', '1.', 'hearl', '2.', 'schumach', '3.', 'menko', 'fh', '4.', 'olschwang', '5.', 'boardman', 'la', '6.', 'gill', 'jj', '7.', 'keller', 'jj', '8.', 'westerman', '9.', 'scott', 'rj', '10.', 'lim', '11.', 'trimbath', 'jd', '12.', 'giardiello', 'fm', '13.', 'gruber', 'sb', '14.', 'offerhau', 'gj', '15.', 'de', 'rooij', 'fw', '16.', 'wilson', 'jh', '17.', 'hansmann', '18.', 'moslein', '19.', 'royer-pokora', '20.', 'vogel', '21.', 'phillip', 'rk', '22.', 'spigelman', 'ad', '23.', 'houlston', 'rs', 'frequenc', 'spectrum', 'cancer', 'peutz-jegh', 'syndrom', 'clin', 'cancer', 're', '2006', '12:3209-15.', 'doi:10.1158/1078-0432.ccr-06-0083', 'openurlabstract/fre', 'full', 'text', '28', '1.', 'pharoah', 'pd', '2.', 'guilford', '3.', 'calda', 'intern', 'gastric', 'cancer', 'linkag', 'c.', 'incid', 'gastric', 'cancer', 'breast', 'cancer', 'cdh1', 'e-cadherin', 'mutat', 'carrier', 'hereditari', 'diffus', 'gastric', 'cancer', 'famili', 'gastroenterolog', '2001', '121:1348-53.', 'doi:10.1053/gast.2001.29611', 'openurlcrossrefpubmedweb', 'scienc', '29', '1.', 'hansford', '2.', 'kaurah', '3.', 'li-chang', '4.', 'woo', '5.', 'senz', '6.', 'pinheiro', '7.', 'schrader', 'ka', '8.', 'schaeffer', 'df', '9.', 'shumanski', '10.', 'zogopoulo', '11.', 'santo', 'ta', '12.', 'claro', '13.', 'carvalho', '14.', 'nielsen', '15.', 'padilla', '16.', 'lum', '17.', 'talhouk', '18.', 'baker-lang', '19.', 'richardson', '20.', 'lewi', '21.', 'lindor', 'nm', '22.', 'pennel', '23.', 'macmillan', '24.', 'fernandez', '25.', 'keller', '26.', 'lynch', '27.', 'shah', 'sp', '28.', 'guilford', '29.', 'galling', '30.', 'corso', '31.', 'roviello', '32.', 'calda', '33.', 'oliveira', '34.', 'pharoah', 'pd', '35.', 'huntsman', 'dg', 'hereditari', 'diffus', 'gastric', 'cancer', 'syndrom', 'cdh1', 'mutat', 'beyond', 'jama', 'oncol', '2015', '1:23-32.', 'doi:10.1001/jamaoncol.2014.168', 'openurlcrossrefpubm', '30', '1.', 'steffen', '2.', 'nowakowska', '3.', 'niwinska', '4.', 'czapczak', '5.', 'kluska', '6.', 'piatkowska', '7.', 'wisniewska', '8.', 'paszko', 'germlin', 'mutat', '657del5', 'nbs1', 'gene', 'contribut', 'significantli', 'incid', 'breast', 'cancer', 'central', 'poland', 'int', 'cancer', '2006', '119:472-5.', 'doi:10.1002/ijc.21853', 'openurlcrossrefpubmedweb', 'scienc', '31', '1.', 'petitjean', '2.', 'math', '3.', 'kato', '4.', 'ishioka', '5.', 'tavtigian', 'sv', '6.', 'hainaut', '7.', 'olivi', 'impact', 'mutant', 'p53', 'function', 'properti', 'tp53', 'mutat', 'pattern', 'tumor', 'phenotyp', 'lesson', 'recent', 'develop', 'iarc', 'tp53', 'databas', 'hum', 'mutat', '2007', '28:622-9.', 'doi:10.1002/humu.20495', 'openurlcrossrefpubmedweb', 'scienc', '32', '1.', 'masciari', '2.', 'dillon', 'da', '3.', 'rath', '4.', 'robson', '5.', 'weitzel', 'jn', '6.', 'balmana', '7.', 'gruber', 'sb', '8.', 'ford', 'jm', '9.', 'euhu', '10.', 'lebensohn', '11.', 'telli', '12.', 'pochebit', 'sm', '13.', 'lypa', '14.', 'garber', 'je', 'breast', 'cancer', 'phenotyp', 'women', 'tp53', 'germlin', 'mutat', 'li-fraumeni', 'syndrom', 'consortium', 'effort', 'breast', 'cancer', 're', 'treat', '2012', '133:1125-30.', 'doi:10.1007/s10549-012-1993-9', 'openurlcrossrefpubmedweb', 'scienc', '33', '1.', 'risbridg', 'gp', '2.', 'taylor', 'ra', '3.', 'clouston', '4.', 'sliwinski', '5.', 'thorn', '6.', 'hunter', '7.', 'li', '8.', 'mitchel', '9.', 'murphi', '10.', 'frydenberg', '11.', 'pook', '12.', 'pedersen', '13.', 'toivanen', '14.', 'wang', '15.', 'papargiri', '16.', 'lawrenc', 'mg', '17.', 'bolton', 'dm', 'patient-deriv', 'xenograft', 'reveal', 'intraduct', 'carcinoma', 'prostat', 'promin', 'patholog', 'brca2', 'mutat', 'carrier', 'prostat', 'cancer', 'correl', 'poor', 'prognosi', 'eur', 'urol', '2015', '67:496-503.', 'doi:10.1016/j.eururo.2014.08.007', 'openurlcrossrefpubm', '34', '1.', 'sangrajrang', '2.', 'hopper', 'jl', '3.', 'southey', 'mc', '4.', 'andruli', 'il', '5.', 'john', 'em', '6.', 'chenevix-trench', '1.', 'tavtigian', 'sv', '2.', 'oefner', 'pj', '3.', 'babikyan', '4.', 'hartmann', '5.', 'healey', '6.', 'le', 'calvez-kelm', '7.', 'lesueur', '8.', 'byrn', 'gb', '9.', 'chuang', 'sc', '10.', 'forey', '11.', 'feuchting', '12.', 'gioia', '13.', 'hall', '14.', 'hashib', '15.', 'hert', '16.', 'mckay-chopin', '17.', 'thoma', '18.', 'valle', 'mp', '19.', 'voegel', '20.', 'webb', 'pm', '21.', 'whiteman', 'dc', 'australian', 'cancer', 'breast', 'cancer', 'famili', 'kathleen', 'cuningham', 'foundat', 'consortium', 'research', 'famili', 'aspect', 'breast', 'sangrajrang', 'hopper', 'jl', 'southey', 'mc', 'andruli', 'il', 'john', 'em', 'chenevix-trench', 'g.', 'rare', 'evolutionarili', 'unlik', 'missens', 'substitut', 'atm', 'confer', 'increas', 'risk', 'breast', 'cancer', 'hum', 'genet', '2009', '85:427-46.', 'doi:10.1016/j.ajhg.2009.08.018', 'openurlcrossrefpubmedweb', 'scienc', '35', '1.', 'tan', 'mh', '2.', 'mester', 'jl', '3.', 'ngeow', '4.', 'rybicki', 'la', '5.', 'orloff', 'ms', '6.', 'eng', 'lifetim', 'cancer', 'risk', 'individu', 'germlin', 'pten', 'mutat', 'clin', 'cancer', 're', '2012', '18:400-7.', 'doi:10.1158/1078-0432.ccr-11-2283', 'openurlabstract/fre', 'full', 'text', '36', '1.', 'wang', '2.', 'karikomi', '3.', 'naidu', '4.', 'rajmohan', '5.', 'caserta', '6.', 'chen', 'hz', '7.', 'rawahneh', '8.', 'moffitt', '9.', 'stephen', 'ja', '10.', 'fernandez', 'sa', '11.', 'weinstein', '12.', 'wang', '13.', 'sade', '14.', 'la', 'perl', '15.', 'stromberg', '16.', 'rosol', 'tj', '17.', 'eng', '18.', 'ostrowski', 'mc', '19.', 'leon', 'allele-specif', 'tumor', 'spectrum', 'pten', 'knockin', 'mice', 'proc', 'natl', 'acad', 'sci', 'usa', '2010', '107:5142-7.', 'doi:10.1073/pnas.0912524107', 'openurlabstract/fre', 'full', 'text']\n"
     ]
    }
   ],
   "source": [
    "def clean(text):\n",
    "    # Convert to ascii and replace returns with spaces and lower case\n",
    "    text = re.sub(r\"[\\x07]+\", ' ', unidecode.unidecode(text)).lower()\n",
    "    \n",
    "    # split into words\n",
    "    words = nltk.tokenize.word_tokenize(text)\n",
    "    \n",
    "    # Drop all single character words and stop words - effectively removes punctuation\n",
    "    words = [w for w in words if len(w) > 1 and w not in stop_words]\n",
    "    \n",
    "    # Stem (lowercase's as a side affect as well)\n",
    "    porter = nltk.stem.porter.PorterStemmer()\n",
    "    words = [porter.stem(w) for w in words]\n",
    "    \n",
    "    return words\n",
    "\n",
    "print(clean(articles.iloc[42].content))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 20.8 s, sys: 41.5 ms, total: 20.8 s\n",
      "Wall time: 20.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "texts = [clean(article.content) for article in articles.iloc[0:100].itertuples()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRECLINICAL STUDY\u0007An integrative genomic and transcriptomic analysis reveals\u0007molecular pathways and networks regulated by copy number\u0007aberrations in basal-like, HER2 and luminal cancers\u0007Rachael Natrajan Æ Britta Weigelt Æ Alan Mackay Æ Felipe C. Geyer Æ\u0007Anita Grigoriadis Æ David S. P. Tan Æ Chris Jones Æ Christopher J. Lord Æ\u0007Radost Vatcheva Æ Socorro M. Rodriguez-Pinilla Æ Jose Palacios Æ\u0007Alan Ashworth Æ Jorge S. Reis-Filho\u0007Received: 24 July 2009 / Accepted: 28 July 2009 / Published online: 18 August 2009\u0007  Springer Science+Business Media, LLC. 2009\u0007Abstract Breast cancer is a heterogeneous disease\u0007caused by the accumulation of genetic changes in neo-\u0007plastic cells. We hypothesised that molecular subtypes of\u0007breast cancer may be driven by specific constellations of\u0007genes whose expression is regulated by gene copy\u0007number aberrations. To address this question, we ana-\u0007lysed a series of 48 microdissected grade III ductal\u0007carcinomas using high resolution microarray comparative\u0007genomic hybridisation and mRNA expression arrays.\u0007There were 5,931 genes whose expression significantly\u0007correlates with copy number identified; out of these,\u00071,897 genes were significantly differentially expressed\u0007between basal-like, HER2 and luminal tumours. Inge-\u0007nuity Pathway Analysis (IPA) revealed that ‘G1/S cell\u0007cycle regulation’ and ‘BRCA1 in DNA damage control’\u0007pathways were significantly enriched for genes whose\u0007expression correlates with copy number and are differ-\u0007entially expressed between the molecular subtypes of\u0007breast cancer. IPA of genes whose expression signifi-\u0007cantly correlates with copy number in each molecular\u0007subtype individually revealed that canonical pathways\u0007involved in oestrogen receptor (ER) signalling and DNA\u0007repair are enriched for these genes. We also identified\u000732, 157 and 265 genes significantly overexpressed when\u0007amplified in basal-like, HER2 and luminal cancers,\u0007respectively. These lists include known and novel\u0007potential therapeutic targets (e.g. HER2 and PPM1D in\u0007HER2 cancers). Our results provide strong circumstantial\u0007evidence that different patterns of genetic aberrations in\u0007distinct molecular subtypes of breast cancer contribute to\u0007their specific transcriptomic profiles and that biological\u0007phenomena characteristic of each subtype (e.g. prolifer-\u0007ation, HER2 and ER signalling) may be driven by spe-\u0007cific patterns of copy number aberrations.\u0007Keywords Breast cancer   Molecular subtypes  \u0007Copy number   Gene expression   Data integration  \u0007Molecular pathways\u0007Rachael Natrajan and Britta Weigelt contributed equally to the\u0007present study.\u0007Electronic supplementary material The online version of this\u0007article (doi:10.1007/s10549-009-0501-3) contains supplementary\u0007material, which is available to authorized users.\u0007R. Natrajan (&)   A. Mackay   F. C. Geyer  \u0007D. S. P. Tan   C. J. Lord   R. Vatcheva   A. Ashworth  \u0007J. S. Reis-Filho (&)\u0007The Breakthrough Breast Cancer Research Centre, Institute\u0007of Cancer Research, 237 Fulham Road, London SW3 6JB, UK\u0007e-mail: Rachael.Natrajan@icr.ac.uk\u0007J. S. Reis-Filho\u0007e-mail: Jorge.Reis-Filho@icr.ac.uk\u0007B. Weigelt\u0007Cancer Research UK, London Research Institute,\u0007London WC2A 3PX, UK\u0007A. Grigoriadis\u0007Breakthrough Breast Research Unit, Guy’s Hospital,\u0007London SE1 9RT, UK\u0007C. Jones\u0007Paediatric Oncology, Institute of Cancer Research,\u0007Sutton SM2 5NG, UK\u0007S. M. Rodriguez-Pinilla\u0007Centro Nacional de Investigaciones Oncologicas (CNIO),\u0007Madrid, Spain\u0007J. Palacios\u0007Servicio de Anatomia Patologica, Hospital Virgen del Rocı́o,\u0007Seville, Spain\u0007123\u0007Breast Cancer Res Treat (2010) 121:575–589\u0007DOI 10.1007/s10549-009-0501-3\u0007Introduction\u0007Breast cancer is a heterogeneous disease comprising multi-\u0007ple entities, which have distinct biological features and\u0007clinical behaviour. In recent years, it has become apparent\u0007that breast cancer heterogeneity may be underpinned by\u0007distinct genomic changes [1–7]. Microarray-based gene\u0007expression profiling of breast cancer has led to the identifi-\u0007cation of at least five distinct classes of breast cancers:\u0007luminal A, luminal B, HER2, normal breast-like and basal-\u0007like breast cancer. It has been hypothesised that the differ-\u0007ences between these molecular subgroups of breast cancer\u0007stem from differences in their cell of origin [8, 9]. However,\u0007luminal, HER2 and basal-like breast cancers have been\u0007shown to harbour discrete patterns of genetic aberrations,\u0007which may also contribute to their distinct transcriptomic\u0007profiles and clinical phenotypes [1–3, 5, 10].\u0007Gene copy number aberrations have been observed in\u0007[95% of breast cancers and involve a substantial proportion\u0007of the genome of breast cancer cells [1–3, 5, 6, 10–12]. It is\u0007unclear, however, as to what extent copy number aberrations\u0007drive the expression of genes and molecular networks and\u0007pathways in breast cancer. Previous efforts in identifying\u0007these networks and pathways were either carried out in breast\u0007cancer cell lines [13] or made use of non-microdissected\u0007tissue [2–5, 11], which reduces the sensitivity of detection in\u0007copy number aberrations and confounds the transcriptomic\u0007profiles due to contamination with diploid stromal and\u0007inflammatory cells.\u0007The aims of this study were twofold: (1) to determine\u0007the genes whose expression levels correlate with gene copy\u0007numbers and the genes which are overexpressed when\u0007amplified in grade III (GIII) breast cancers; and (2) to\u0007determine canonical pathways and networks significantly\u0007enriched for these genes. We microdissected and analysed\u0007a series of 48 GIII invasive ductal carcinomas of no special\u0007type (IDC-NSTs) with high resolution microarray-based\u0007comparative genomic hybridisation (aCGH) and mRNA\u0007expression arrays. Our study focused on GIII IDC-NSTs\u0007only to avoid bias introduced by distinct patterns of\u0007genetic aberrations between grade I and III cancers [12]\u0007and between invasive ductal carcinomas and special types\u0007[14–16].\u0007Materials and methods\u0007Sample selection\u0007A series of 64 consecutive GIII IDC-NSTs cases were\u0007retrieved from University Hospital La Paz, Madrid, Spain.\u0007This project has been approved by the local Research\u0007Ethics Committees of the authors’ institutions.\u0007Four-lm thick sections of the tumours were subjected to\u0007immunohistochemistry with antibodies against oestrogen\u0007receptor, (ER, ID5, 1:30, Novocastra), progesterone\u0007receptor (PR, 1A6, 1:30, Novocastra), cytokeratin (CK) 5/6\u0007(D5/16 B4, 1:25, DAKO) and epidermal growth factor\u0007receptor (EGFR) (EGFR, 31G7, 1:50, Zymed) essentially\u0007as previously described [1]. Immunohistochemistry for\u0007HER2 was performed using the Herceptest kit (Dako,\u0007Glostrup, Denmark). HER2 expression was scored\u0007according to the ASCO/CAP guidelines [17]; only cases\u0007displaying a 3? score were considered positive, cases dis-\u0007playing a 2? score were labelled as equivocal. In equivocal\u0007cases, HER2 gene status was determined using the FDA\u0007approved SpotLight chromogenic in situ hybridisation\u0007(CISH) HER2 probes. All markers were scored by two\u0007pathologists (SMP-R and JSR-F), blinded to the results of\u0007the aCGH, expression array and CISH analysis. Tumours\u0007were classified into HER2, luminal and basal-like sub-\u0007groups utilising the criteria of Nielsen et al. [18].\u0007Nucleic acid extraction\u0007Ten consecutive 8-lm thick, frozen sections of 64 GIII\u0007tumours were microdissected as previously described [14,\u000719] to obtain a percentage of tumour cells in the remaining\u0007tissue greater than 90%. DNA was extracted as previously\u0007described [12]. DNA concentration was measured with\u0007Picogreen  according to the manufacturer’s instructions\u0007(Invitrogen, Paisley, UK). RNA was extracted using Trizol\u0007according to the manufacturers’ instructions (Invitrogen,\u0007UK) and quantified using the Agilent 2100 Bioanalyzer\u0007with RNA Nano LabChip Kits (Agilent Biosystems). In 16\u0007cases, DNA and/or RNA were of insufficient quantity for\u0007aCGH and/or expression array analysis and were excluded\u0007from the study. The final dataset comprises 48 cases, whose\u0007clinico-pathological characteristics are summarised in\u0007Supplementary Table 1.\u0007Microarray-based comparative genomic\u0007hybridisation\u0007The 32K bacterial artificial chromosome (BAC) re-array\u0007collection (CHORI) tiling path aCGH platform used for\u0007this study was constructed in the Breakthrough Breast\u0007Cancer Research Centre [14]. DNA labelling, array hy-\u0007bridisations and image acquisition were performed as\u0007described in Natrajan et al. [1].\u0007Cases with [10% of clones missing and clones where\u0007data were not available in C10% of cases were excluded.\u0007Log2 ratios were normalised for spatial and intensity-\u0007dependent biases using a two-dimensional LOESS regres-\u0007sion followed by a BAC-dependent bias correction as\u0007576 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007previously described [14, 20]. Polymorphic BACs identi-\u0007fied in an analysis of 50 male:female and female:female\u0007hybridisations were filtered out. This left a final dataset of\u000731,544 clones with unambiguous mapping information\u0007according to the March 2006 build (hg18) of the human\u0007genome (http://www.ensembl.org). Data were smoothed\u0007using the circular binary segmentation (cbs) algorithm [6].\u0007A categorical analysis was applied to the BACs after clas-\u0007sifying them as representing gain, loss or no-change\u0007according to their smoothed Log2 ratio values [21].\u0007Threshold values were chosen to correspond to three stan-\u0007dard deviations of the normal ratios obtained from the fil-\u0007tered clones mapping to chromosomes 1-22, assessed in\u0007multiple hybridisations between DNA extracted from a pool\u0007of male and female blood donors as previously described [21,\u000722] (Log2 ratio of ±0.08). Low level gain was defined as a\u0007smoothed Log2 ratio of between 0.08 and 0.45, corre-\u0007sponding to approximately 3–5 copies of the locus, whilst\u0007gene amplification was defined as having a Log2 ratio[0.45,\u0007corresponding to more than 5 copies [19].\u0007Gene expression profiling\u0007Gene expression profiling was performed using the Illumina\u0007human WG6 version 2 expression array according to the\u0007manufacturer’s protocol. Briefly, 200 ng of each RNA\u0007sample was amplified using the Illumina Totalprep RNA\u0007amplification kit following the manufacturer’s instructions\u0007(Ambion, UK). Labelled cRNA was subsequently hybri-\u0007dised to Illumina Sentrix-human 6 version 2 expression\u0007bead-chips. Full details on RNA amplification and hybrid-\u0007isation can be found at www.illumina.com. Raw gene\u0007expression values were robust-spline normalised using the\u0007Bioconductor lumi package (http://www.bioconductor.org/\u0007packages/2.3/bioc/html/lumi.html) in R. Genes were map-\u0007ped to their genomic location using the lumiHumanAllv2\u0007annotation database available from Bioconductor. Only\u0007Illumina transcript probes with detection P values\\0.01 in\u0007[25% of samples were included; this resulted in a dataset\u0007of 12,699 transcriptionally regulated probes with accurate\u0007and unequivocal mapping information. Gene expression\u0007data are publicly available at ArrayExpress http://www.ebi.\u0007ac.uk/microarray-as/ae/ (accession number: E-TABM-543).\u0007Integrated aCGH and expression analysis\u0007To identify genes whose expression correlated with genetic\u0007copy number changes, cbs-smoothed aCGH data were used\u0007to assign the median aCGH states for each of the 12,699\u0007genes in the gene expression dataset using the median\u0007values for all BACs that overlap with the genomic posi-\u0007tions of each gene. This resulted in a 1:1 matrix of\u0007expression data and aCGH values used in correlations.\u0007Pearson’s correlations were performed between mRNA\u0007expression log intensity values and median cbs-smoothed\u0007ratios derived from aCGH analysis for each gene. P values\u0007for each test were adjusted with Benjamini and Hochberg\u0007multiple P value adjustment [6, 23].\u0007To identify genes consistently overexpressed when\u0007amplified in regions of recurrent amplification, Mann–\u0007Whitney U tests were performed and adjusted with Ben-\u0007jamini and Hochberg multiple P value adjustment. An\u0007adjusted P value of \\0.05 was considered significant.\u0007For unsupervised analysis, we performed hierarchical\u0007clustering of the 48 primary tumours using 848 unique genes\u0007identified from the 1,300 intrinsic genes, as described by Hu\u0007et al. [24]. Genes were median centred, and average linkage\u0007clustering was performed using Pearson’s correlation as the\u0007similarity metric on Cluster 3 [25]. Results were subse-\u0007quently visualised using Java Treeview (http://jtreeview.\u0007sourceforge.net/). SAM software [26] was employed to\u0007determine the genes significantly differentially expressed\u0007between basal-like, HER2 and luminal tumours, adopting a\u0007false discovery rate of 3% after 1,000 permutations.\u0007Ingenuity pathway analysis\u0007To determine pathways and networks that were significantly\u0007regulated in the gene expression data of basal-like, HER2 and\u0007luminal GIII tumours, we performed pathway analysis using\u0007the Ingenuity Pathway Analysis (IPA) program (http://www.\u0007ingenuity.com). HUGO gene identifiers were mapped to\u0007networks available in the Ingenuity database and ranked by\u0007score. The score indicates the likelihood of the genes in a\u0007network being found together due to random chance. Using a\u000799% confidence level, scores of C3 are considered signifi-\u0007cant. IPA was performed essentially as previously described\u0007[16, 27] with the Gene-sets described in Fig. 1.\u0007Exact hypergeometric probability analysis\u0007for gene list enrichment\u0007Analysis for gene list enrichment was performed using\u0007hypergeometric probability analysis, whereby the number\u0007of genes in common between two groups was identified\u0007and a representation factor (the number of overlapping\u0007genes divided by the expected number of overlapping\u0007genes drawn from two independent groups) was calculated.\u0007A representation factor [1 indicates more overlap than\u0007expected of two independent groups, whereas a represen-\u0007tation factor \\1 indicates less overlap than expected. The\u0007probability of finding an overlap of that number of genes\u0007was then calculated using the hypergeometric probability\u0007formula: C (D, x) 9 C (N - D, n - x)/C (N, n). http://\u0007elegans.uky.edu/MA/progs/representation.stats.html.\u0007Breast Cancer Res Treat (2010) 121:575–589 577\u0007123\u0007Validation of aCGH and microarray-based\u0007expression profiling\u0007All cases were reviewed by two pathologists (SMP-R and\u0007JSR-F), and representative areas of the 48 cases were\u0007included in a tissue microarray containing two 1-mm rep-\u0007licate cores from each case. These cases were used to\u0007validate selected amplifications found in aCGH.\u0007Immunohistochemistry for cyclin D1 was performed\u0007with the SP4 rabbit monoclonal antibody (Neomarkers,\u0007Suffolk, UK) at 1:50 dilution as previously described [28]\u0007and scored using the Allred scoring system [28]. The\u0007immunohistochemical analysis was performed by two of\u0007the authors (SMP-R and JSR-F) with observers blinded to\u0007the results of the aCGH, gene expression and CISH\u0007analyses.\u0007Chromogenic in situ hybridisation was performed using\u0007the Zymed/Invitrogen SpotLight CISH probes against\u0007HER2, CCND1 and EGFR [21, 28, 29], and an in-house\u0007generated PPM1D probe [30]. Hybridisations and washes\u0007were performed as previously described [21, 28–30]. Gene\u0007amplification was defined as the presence of [5 signals,\u0007large gene clusters or a combination of signals and clusters\u0007in [50% of morphologically unequivocal neoplastic cells\u0007as previously described [21, 28–30]. CISH analysis was\u0007performed by two of the authors (SMP-R and FCG) with\u0007observers blinded to the results of the aCGH, gene\u0007expression and immunohistochemical analyses.\u0007Fig. 1 Work flow diagram of\u0007integration of aCGH and gene\u0007expression data. a Analysis of\u0007cohort of 48 grade III invasive\u0007ductal carcinomas of no special\u0007type. b Analysis of 48 grade III\u0007invasive ductal carcinomas of\u0007no special type assigned to the\u0007molecular subtypes basal-like\u0007(n = 13), HER2 (n = 13) and\u0007luminal (n = 22) separately.\u00071Nielsen et al. [18], MWU\u0007Mann–Whitney U test, adjp\u0007adjusted P value\u0007578 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007Quantitative real-time reverse transcriptase PCR\u0007RNA from 48 microdissected cases was available for quan-\u0007titative real-time reverse transcriptase PCR (qRT-PCR)\u0007analysis. First strand synthesis was performed as previously\u0007described [1], and qRT-PCR was performed with TaqMan\u0007chemistry on the ABI Prism 7900HT (Applied Biosystems),\u0007using the standard curve method [1, 21, 30]. Assays for\u0007PPM1D and TFRC were purchased from Applied Biosys-\u0007tems. PPM1D expression levels were normalised to gene\u0007TFRC (Assay on demand ID: Hs00174609_m1-TFRC and\u0007Hs00186230_m1-PPM1D) [30].\u0007Results\u0007Breast cancer molecular subtype classification\u0007To determine the basal-like, luminal and HER2 molecular\u0007subtype classes, we subjected 48 GIII IDC-NSTs to a val-\u0007idated immunohistochemical surrogate of the molecular\u0007subtypes of breast cancer described by Nielsen et al. [18].\u0007This taxonomy was employed given that it can be used to\u0007classify archival tumour samples and more accurately\u0007identifies the clinically relevant subgroup of HER2-posi-\u0007tive/amplified tumours (i.e. amenable to therapy with anti-\u0007HER2 agents). Using Hu et al.’s [24] or Parker et al.’s [31]\u0007microarray-based classification, a proportion of ER-positive\u0007HER2 amplified cancers are classified as of luminal B\u0007phenotype. Based on the expression of HER2, ER, CK 5/6\u0007and EGFR, 13, 22 and 13 tumours were classified as HER2,\u0007luminal and basal-like phenotype, respectively (Supple-\u0007mentary Table 1). As expected, a good agreement between\u0007the classification of tumours into basal-like, HER2 and\u0007luminal using Nielsen et al.’s [18] immunohistochemical\u0007surrogate and by expression profiling using the Hu et al.’s\u0007centroids [24] was observed (Kappa score = 0.753, P \\ 1\u0007910-11, Supplementary Table 1). As expected [31], cases\u0007harbouring HER2 amplification were consistently classified\u0007as of HER2 phenotype by Nielsen et al. [18] and fell into\u0007either HER2 or luminal subgroup using the Hu et al.’s\u0007centroids [24].\u0007Identification of genes whose expression is associated\u0007with gene copy number\u0007To define genes whose expression is significantly corre-\u0007lated with gene copy number, we extracted DNA and RNA\u0007from microdissected samples and performed aCGH anal-\u0007ysis, using a BAC array platform (actual resolution 50 kb)\u0007and genome-wide gene expression profiling using the\u0007Illumina WG6 v2 platform. We next overlaid the aCGH\u0007and gene expression data of our 48 GIII IDC-NSTs. Out of\u0007the 12,699 significantly regulated genes, 5,931 (46.70%)\u0007displayed mRNA expression levels that significantly cor-\u0007related with copy number (Pearson’s correlation adjusted\u0007P \\ 0.05, Supplementary Table 2, Gene-set A, Fig. 1a).\u0007This list included genes that have previously been found to\u0007have increased expression levels determined by gains in\u0007copy number, such as ORAOV1, ASH2L, FADD, RPS6\u0007KB1, NME1, EIF4EBP2, GRB7, INTS2, PERLD1, MYST3,\u0007PIGS, INTS4, TLK2, PPM1D, PROSC, STARD3, ERBB2\u0007and CTTN [3–6, 11].\u0007To identify potential amplicon drivers, we determined\u0007genes whose expression correlates with copy number and are\u0007recurrently amplified (n = 811, Pearson’s correlation\u0007adjusted P \\ 0.05; amplified in 2 or more tumours, Sup-\u0007plementary Table 3; Gene-set B) and genes significantly\u0007overexpressed when amplified (n = 469, Mann–Whitney U\u0007test adjusted P \\ 0.05, Supplementary Table 3; Gene-set\u0007C). Not surprisingly, these lists of genes included known\u0007drivers of amplicons on 1q, 8p, 8q, 11q13, 17q12-q21,\u000717q22-q25 and 20q13 [1, 3–5, 7, 11, 32–36]. This unbiased\u0007approach accurately identified known amplicon drivers,\u0007including HER2, STARD3, GRB7 [37] on 17q12, ORAOV1\u0007[38] on 11q13 and PPM1D [1, 5, 30] on 17q23.2, genes\u0007whose expression and activity have been shown to be\u0007required for the proliferation and/or survival of cancer cells\u0007harbouring their amplification. In addition, this analysis has\u0007also revealed additional genes on chromosome 17q that do\u0007not pertain to the smallest region of amplification on 17q12-\u0007q21 or on 17q22-q25, but are significantly overexpressed\u0007when amplified (e.g. MRPL45, FBXL20, MED1 and ORM\u0007DL3, Fig. 2a), suggesting that the reported complex patterns\u0007of genetic aberrations on chromosome 17 [39] may have a\u0007biological impact (i.e. overexpression of specific genes) and\u0007are not mere genomic ‘noise’. Similar findings were\u0007observed in the genomic regions adjacent to the amplicons\u0007on 11q13.2-q13.3 and 17q21-q23 (Fig. 2b,c).\u0007On the other hand, the 8q23.2-24.3 amplicon displayed a\u0007significantly more complex pattern of genetic aberrations\u0007without a well-defined core. Our unbiased analysis dem-\u0007onstrates that UTP23, RAD21, MED30, MRPL13, MTBP,\u0007C8orf76, ZHX1, C8orf32, FBXO32, NDUFB9, KIAA0196,\u0007NSMCE2, DDEF1, EFR3A, PHF20L1, ST3GAL1 and\u0007TRAPPC9 (Fig. 2d) are overexpressed when amplified and\u0007constitute potential drivers of this amplicon. It should be\u0007noted that MYC, which maps to this region, has expression\u0007levels that correlate with copy number, but is not consis-\u0007tently overexpressed when amplified, as previously\u0007described [40]. Taken together with the results of previous\u0007functional and correlative studies [3, 4, 7, 11, 13, 37, 38],\u0007our analysis seems to identify accurately genes whose\u0007expression correlates with copy number and potential\u0007amplicon drivers.\u0007Breast Cancer Res Treat (2010) 121:575–589 579\u0007123\u0007Molecular pathways regulated by gene copy number\u0007differ in basal-like, HER2 and luminal cancers\u0007To determine which of the 5,931 genes whose expression\u0007levels are significantly correlated with gene copy number are\u0007associated with each molecular subtype of GIII breast can-\u0007cers (i.e. basal-like, HER2 and luminal), multiclass signifi-\u0007cance analysis of microarray (SAM) was performed [26]\u0007using 1,000 permutations and accepting a false discovery\u0007rate of 3%. SAM analysis revealed that 1,897 genes (Gene-\u0007set D; Fig. 1a) were differentially expressed between GIII\u0007IDC-NSTs of basal-like, HER2 and luminal molecular sub-\u0007types (Supplementary Table 4). As expected, genes per-\u0007taining to the HER2 amplicon, PERLD1, C17orf37, GRB7,\u0007STARD3 and HER2 (ERBB2) were significantly overex-\u0007pressed in cancers of HER2 phenotype and were the top\u0007ranking genes in the SAM analysis.\u0007Ingenuity Pathway Analysis (IPA) of these 1,897 tran-\u0007scripts revealed networks and pathways enriched for genes\u0007whose expression correlates with copy number and are\u0007differentially expressed between molecular subtypes.\u0007These included networks related to DNA replication and\u0007repair, and cell cycle (Fig. 3; Supplementary Fig. 1; Sup-\u0007plementary Table 5). For instance, the network ‘Gene\u0007Expression, DNA Replication, Recombination, and Repair,\u0007Cell Cycle, Cancer’ (score 40, Fig. 3) encompasses MYC\u0007and other key genes such as MDC1, a mediator of DNA\u0007damage checkpoint; MSH3, MLH1 and MLH3, which play\u0007a role in mismatch repair; and RAD50, which is involved in\u0007DNA double-strand break repair. Basal-like cancers dis-\u0007played recurrent copy number gains/amplifications and\u0007upregulation of MYC and MDC1 and losses and down-\u0007regulation of RAD50 and genes related to mismatch repair\u0007(i.e. MSH3, MLH1 and MLH3) compared to HER2 and\u0007luminal cancers (Fig. 3). The network ‘Gene Expression,\u0007DNA Replication, Recombination, and Repair, Cell Cycle’\u0007(score 40, Supplementary Fig. 1) was centred around key\u0007genes involved in cell cycle such as CCND1, E2F3,\u0007CCNA2 and CCNDBP1, with basal-like tumours showing\u0007recurrent deletions and downregulation of CCND1 and\u0007Fig. 2 Matched heatmaps of gene expression and aCGH within\u0007regions of recurrent amplification. a 17q12-q21.1, b 11q13.2-q13.3, c\u000717q21-q23 and d 8q23.3-q24.3. For each amplicon, genes within the\u0007amplified region are recovered and median aCGH values and states\u0007are assigned. Samples are separated into those harbouring amplifica-\u0007tion within the region and those that do not. Expression and aCGH\u0007values are depicted in two matching heatmaps (aCGH states on the\u0007left and expression values on the right) in which the genes are ordered\u0007according to their chromosomal position. Bar plots show the result of\u0007a Mann–Whitney U test for expression as a continuous variable and\u0007gene amplification as the grouping variable. Bars in red show\u0007adjusted P values of less than 0.05. aCGH: green copy number loss;\u0007black no copy number change; dark red copy number gain; bright red\u0007gene amplification. Gene expression: green downregulation, red\u0007upregulation; MWU Mann–Whitney U test, adjp adjusted P value\u0007580 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007copy number gains and upregulation of E2F3 and HER2,\u0007whereas luminal cancers harboured recurrent amplifica-\u0007tions and overexpression of CCND1 and less frequent gains\u0007of E2F3 (Supplementary Fig. 1).\u0007In agreement with the above, IPA also identified the\u0007canonical pathways ‘Cell Cycle: G1/S Checkpoint Regu-\u0007lation’ and ‘Role of BRCA1 in DNA Damage Response’\u0007as significantly enriched for genes whose expression cor-\u0007relates with copy number and are significantly differentially\u0007expressed in breast cancer molecular subtypes (P \\ 0.001\u0007and P = 0.0069, respectively) (Supplementary Table 6).\u0007This analysis revealed that the expression levels of distinct\u0007components of the G1/S checkpoint regulation pathway are\u0007differentially expressed in basal-like, luminal and HER2\u0007cancers due to specific patterns of copy number aberrations\u0007(Fig. 4). Recurrent gains/amplifications and upregulation of\u0007CDK6, p16INK4A, E2F, CCNE1 and MYC were observed\u0007in basal-like breast cancers, whereas CCNE1 and CCND1\u0007were recurrently gained/amplified and upregulated in HER2\u0007tumours, and CCND1 in luminal tumours. SKP1 and SKP2\u0007were recurrently deleted and downregulated in basal-like\u0007breast cancers, whereas recurrent gains and overexpression\u0007of SKP2 were found in HER2 cancers, and recurrent gains\u0007and overexpression of both genes were observed in luminal\u0007cancers. Interestingly, p16 (CDKN2A), a gene that has been\u0007reported to be overexpressed in the majority of basal-like\u0007breast cancers [41], was recurrently gained and overex-\u0007pressed in these tumours; whereas HER2 and luminal can-\u0007cers harboured recurrent deletions and downregulation of\u0007this gene. Genes involved in BRCA1 DNA damage control\u0007Fig. 3 Network enriched for genes whose expression correlates with\u0007copy number and are significantly differentially expressed between\u0007basal-like, HER2 and luminal GIII invasive ductal carcinomas of no\u0007special type, defined by Ingenuity Pathway Analysis software. a\u0007‘Gene Expression, DNA Replication, Recombination, and Repair,\u0007Cell Cycle, Cancer’ network (score 40) in (I) basal-like, (II) HER2\u0007and (III) luminal tumours. Green downregulation, red upregulation. b\u0007Matched heatmaps of aCGH categorical data for each of the genes\u0007differentially expressed in (I) basal-like, (II) HER2 and (III) luminal\u0007tumours. Green copy number loss; black no copy number change;\u0007dark red copy number gain; bright red gene amplification\u0007Breast Cancer Res Treat (2010) 121:575–589 581\u0007123\u0007and DNA repair mechanisms, including RB, MLH1 and\u0007RAD50, were recurrently deleted and downregulated and\u0007E2F3, MRE11A and FANCE were recurrently copy number\u0007gained and upregulated in basal-like cancers compared\u0007with HER2 and luminal tumours (Supplementary Fig. 2).\u0007Although BRCA1 did not reach significance in our multi-\u0007class SAM analysis, its expression is associated with copy\u0007number (Pearson’s correlation adjusted P = 0.0433), and\u0007lower expression is associated with basal-like tumours\u0007(P = 0.0283, t-test, homoscedastic; adjusted P = 0.01499,\u0007Mann–Whitney U test). In addition, ‘Protein Ubiquitina-\u0007tion’ (P \\ 0.001) and ‘IGF-1’ (P = 0.0018) were amongst\u0007the top canonical pathways significantly enriched for genes\u0007whose expression correlates with copy number (Supple-\u0007mentary Table 6).\u0007These results suggest that the transcriptional regulation of\u0007molecular pathways may be determined by the pattern of\u0007genetic aberrations in breast cancers and that distinct\u0007pathways and networks are driven by specific patterns of copy\u0007number aberrations in basal-like, HER2 and luminal tumours.\u0007To corroborate the above hypotheses, we overlaid the aCGH\u0007and expression data for each subtype separately (Pearson’s\u0007correlation adjusted P \\ 0.05, Supplementary Table 7). This\u0007resulted in a list of 730 (Gene-set E1), 933 (Gene-set F1) and\u00072,137 (Gene-set G1; see Fig. 1b) genes whose expression is\u0007associated with copy number in 13 basal-like, 13 HER2 and 22\u0007luminal cancers, respectively. A significant overlap as defined\u0007by exact hypergeometric probability analysis was found for\u0007the genes identified as copy number regulated with both\u0007approaches in basal-like (Gene-set D vs. Gene-set E1, Rep-\u0007resentation factor: 2.0, P \\ 10-25), HER2 (Gene-set D vs.\u0007Gene-set F1, Representation factor: 2.1, P \\ 10-37) and\u0007luminal (Gene-set D vs. Gene-set G1, Representation factor:\u00071.9, P \\ 10-71) cancers.\u0007Ingenuity Pathway Analysis of Gene-sets E1, F1 and G1\u0007(Pearson’s correlation adjusted P \\ 0.05) revealed\u0007Fig. 4 Canonical pathway ‘Cell Cycle: G1/S Checkpoint Regulation’\u0007significantly enriched for genes whose expression correlates with\u0007copy number and differentially expressed between basal-like, HER2\u0007and luminal GIII tumours (P \\ 0.001), defined by Ingenuity Pathway\u0007Analysis software. a Canonical pathway in (I) basal-like, (II) HER2\u0007and (III) luminal tumours; green downregulation, red upregulation. b\u0007Matched heatmaps of aCGH data of (I) basal-like, (II) HER2 and (III)\u0007luminal tumours. Green copy number loss; black no copy number\u0007change; dark red copy number gain; bright red gene amplification\u0007582 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007networks and pathways independently associated with each\u0007molecular subtype (Supplementary Tables 6, 8). This\u0007analysis revealed a heterogeneous pattern of losses, gains\u0007and amplifications of genes whose expression levels are\u0007correlated with copy number between the tumour subtypes\u0007leading to the activation of distinct networks and pathways.\u0007In basal-like cancers (Gene-set E1), the ‘Oestrogen\u0007Receptor Signalling’ (P = 0.00178, Fig. 5) and ‘Protein\u0007Ubiquitination’ (P = 0.00181) canonical pathways were\u0007significantly enriched for genes whose expression corre-\u0007lates with copy number (Supplementary Table 6). Of these\u0007genes pertaining to the ‘Oestrogen Receptor Signalling’\u0007pathway, the vast majority were lost and downregulated,\u0007with the exception of PgR, which was gained in 3 and\u0007amplified in one basal-like case (B11, which expressed\u0007high mRNA and protein levels of progesterone receptor\u0007(PR), Supplementary Table 1). A strong correlation was\u0007observed between ER and PR status as defined by immu-\u0007nohistochemistry and ESR1 and PgR mRNA expression\u0007levels as defined by microarray-based expression analysis\u0007(Mann–Whitney U test, P = 5.5 9 10-9 and P = 1.2 9 10-7,\u0007respectively; Supplementary Table 1). The ‘Oestrogen\u0007Receptor Signalling’ canonical pathway was also signifi-\u0007cantly enriched for genes whose expression correlates with\u0007copy number in HER2 cancers (Gene-set F1, P = 0.01288,\u0007Fig. 5); however, the genes that were copy number regu-\u0007lated in each subtype were largely distinct. In HER2\u0007tumours, RAF1 and ERK1/2 were recurrently lost, whereas\u0007gains of multiple components of the RNA polymerase II\u0007complex, including POLR2K and TATA-binding protein-\u0007associated factor genes (i.e. TAF4 and TAF15), were\u0007observed. In HER2 cancers, the ‘Nucleotide Excision Repair\u0007Pathway’ was significantly enriched for genes whose\u0007expression correlates with copy number (P = 0.0214), with\u0007recurrent gains and amplifications of POLR2K and losses of\u0007RPA2 (Fig. 5; Supplementary Table 6).\u0007These results suggest that several of the molecular\u0007pathways and networks specifically associated with basal-\u0007like, HER2 and luminal cancers are driven, at least in part,\u0007by specific genetic aberrations. Importantly, DNA copy\u0007number changes appear to be involved in the downregu-\u0007lation of oestrogen receptor signalling, one of the hallmark\u0007features of basal-like cancers [42, 43], and in the dys-\u0007functional status of gene pathways and networks related to\u0007various types of DNA repair mechanisms.\u0007Pathway analysis identifies novel potential therapeutic\u0007targets in GIII IDC-NSTs of basal-like, HER2\u0007and luminal molecular subtype\u0007Previous studies have demonstrated that genes whose\u0007expression correlates with copy number and are recurrently\u0007amplified and genes consistently overexpressed when\u0007amplified are likely to be amplicon drivers and may con-\u0007stitute therapeutic targets [1, 3, 5, 30, 36, 44]. To identify\u0007potential therapeutic targets specific for subgroups of\u0007basal-like, HER2 and luminal cancers, we defined (1)\u0007genes whose expression correlates with copy number in\u0007each molecular subtype individually (Pearson’s correlation\u0007adjusted P \\ 0.05) and are recurrently amplified in each\u0007group and (2) genes overexpressed when amplified in each\u0007group individually (Mann–Whitney U adjusted P \\ 0.05)\u0007(Fig. 1b). The first analysis resulted in a list of 32, 157 and\u0007265 genes in basal-like, HER2 and luminal tumours,\u0007respectively (Gene-sets E2, F2 and G2, respectively),\u0007whereas the second resulted in 16, 33 and 159 genes in\u0007basal-like, HER2 and luminal tumours, respectively (Gene-\u0007sets E3, F3 and G3, respectively), (Supplementary Tables 9,\u000710). A significant overlap as defined by exact hypergeo-\u0007metric probability analysis was found between the genes\u0007identified as recurrently amplified and genes overexpressed\u0007when amplified in basal-like (Gene-set E2 vs. Gene-set E3,\u0007Representation factor: 372.0, P \\ 10-40), HER2 (Gene-set\u0007F2 vs. Gene-set F3, Representation factor: 68.8, P \\ 10-50)\u0007and luminal (Gene-set G2 vs. Gene-set G3, Representation\u0007factor: 44.3, P \\ 10-252) tumours.\u0007Genes whose expression correlates with copy number and\u0007are recurrently amplified in basal-like cancers (Gene-set E2)\u0007mapped to 1q22-q24, 1q44, 8q24.1, 10p15 and 19q12-q13.\u0007IPA revealed that the networks ‘Gene Expression, Cancer,\u0007Gastrointestinal Disease’ (score 46) and ‘Gene Expression,\u0007Cancer, Cell Death’ (score 17) were significantly enriched\u0007for these genes (Supplementary Table 11; Supplementary\u0007Fig. 3). Interestingly, different components of these net-\u0007works were amplified in individual cases. Given the limited\u0007sample size, only genes mapping to 19q12-q13 were sig-\u0007nificantly overexpressed when amplified in basal-like can-\u0007cers after multiple comparison adjustment (Gene-set E3;\u0007Fig. 1b). These genes, including C19orf2, MED30, SUPT\u00075H, SERTAD1 and SERTAD3, were part of the network\u0007‘Cancer, Cell Death, Endocrine System Disorders’ (score\u000740, Supplementary Fig. 3). Notably, the p21-activated\u0007kinase 4 (PAK4) was present in both ‘Gene expression,\u0007Cancer, Cell Death’ and ‘Cancer, Cell Death, Endocrine\u0007System Disorders’ networks and has recently emerged as a\u0007potential therapeutic target [45].\u0007In HER2 cancers, the network ‘Lipid Metabolism,\u0007Molecular Transport, Nucleic Acid Metabolism’ (score 37)\u0007was significantly enriched for recurrently amplified genes\u0007whose expression correlates with copy number (Gene-set\u0007F2) and the network ‘Amino Acid Metabolism, Post-\u0007Translational Modification, Small Molecule Biochemistry’\u0007(score 29, Fig. 6; Supplementary Tables 11, 12) was sig-\u0007nificantly enriched for genes overexpressed when amplified\u0007(Gene-set F3; Fig. 1b). These networks were composed of\u0007genes mapping not only to the HER2 amplicon (17q12-\u0007Breast Cancer Res Treat (2010) 121:575–589 583\u0007123\u0007q21), but also to distinct amplicons, including 8p12, 8q22-\u0007q24, 11q13-q14, 17p11 and 17q23, which were amplified\u0007in multiple combinations in each tumour (Fig. 6; Supple-\u0007mentary Tables 9, 10). Interestingly, PPM1D, a negative\u0007regulator of the p53 pathway and potential therapeutic\u0007target for a subgroup of patients with HER2 amplified\u0007breast cancers [1], was present in both networks.\u0007In luminal tumours, the ‘ERK/MAPK Signalling’ (P =\u00070.00631 and P = 0.00978, respectively) and ‘PI3K/AKT\u0007Signalling’ (P = 0.01445 and P = 0.01047, respectively)\u0007canonical pathways were significantly enriched for recur-\u0007rently amplified genes whose expression correlates with\u0007copy number (Gene-set G2) and for genes overexpressed\u0007when amplified (Gene-set G3) (e.g., PAK1, PRKAR1A,\u0007RPS6KB1, PPP2R5A) (Supplementary Table 6; Fig. 7;\u0007Supplementary Fig. 4). Protein kinases of these two canon-\u0007ical pathways may constitute targets for therapeutic inter-\u0007vention by PI3K and/or MEK inhibitors [13, 46, 47].\u0007Luminal breast cancers were also significantly enriched for\u0007genes in the networks ‘Cell Cycle, Reproductive System\u0007Development and Function, Haematological System\u0007Development and Function’ (score 40) and ‘Reproductive\u0007System Disease, Cellular Assembly and Organisation,\u0007Cellular Movement’ (score 33) (Supplementary Tables 11,\u000712; Fig. 7; Supplementary Fig. 4). Of note, PPM1D was\u0007part of the cell cycle network, providing additional evidence\u0007to our previous findings that PPM1D may also be a potential\u0007therapeutic target in a subgroup luminal breast cancers\u0007harbouring amplification of this gene [1].\u0007Validation of aCGH and microarray-based\u0007expression profiling analysis\u0007Two tissue microarrays containing the 48 tumours in our\u0007study were subjected to CISH with probes for CCND1,\u0007EGFR, PPM1D and HER2. A perfect agreement between\u0007aCGH and CISH results for EGFR, PPM1D and HER2 was\u0007observed (Supplementary Table 1; Supplementary Fig. 5),\u0007whereas for CCND1, all but two cases showed concordant\u0007results (unweighted Kappa score = 0.86297; 95% CI\u00070.7099–1.0000).\u0007A good correlation was observed between the micro-\u0007array and qRT-PCR–defined expression levels of PPM1D\u0007(Pearson’s r value = 0.81421, P = 1.8 9 10-11). In\u0007agreement with the results of the aCGH and expression\u0007array analysis, CISH and qRT-PCR-based assessment of\u0007PPM1D copy numbers and mRNA expression also\u0007Fig. 5 Canonical pathways enriched for genes whose expression\u0007correlates with copy number in 13 basal-like and 13 HER2 GIII\u0007tumours, defined by Ingenuity Pathway Analysis software. a Canon-\u0007ical pathways (I) ‘Oestrogen Receptor Signalling’ (P = 0.0018) in\u0007basal-like, (II) ‘Oestrogen Receptor Signalling’ (P = 0.013) in HER2\u0007and (III) ‘Nucleotide Excision Repair Pathway’ (P = 0.0214) in\u0007HER2 GIII tumours. Note that the expression levels of genes\u0007significantly correlate with gene copy number. b Matched heatmaps\u0007of aCGH data of (I) basal-like, (II) HER2 and (III) HER2 tumours.\u0007Green copy number loss; black no copy number change; dark red\u0007copy number gain; bright red gene amplification\u0007584 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007demonstrated that PPM1D is overexpressed when ampli-\u0007fied (Mann–Whitney U test = 5.0, P = 1.4 9 10-5).\u0007Immunohistochemical analysis of cyclin D1 using a\u0007semiquantitative scoring system showed a good correlation\u0007with CCND1 mRNA levels as defined by microarray\u0007analysis (Spearman’s correlation coefficient = 0.68943,\u0007P \\ 1 9 10-6). Despite the impact of ER expression on\u0007cyclin D1 expression levels [28], CISH and immunohis-\u0007tochemistry based assessment of CCND1 copy numbers\u0007and cyclin D1 protein expression, respectively, confirmed\u0007that CCND1 is overexpressed when amplified (Mann–\u0007Whitney U test = 62.0, P = 0.00694), in agreement with\u0007aCGH and expression array analysis. Our results provide\u0007direct evidence for the accuracy of the aCGH and expres-\u0007sion profiling methods employed in this study.\u0007Discussion\u0007The actual impact of gene copy number changes on the\u0007transcriptomic profiles of breast cancer molecular subtypes\u0007has been a matter of contention. Whilst HER2 tumours by\u0007definition are biologically driven by amplification and\u0007overexpression of ERBB2, the impact of genomic altera-\u0007tions on the transcriptome and phenotype of basal-like and\u0007luminal tumours is less clear [3]. Here, we show that\u0007Fig. 6 Top ranked networks in 13 HER2 tumours enriched for genes\u0007whose expression correlates with copy number and are recurrently\u0007amplified (Pearson’s correlation) and genes that are overexpressed\u0007when amplified (Mann–Whitney U test), defined by Ingenuity\u0007Pathway Analysis. a Top ranked networks in HER2 tumours (I)\u0007‘Lipid Metabolism, Molecular Transport, Nucleic Acid Metabolism’\u0007(Pearson’s correlation) (score 37), (II) ‘Amino Acid Metabolism,\u0007Post-Translational Modification, Small Molecule Biochemistry’\u0007(Mann–Whitney U test) (score 29); Green downregulation, Red\u0007upregulation. b Matched heatmap of aCGH data of HER2 tumours.\u0007Green copy number loss; black no copy number change; dark red\u0007copy number gain; red gene amplification\u0007Breast Cancer Res Treat (2010) 121:575–589 585\u0007123\u0007distinct gene networks and canonical pathways are enri-\u0007ched for genes whose expression levels correlate with copy\u0007number in basal-like, HER2 and luminal tumours, provid-\u0007ing evidence that differences between these molecular\u0007subtypes may be determined, at least in part, by their pat-\u0007terns of genetic aberrations.\u0007Meta-analyses of microarray expression data have\u0007demonstrated that the main separation between molecular\u0007subtypes is driven by ER, HER2 and proliferation and that\u0007proliferation is a major determinant of breast cancer clin-\u0007ical behaviour [48, 49]. Our analysis demonstrates that\u0007networks and canonical pathways directly related to ER\u0007signalling (Fig. 5), HER2 signalling (Fig. 6) and prolifer-\u0007ation (Fig. 4) are enriched for genes whose expression\u0007correlates with copy number. It should be noted that\u0007although ER pathway was enriched for genes whose\u0007expression correlates with copy number, ESR1 mRNA\u0007levels and DNA copy number failed to show any\u0007statistically significant correlation (Pearson’s correlation\u0007adjusted P = 0.5667) in our study, and ESR1 was only\u0007amplified in a single ER-positive case (1.8%). This concurs\u0007with previous reports suggesting that only a small minority\u0007of ER-positive cancers harbour ESR1 gene amplification\u0007and that ER expression is not significantly associated with\u0007ESR1 copy numbers [50–53].\u0007Our group and others have demonstrated that tumours\u0007arising in patients harbouring BRCA1 germline mutations\u0007predominantly display a basal-like phenotype and that a\u0007subgroup of sporadic basal-like breast cancers harbours a\u0007dysfunctional BRCA1 pathway [54–57]. Given the rarity of\u0007BRCA1 somatic mutations, several mechanisms of BRCA1\u0007pathway dysfunction in sporadic breast cancers have been\u0007proposed [43, 56, 57], including BRCA1 gene promoter\u0007methylation and overexpression of ID4, a negative regu-\u0007lator of BRCA1 [57]. In this study, we provide evidence\u0007that the BRCA1 pathway may be dysfunctional in sporadic\u0007Fig. 7 Top ranking networks and canonical pathways identified by\u0007Ingenuity Pathway Analysis in 22 luminal tumours enriched for genes\u0007whose expression correlates with copy number and are recurrently\u0007amplified (Pearson’s correlation). a Top ranked network and canonical\u0007pathways in luminal tumours (I) ‘Cell Cycle, Reproductive System\u0007Development and Function, Haematological System Development and\u0007Function’ (score = 40), (II) Canonical pathway ‘ERK/MAPK Signal-\u0007ling’ (P = 0.0063), (III) Canonical pathway ‘PI3K/AKT Signalling’\u0007(P = 0.0145); Green downregulation, Red upregulation. b Matched\u0007heatmap of aCGH data of HER2 tumours. Green copy number loss;\u0007black no copy number change; dark red copy number gain; red gene\u0007amplification\u0007586 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007basal-like cancers due to gene copy number aberrations\u0007affecting genes pertaining to this pathway. The ‘BRCA1\u0007DNA Damage Response’ pathway was enriched for genes\u0007whose expression correlates with copy number and are\u0007differentially expressed between the molecular subtypes\u0007(Supplementary Fig. 2). BRCA1 expression itself was\u0007found to be associated with copy number and downregu-\u0007lated in basal-like tumours, suggesting a role for homolo-\u0007gous recombination DNA repair defects in this subgroup of\u0007sporadic breast cancer. In addition, we provide evidence to\u0007suggest that overexpression of the BRCA1 negative regu-\u0007lator ID4 in basal-like cancers is driven by copy number\u0007gains, as (1) the expression levels of this gene correlated\u0007with copy number in the whole cohort (Pearson’s correla-\u0007tion adjusted P \\ 0.01669, Supplementary Table 2), (2)\u0007ID4 was expressed at significantly higher levels in basal-\u0007like breast cancer (Supplementary Table 4), and (3) ID4\u0007copy number gains were significantly more frequent in\u0007basal-like than in non–basal-like tumours (9/13 basal-like\u0007cancers and 10/35 non–basal-like cancers; Fisher’s exact\u0007test, P = 0.0187). Taken together, our results support the\u0007contention that BRCA1 pathway dysfunction in sporadic\u0007basal-like breast cancers may be driven by genetic aber-\u0007rations. Given the reported sensitivity of tumours with\u0007BRCA1 loss of function to PARP inhibitors, our findings\u0007provide further credence to the molecular basis for its\u0007potential efficacy in sporadic basal-like breast cancers [58].\u0007By comparing our list of copy number regulated genes\u0007with that of published ‘intrinsic gene’ lists, we observed\u0007that 61/207 (29%) intrinsic genes described by Perou et al.\u0007[8] were present in our dataset, 166/365 (45.5%) intrinsic\u0007genes by Sorlie et al. [59], 435/929 (46.8%) unique iden-\u0007tifiers of the ‘Intrinsic/UNC’ gene list by Hu et al. [24] and\u0007684/1,488 (46%) intrinsic genes by Parker et al. [31] were\u0007significantly associated with gene copy number. The sub-\u0007stantial enrichment of the published ‘intrinsic gene’ lists\u0007with genes whose expression correlates with copy number\u0007further emphasises the potential importance of genetic\u0007alterations in the molecular classification of breast cancers.\u0007Furthermore, it suggests that the differential expression of\u0007the ‘intrinsic genes’ in the distinct molecular subtypes may\u0007not only stem from their cell of origin [8, 9], but also as a\u0007result of genetic instability and subsequent acquisition of\u0007stochastic genetic aberrations.\u0007To identify possible therapeutic targets, we used our\u0007integrative approach to distinguish recurrently amplified\u0007genes whose expression is associated with copy number\u0007and also genes that are consistently overexpressed when\u0007amplified. Our genome-wide, unbiased approach identi-\u0007fied genes that have been functionally validated as\u0007potential therapeutic targets in specific breast cancer\u0007molecular subtypes (e.g. HER2, PPM1D, PAK1 and\u0007PPAPDC1B—Supplementary Tables 9, 10) [1, 13, 44]\u0007and canonical pathways that have been exploited in\u0007preclinical therapeutic models for breast cancer (e.g.\u0007PI3K/AKT signalling in luminal cancers—Fig. 7 and\u0007Supplementary Fig. 4) [47], as well as a comprehensive\u0007list of genes that may constitute amplicon drivers and\u0007novel therapeutic targets.\u0007Molecular subtype-specific analyses of genes consis-\u0007tently significantly overexpressed when amplified high-\u0007lighted networks and canonical pathways enriched for\u0007potential therapeutic targets. The networks identified in\u0007basal-like tumours were found to be enriched for multiple\u0007genes involved in transcriptional regulation, suggesting\u0007that preclinical studies testing these transcriptional regu-\u0007lators as potential targets may be warranted. The p21-\u0007activated kinase 4 PAK4, which plays a role in integrin-\u0007mediated breast cancer cell motility [60], was significantly\u0007overexpressed when amplified in basal-like cancers. Novel\u0007strategies for targeting PAK4 and its family members in\u0007cancer are under investigation [45]. Ingenuity Pathway\u0007Analysis of HER2 cancers substantiates previous findings\u0007for the use of PPM1D-targeted therapies in this subgroup\u0007[1, 35] and revealed that PI3K and ERK/MAPK canonical\u0007pathways are significantly enriched for genes that are\u0007amplified and overexpressed in luminal tumours (e.g.\u0007PAK1, PPP2R5A and PRKAR1A). Interestingly, PAK1\u0007overexpressing luminal breast cancer cell lines have\u0007recently been shown to be significantly more sensitive to\u0007MEK inhibition [13]. Moreover, the PI3K catalytic subunit\u0007inhibitor BEZ235 has been reported to cause selective cell\u0007growth inhibition in ER-positive breast cancer cell lines\u0007[47] and may constitute an interesting therapeutic approach\u0007for subgroups of patients with luminal breast cancer har-\u0007bouring amplification of these genes.\u0007In conclusion, high resolution integrative genomic\u0007analysis of basal-like, HER2 and luminal GIII breast can-\u0007cers using microdissected tissues identified pathways and\u0007networks specific to each tumour type that are regulated by\u0007changes at the DNA level, and has identified key genes that\u0007govern distinct pathways. These results provide strong\u0007circumstantial evidence to suggest that the diversity of\u0007breast cancer and the molecular subtypes may stem, at least\u0007in part, from the distinct pattern of genetic aberrations\u0007found in these cancers. Furthermore, we have identified\u0007canonical pathways enriched for genes recurrently ampli-\u0007fied or overexpressed when amplified in basal-like, HER2\u0007and luminal cancers. Functional genomic studies to test the\u0007biological importance of these pathways and the thera-\u0007peutic implication of inhibition of these pathways in\u0007appropriate preclinical models are warranted.\u0007Acknowledgments This study was funded in part by Breakthrough\u0007Breast Cancer. B. Weigelt is supported by a Cancer Research UK\u0007Fellowship, J Palacios by RETICS: RD06/0020/0013 and PI 08/\u0007Breast Cancer Res Treat (2010) 121:575–589 587\u0007123\u00076080971 from Instituto de Salud Carlos III and P07-CVI-03100 from\u0007Consejerı́a de Innovación Junta de Andalucı́a.\u0007References\u00071. Natrajan R, Lambros MB, Rodriguez-Pinilla SM et al (2009)\u0007Tiling path genomic profiling of grade 3 invasive ductal breast\u0007cancers. Clin Cancer Res 15:2711–2722\u00072. Bergamaschi A, Kim YH, Wang P et al (2006) Distinct patterns\u0007of DNA copy number alteration are associated with different\u0007clinicopathological features and gene-expression subtypes of\u0007breast cancer. Genes Chromosomes Cancer 45:1033–1040\u00073. Chin K, DeVries S, Fridlyand J et al (2006) Genomic and tran-\u0007scriptional aberrations linked to breast cancer pathophysiologies.\u0007Cancer Cell 10:529–541\u00074. Chin SF, Teschendorff AE, Marioni JC et al (2007) High-reso-\u0007lution aCGH and expression profiling identifies a novel genomic\u0007subtype of ER negative breast cancer. Genome Biol 8:R215\u00075. Hu X, Stern HM, Ge L et al (2009) Genetic alterations and\u0007oncogenic pathways associated with breast cancer subtypes. Mol\u0007Cancer Res 7:511–522\u00076. Mackay A, Tamber N, Fenwick K, et al. (2009) A high-resolution\u0007integrated analysis of genetic and expression profiles of breast\u0007cancer cell lines. Breast Cancer Res Treat. doi:10.1007/s10549-\u0007008-0296-7\u00077. Neve RM, Chin K, Fridlyand J et al (2006) A collection of breast\u0007cancer cell lines for the study of functionally distinct cancer\u0007subtypes. Cancer Cell 10:515–527\u00078. Perou CM, Sorlie T, Eisen MB et al (2000) Molecular portraits of\u0007human breast tumours. Nature 406:747–752\u00079. Sorlie T, Perou CM, Tibshirani R et al (2001) Gene expression\u0007patterns of breast carcinomas distinguish tumor subclasses with\u0007clinical implications. Proc Natl Acad Sci USA 98:10869–10874\u000710. Hicks J, Krasnitz A, Lakshmi B et al (2006) Novel patterns of\u0007genome rearrangement and their association with survival in\u0007breast cancer. Genome Res 16:1465–1479\u000711. Adelaide J, Finetti P, Bekhouche I et al (2007) Integrated pro-\u0007filing of basal and luminal breast cancers. Cancer Res 67:11565–\u000711575\u000712. Natrajan R, Lambros MB, Geyer FC et al (2009) Loss of 16q in\u0007high grade breast cancer is associated with estrogen receptor\u0007status: Evidence for progression in tumors with a luminal phe-\u0007notype? Genes Chromosomes Cancer 48:351–365\u000713. Heiser LM, Wang NJ, Talcott CL et al (2009) Integrated analysis\u0007of breast cancer cell lines reveals unique signaling pathways.\u0007Genome Biol 10:R31\u000714. Marchio C, Iravani M, Natrajan R et al (2008) Genomic and\u0007immunophenotypical characterization of pure micropapillary\u0007carcinomas of the breast. J Pathol 215:398–410\u000715. Simpson PT, Reis-Filho JS, Lambros MB et al (2008) Molecular\u0007profiling pleomorphic lobular carcinomas of the breast: evidence\u0007for a common molecular genetic pathway with classic lobular\u0007carcinomas. J Pathol 215:231–244\u000716. Weigelt B, Horlings HM, Kreike B et al (2008) Refinement of\u0007breast cancer classification by molecular characterization of his-\u0007tological special types. J Pathol 216:141–150\u000717. Wolff AC, Hammond ME, Schwartz JN et al (2007) American\u0007Society of Clinical Oncology/College of American Pathologists\u0007guideline recommendations for human epidermal growth factor\u0007receptor 2 testing in breast cancer. J Clin Oncol 25:118–145\u000718. Nielsen TO, Hsu FD, Jensen K et al (2004) Immunohistochemical\u0007and clinical characterization of the basal-like subtype of invasive\u0007breast carcinoma. Clin Cancer Res 10:5367–5374\u000719. Marchio C, Natrajan R, Shiu KK et al (2008) The genomic profile\u0007of HER2-amplified breast cancers: the influence of ER status.\u0007J Pathol 216:399–407\u000720. Huang J, Gusnanto A, O’Sullivan K et al (2007) Robust smooth\u0007segmentation approach for array CGH data analysis. Bioinfor-\u0007matics 23:2463–2469\u000721. Arriola E, Marchio C, Tan DS et al (2008) Genomic analysis of\u0007the HER2/TOP2A amplicon in breast cancer and breast cancer\u0007cell lines. Lab Invest 88:491–503\u000722. Reis-Filho JS, Simpson PT, Jones C et al (2005) Pleomorphic\u0007lobular carcinoma of the breast: role of comprehensive molecular\u0007pathology in characterization of an entity. J Pathol 207:1–13\u000723. Benjamini Y, Hochberg Y (1995) Controlling the false discovery\u0007rate: a practical and powerful approach to multiple testing. J Roy\u0007Stat Soc B 57:289–300\u000724. Hu Z, Fan C, Oh DS et al (2006) The molecular portraits of breast\u0007tumors are conserved across microarray platforms. BMC Genom-\u0007ics 7:96\u000725. Eisen MB, Spellman PT, Brown PO et al (1998) Cluster analysis\u0007and display of genome-wide expression patterns. Proc Natl Acad\u0007Sci USA 95:14863–14868\u000726. Tusher VG, Tibshirani R, Chu G (2001) Significance analysis of\u0007microarrays applied to the ionizing radiation response. Proc Natl\u0007Acad Sci USA 98:5116–5121\u000727. Weigelt B, Kreike B, Reis-Filho JS (2008) Metaplastic breast car-\u0007cinomas are basal-like breast cancers: a genomic profiling analysis.\u0007Breast Cancer Res Treat. doi:10.1007/s10549-008-0197-9\u000728. Reis-Filho JS, Savage K, Lambros MB et al (2006) Cyclin D1\u0007protein overexpression and CCND1 amplification in breast car-\u0007cinomas: an immunohistochemical and chromogenic in situ\u0007hybridisation analysis. Mod Pathol 19:999–1009\u000729. Reis-Filho JS, Pinheiro C, Lambros MB et al (2006) EGFR\u0007amplification and lack of activating mutations in metaplastic\u0007breast carcinomas. J Pathol 209:445–453\u000730. Tan DS, Lambros MB, Rayter S et al (2009) PPM1D is a\u0007potential therapeutic target in ovarian clear cell carcinomas. Clin\u0007Cancer Res 15:2269–2280\u000731. Parker JS, Mullins M, Cheang MC et al (2009) Supervised risk\u0007predictor of breast cancer based on intrinsic subtypes. J Clin\u0007Oncol 27:1160–1167\u000732. Cheng KW, Lahad JP, Kuo WL et al (2004) The RAB25 small\u0007GTPase determines aggressiveness of ovarian and breast cancers.\u0007Nat Med 10:1251–1256\u000733. Fiddes RJ, Campbell DH, Janes PW et al (1998) Analysis of Grb7\u0007recruitment by heregulin-activated erbB receptors reveals a novel\u0007target selectivity for erbB3. J Biol Chem 273:7717–7724\u000734. Hata T, Furukawa T, Sunamura M et al (2005) RNA interference\u0007targeting aurora kinase a suppresses tumor growth and enhances\u0007the taxane chemosensitivity in human pancreatic cancer cells.\u0007Cancer Res 65:2899–2905\u000735. Rayter S, Elliott R, Travers J et al (2008) A chemical inhibitor of\u0007PPM1D that selectively kills cells overexpressing PPM1D.\u0007Oncogene 27:1036–1044\u000736. Reis-Filho JS, Simpson PT, Turner NC et al (2006) FGFR1\u0007emerges as a potential therapeutic target for lobular breast car-\u0007cinomas. Clin Cancer Res 12:6652–6662\u000737. Kao J, Pollack JR (2006) RNA interference-based functional\u0007dissection of the 17q12 amplicon in breast cancer reveals con-\u0007tribution of coamplified genes. Genes Chromosomes Cancer\u000745:761–769\u000738. Jiang L, Zeng X, Yang H et al (2008) Oral cancer overexpressed\u00071 (ORAOV1): a regulator for the cell growth and tumor angio-\u0007genesis in oral squamous cell carcinoma. Int J Cancer 123:1779–\u00071786\u000739. Orsetti B, Nugoli M, Cervera N et al (2004) Genomic and\u0007expression profiling of chromosome 17 in breast cancer reveals\u0007588 Breast Cancer Res Treat (2010) 121:575–589\u0007123\u0007complex patterns of alterations and novel candidate genes. Can-\u0007cer Res 64:6453–6460\u000740. Guan Y, Kuo WL, Stilwell JL et al (2007) Amplification of PVT1\u0007contributes to the pathophysiology of ovarian and breast cancer.\u0007Clin Cancer Res 13:5745–5755\u000741. Subhawong AP, Subhawong T, Nassar H et al (2009) Most basal-\u0007like breast carcinomas demonstrate the same Rb-/p16? immu-\u0007nophenotype as the HPV-related poorly differentiated squamous\u0007cell carcinomas which they resemble morphologically. Am J Surg\u0007Pathol 33:163–175\u000742. Reis-Filho JS, Tutt AN (2008) Triple negative tumours: a critical\u0007review. Histopathology 52:108–118\u000743. Rakha EA, Reis-Filho JS, Ellis IO (2008) Basal-like breast can-\u0007cer: a critical review. J Clin Oncol 26:2568–2581\u000744. Bernard-Pierrot I, Gruel N, Stransky N et al (2008) Character-\u0007ization of the recurrent 8p11-12 amplicon identifies PPAPDC1B,\u0007a phosphatase protein, as a new therapeutic target in breast can-\u0007cer. Cancer Res 68:7165–7175\u000745. Eswaran J, Soundararajan M, Knapp S (2009) Targeting group II\u0007PAKs in cancer and metastasis. Cancer Metastasis Rev 28:209–217\u000746. Ihle NT, Powis G (2009) Take your PIK: phosphatidylinositol 3-\u0007kinase inhibitors race through the clinic and toward cancer\u0007therapy. Mol Cancer Ther 8:1–9\u000747. Crowder RJ, Phommaly C, Tao Y, et al. (2009) PIK3CA and\u0007PIK3CB inhibition produce synthetic lethality when combined\u0007with estrogen deprivation in estrogen receptor-positive breast\u0007cancer. Cancer Res 69:3955–3962\u000748. Desmedt C, Haibe-Kains B, Wirapati P et al (2008) Biological\u0007processes associated with breast cancer clinical outcome depend on\u0007the molecular subtypes. Clin Cancer Res 14:5158–5165\u000749. Wirapati P, Sotiriou C, Kunkel S et al (2008) Meta-analysis of gene\u0007expression profiles in breast cancer: toward a unified understanding\u0007of breast cancer subtyping and prognosis signatures. Breast Cancer\u0007Res 10:R65\u000750. Reis-Filho JS, Drury S, Lambros MB et al (2008) ESR1 gene\u0007amplification in breast cancer: a common phenomenon? Nat\u0007Genet 40:809–810 author reply 10-2\u000751. Brown LA, Hoog J, Chin SF et al (2008) ESR1 gene amplifica-\u0007tion in breast cancer: a common phenomenon? Nat Genet\u000740:806–807 author reply 10-2\u000752. Horlings HM, Bergamaschi A, Nordgard SH et al (2008) ESR1\u0007gene amplification in breast cancer: a common phenomenon? Nat\u0007Genet 40:807–808 author reply 10-2\u000753. Vincent-Salomon A, Gruel N, Lucchesi C et al (2007) Identifi-\u0007cation of typical medullary breast carcinoma as a genomic sub-\u0007group of basal-like carcinomas, a heterogeneous new molecular\u0007entity. Breast Cancer Res 9:R24\u000754. Lakhani SR, Reis-Filho JS, Fulford L et al (2005) Prediction of\u0007BRCA1 status in patients with breast cancer using estrogen\u0007receptor and basal phenotype. Clin Cancer Res 11:5175–5180\u000755. Foulkes WD, Stefansson IM, Chappuis PO et al (2003) Germline\u0007BRCA1 mutations and a basal epithelial phenotype in breast\u0007cancer. J Natl Cancer Inst 95:1482–1485\u000756. Turner NC, Reis-Filho JS (2006) Basal-like breast cancer and the\u0007BRCA1 phenotype. Oncogene 25:5846–5853\u000757. Turner NC, Reis-Filho JS, Russell AM et al (2007) BRCA1\u0007dysfunction in sporadic basal-like breast cancer. Oncogene 26:\u00072126–2132\u000758. Ashworth A (2008) A synthetic lethal therapeutic approach:\u0007poly(ADP) ribose polymerase inhibitors for the treatment of\u0007cancers deficient in DNA double-strand break repair. J Clin\u0007Oncol 26:3785–3790\u000759. Sorlie T, Tibshirani R, Parker J et al (2003) Repeated observation\u0007of breast tumor subtypes in independent gene expression data\u0007sets. Proc Natl Acad Sci USA 100:8418–8423\u000760. Zhang H, Li Z, Viklund EK et al (2002) P21-activated kinase 4\u0007interacts with integrin alpha v beta 5 and regulates alpha v beta 5-\u0007mediated cell migration. J Cell Biol 158:1287–1297\u0007Breast Cancer Res Treat (2010) 121:575–589 589\u0007123\n"
     ]
    }
   ],
   "source": [
    "print(articles.iloc[2].content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['preclin', 'studi', 'integr', 'genom', 'transcriptom', 'analysi', 'reveal', 'molecular', 'pathway', 'network', 'regul', 'copi', 'number', 'aberr', 'basal-lik', 'her2', 'lumin', 'cancer', 'rachael', 'natrajan', 'ae', 'britta', 'weigelt', 'ae', 'alan', 'mackay', 'ae', 'felip', 'c.', 'geyer', 'ae', 'anita', 'grigoriadi', 'ae', 'david', 's.', 'p.', 'tan', 'ae', 'chri', 'jone', 'ae', 'christoph', 'j.', 'lord', 'ae', 'radost', 'vatcheva', 'ae', 'socorro', 'm.', 'rodriguez-pinilla', 'ae', 'jose', 'palacio', 'ae', 'alan', 'ashworth', 'ae', 'jorg', 's.', 'reis-filho', 'receiv', '24', 'juli', '2009', 'accept', '28', 'juli', '2009', 'publish', 'onlin', '18', 'august', '2009', 'springer', 'science+busi', 'media', 'llc', '2009', 'abstract', 'breast', 'cancer', 'heterogen', 'diseas', 'caus', 'accumul', 'genet', 'chang', 'neo-', 'plastic', 'cell', 'hypothesis', 'molecular', 'subtyp', 'breast', 'cancer', 'may', 'driven', 'specif', 'constel', 'gene', 'whose', 'express', 'regul', 'gene', 'copi', 'number', 'aberr', 'address', 'question', 'ana-', 'lyse', 'seri', '48', 'microdissect', 'grade', 'iii', 'ductal', 'carcinoma', 'use', 'high', 'resolut', 'microarray', 'compar', 'genom', 'hybridis', 'mrna', 'express', 'array', '5,931', 'gene', 'whose', 'express', 'significantli', 'correl', 'copi', 'number', 'identifi', '1,897', 'gene', 'significantli', 'differenti', 'express', 'basal-lik', 'her2', 'lumin', 'tumour', 'inge-', 'nuiti', 'pathway', 'analysi', 'ipa', 'reveal', \"'g1/\", 'cell', 'cycl', 'regul', \"'brca1\", 'dna', 'damag', 'control', 'pathway', 'significantli', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'differ-', 'ential', 'express', 'molecular', 'subtyp', 'breast', 'cancer', 'ipa', 'gene', 'whose', 'express', 'signifi-', 'cantli', 'correl', 'copi', 'number', 'molecular', 'subtyp', 'individu', 'reveal', 'canon', 'pathway', 'involv', 'oestrogen', 'receptor', 'er', 'signal', 'dna', 'repair', 'enrich', 'gene', 'also', 'identifi', '32', '157', '265', 'gene', 'significantli', 'overexpress', 'amplifi', 'basal-lik', 'her2', 'lumin', 'cancer', 'respect', 'list', 'includ', 'known', 'novel', 'potenti', 'therapeut', 'target', 'e.g', 'her2', 'ppm1d', 'her2', 'cancer', 'result', 'provid', 'strong', 'circumstanti', 'evid', 'differ', 'pattern', 'genet', 'aberr', 'distinct', 'molecular', 'subtyp', 'breast', 'cancer', 'contribut', 'specif', 'transcriptom', 'profil', 'biolog', 'phenomena', 'characterist', 'subtyp', 'e.g', 'prolifer-', 'ation', 'her2', 'er', 'signal', 'may', 'driven', 'spe-', 'cific', 'pattern', 'copi', 'number', 'aberr', 'keyword', 'breast', 'cancer', 'molecular', 'subtyp', 'copi', 'number', 'gene', 'express', 'data', 'integr', 'molecular', 'pathway', 'rachael', 'natrajan', 'britta', 'weigelt', 'contribut', 'equal', 'present', 'studi', 'electron', 'supplementari', 'materi', 'onlin', 'version', 'articl', 'doi:10.1007/s10549-009-0501-3', 'contain', 'supplementari', 'materi', 'avail', 'author', 'user', 'r.', 'natrajan', 'a.', 'mackay', 'f.', 'c.', 'geyer', 'd.', 's.', 'p.', 'tan', 'c.', 'j.', 'lord', 'r.', 'vatcheva', 'a.', 'ashworth', 'j.', 's.', 'reis-filho', 'breakthrough', 'breast', 'cancer', 'research', 'centr', 'institut', 'cancer', 'research', '237', 'fulham', 'road', 'london', 'sw3', '6jb', 'uk', 'e-mail', 'rachael.natrajan', 'icr.ac.uk', 'j.', 's.', 'reis-filho', 'e-mail', 'jorge.reis-filho', 'icr.ac.uk', 'b.', 'weigelt', 'cancer', 'research', 'uk', 'london', 'research', 'institut', 'london', 'wc2a', '3px', 'uk', 'a.', 'grigoriadi', 'breakthrough', 'breast', 'research', 'unit', 'guy', \"'s\", 'hospit', 'london', 'se1', '9rt', 'uk', 'c.', 'jone', 'paediatr', 'oncolog', 'institut', 'cancer', 'research', 'sutton', 'sm2', '5ng', 'uk', 's.', 'm.', 'rodriguez-pinilla', 'centro', 'nacion', 'de', 'investigacion', 'oncologica', 'cnio', 'madrid', 'spain', 'j.', 'palacio', 'servicio', 'de', 'anatomia', 'patologica', 'hospit', 'virgen', 'del', 'rocio', 'sevil', 'spain', '123', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', 'doi', '10.1007/s10549-009-0501-3', 'introduct', 'breast', 'cancer', 'heterogen', 'diseas', 'compris', 'multi-', 'ple', 'entiti', 'distinct', 'biolog', 'featur', 'clinic', 'behaviour', 'recent', 'year', 'becom', 'appar', 'breast', 'cancer', 'heterogen', 'may', 'underpin', 'distinct', 'genom', 'chang', '1-7', 'microarray-bas', 'gene', 'express', 'profil', 'breast', 'cancer', 'led', 'identifi-', 'cation', 'least', 'five', 'distinct', 'class', 'breast', 'cancer', 'lumin', 'lumin', 'her2', 'normal', 'breast-lik', 'basal-', 'like', 'breast', 'cancer', 'hypothesis', 'differ-', 'enc', 'molecular', 'subgroup', 'breast', 'cancer', 'stem', 'differ', 'cell', 'origin', 'howev', 'lumin', 'her2', 'basal-lik', 'breast', 'cancer', 'shown', 'harbour', 'discret', 'pattern', 'genet', 'aberr', 'may', 'also', 'contribut', 'distinct', 'transcriptom', 'profil', 'clinic', 'phenotyp', '1-3', '10', 'gene', 'copi', 'number', 'aberr', 'observ', '95', 'breast', 'cancer', 'involv', 'substanti', 'proport', 'genom', 'breast', 'cancer', 'cell', '1-3', '10-12', 'unclear', 'howev', 'extent', 'copi', 'number', 'aberr', 'drive', 'express', 'gene', 'molecular', 'network', 'pathway', 'breast', 'cancer', 'previou', 'effort', 'identifi', 'network', 'pathway', 'either', 'carri', 'breast', 'cancer', 'cell', 'line', '13', 'made', 'use', 'non-microdissect', 'tissu', '2-5', '11', 'reduc', 'sensit', 'detect', 'copi', 'number', 'aberr', 'confound', 'transcriptom', 'profil', 'due', 'contamin', 'diploid', 'stromal', 'inflammatori', 'cell', 'aim', 'studi', 'twofold', 'determin', 'gene', 'whose', 'express', 'level', 'correl', 'gene', 'copi', 'number', 'gene', 'overexpress', 'amplifi', 'grade', 'iii', 'giii', 'breast', 'cancer', 'determin', 'canon', 'pathway', 'network', 'significantli', 'enrich', 'gene', 'microdissect', 'analys', 'seri', '48', 'giii', 'invas', 'ductal', 'carcinoma', 'special', 'type', 'idc-nst', 'high', 'resolut', 'microarray-bas', 'compar', 'genom', 'hybridis', 'acgh', 'mrna', 'express', 'array', 'studi', 'focus', 'giii', 'idc-nst', 'avoid', 'bia', 'introduc', 'distinct', 'pattern', 'genet', 'aberr', 'grade', 'iii', 'cancer', '12', 'invas', 'ductal', 'carcinoma', 'special', 'type', '14-16', 'materi', 'method', 'sampl', 'select', 'seri', '64', 'consecut', 'giii', 'idc-nst', 'case', 'retriev', 'univers', 'hospit', 'la', 'paz', 'madrid', 'spain', 'project', 'approv', 'local', 'research', 'ethic', 'committe', 'author', 'institut', 'four-lm', 'thick', 'section', 'tumour', 'subject', 'immunohistochemistri', 'antibodi', 'oestrogen', 'receptor', 'er', 'id5', '1:30', 'novocastra', 'progesteron', 'receptor', 'pr', '1a6', '1:30', 'novocastra', 'cytokeratin', 'ck', '5/6', 'd5/16', 'b4', '1:25', 'dako', 'epiderm', 'growth', 'factor', 'receptor', 'egfr', 'egfr', '31g7', '1:50', 'zyme', 'essenti', 'previous', 'describ', 'immunohistochemistri', 'her2', 'perform', 'use', 'herceptest', 'kit', 'dako', 'glostrup', 'denmark', 'her2', 'express', 'score', 'accord', 'asco/cap', 'guidelin', '17', 'case', 'display', 'score', 'consid', 'posit', 'case', 'dis-', 'play', 'score', 'label', 'equivoc', 'equivoc', 'case', 'her2', 'gene', 'statu', 'determin', 'use', 'fda', 'approv', 'spotlight', 'chromogen', 'situ', 'hybridis', 'cish', 'her2', 'probe', 'marker', 'score', 'two', 'pathologist', 'smp-r', 'jsr-f', 'blind', 'result', 'acgh', 'express', 'array', 'cish', 'analysi', 'tumour', 'classifi', 'her2', 'lumin', 'basal-lik', 'sub-', 'group', 'utilis', 'criteria', 'nielsen', 'et', 'al', '18', 'nucleic', 'acid', 'extract', 'ten', 'consecut', '8-lm', 'thick', 'frozen', 'section', '64', 'giii', 'tumour', 'microdissect', 'previous', 'describ', '14', '19', 'obtain', 'percentag', 'tumour', 'cell', 'remain', 'tissu', 'greater', '90', 'dna', 'extract', 'previous', 'describ', '12', 'dna', 'concentr', 'measur', 'picogreen', 'accord', 'manufactur', \"'s\", 'instruct', 'invitrogen', 'paisley', 'uk', 'rna', 'extract', 'use', 'trizol', 'accord', 'manufactur', 'instruct', 'invitrogen', 'uk', 'quantifi', 'use', 'agil', '2100', 'bioanalyz', 'rna', 'nano', 'labchip', 'kit', 'agil', 'biosystem', '16', 'case', 'dna', 'and/or', 'rna', 'insuffici', 'quantiti', 'acgh', 'and/or', 'express', 'array', 'analysi', 'exclud', 'studi', 'final', 'dataset', 'compris', '48', 'case', 'whose', 'clinico-patholog', 'characterist', 'summaris', 'supplementari', 'tabl', '1.', 'microarray-bas', 'compar', 'genom', 'hybridis', '32k', 'bacteri', 'artifici', 'chromosom', 'bac', 're-array', 'collect', 'chori', 'tile', 'path', 'acgh', 'platform', 'use', 'studi', 'construct', 'breakthrough', 'breast', 'cancer', 'research', 'centr', '14', 'dna', 'label', 'array', 'hy-', 'bridis', 'imag', 'acquisit', 'perform', 'describ', 'natrajan', 'et', 'al', 'case', '10', 'clone', 'miss', 'clone', 'data', 'avail', 'c10', 'case', 'exclud', 'log2', 'ratio', 'normalis', 'spatial', 'intensity-', 'depend', 'bias', 'use', 'two-dimension', 'loess', 'regres-', 'sion', 'follow', 'bac-depend', 'bia', 'correct', '576', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'previous', 'describ', '14', '20', 'polymorph', 'bac', 'identi-', 'fie', 'analysi', '50', 'male', 'femal', 'femal', 'femal', 'hybridis', 'filter', 'left', 'final', 'dataset', '31,544', 'clone', 'unambigu', 'map', 'inform', 'accord', 'march', '2006', 'build', 'hg18', 'human', 'genom', 'http', '//www.ensembl.org', 'data', 'smooth', 'use', 'circular', 'binari', 'segment', 'cb', 'algorithm', 'categor', 'analysi', 'appli', 'bac', 'clas-', 'sifi', 'repres', 'gain', 'loss', 'no-chang', 'accord', 'smooth', 'log2', 'ratio', 'valu', '21', 'threshold', 'valu', 'chosen', 'correspond', 'three', 'stan-', 'dard', 'deviat', 'normal', 'ratio', 'obtain', 'fil-', 'tere', 'clone', 'map', 'chromosom', '1-22', 'assess', 'multipl', 'hybridis', 'dna', 'extract', 'pool', 'male', 'femal', 'blood', 'donor', 'previous', 'describ', '21', '22', 'log2', 'ratio', '+-0.08', 'low', 'level', 'gain', 'defin', 'smooth', 'log2', 'ratio', '0.08', '0.45', 'corre-', 'spond', 'approxim', '3-5', 'copi', 'locu', 'whilst', 'gene', 'amplif', 'defin', 'log2', 'ratio', '0.45', 'correspond', 'copi', '19', 'gene', 'express', 'profil', 'gene', 'express', 'profil', 'perform', 'use', 'illumina', 'human', 'wg6', 'version', 'express', 'array', 'accord', 'manufactur', \"'s\", 'protocol', 'briefli', '200', 'ng', 'rna', 'sampl', 'amplifi', 'use', 'illumina', 'totalprep', 'rna', 'amplif', 'kit', 'follow', 'manufactur', \"'s\", 'instruct', 'ambion', 'uk', 'label', 'crna', 'subsequ', 'hybri-', 'dise', 'illumina', 'sentrix-human', 'version', 'express', 'bead-chip', 'full', 'detail', 'rna', 'amplif', 'hybrid-', 'isat', 'found', 'www.illumina.com', 'raw', 'gene', 'express', 'valu', 'robust-splin', 'normalis', 'use', 'bioconductor', 'lumi', 'packag', 'http', '//www.bioconductor.org/', 'packages/2.3/bioc/html/lumi.html', 'r.', 'gene', 'map-', 'ped', 'genom', 'locat', 'use', 'lumihumanallv2', 'annot', 'databas', 'avail', 'bioconductor', 'illumina', 'transcript', 'probe', 'detect', 'values\\\\0.01', '25', 'sampl', 'includ', 'result', 'dataset', '12,699', 'transcript', 'regul', 'probe', 'accur', 'unequivoc', 'map', 'inform', 'gene', 'express', 'data', 'publicli', 'avail', 'arrayexpress', 'http', '//www.ebi', 'ac.uk/microarray-as/ae/', 'access', 'number', 'e-tabm-543', 'integr', 'acgh', 'express', 'analysi', 'identifi', 'gene', 'whose', 'express', 'correl', 'genet', 'copi', 'number', 'chang', 'cbs-smooth', 'acgh', 'data', 'use', 'assign', 'median', 'acgh', 'state', '12,699', 'gene', 'gene', 'express', 'dataset', 'use', 'median', 'valu', 'bac', 'overlap', 'genom', 'posi-', 'tion', 'gene', 'result', '1:1', 'matrix', 'express', 'data', 'acgh', 'valu', 'use', 'correl', 'pearson', \"'s\", 'correl', 'perform', 'mrna', 'express', 'log', 'intens', 'valu', 'median', 'cbs-smooth', 'ratio', 'deriv', 'acgh', 'analysi', 'gene', 'valu', 'test', 'adjust', 'benjamini', 'hochberg', 'multipl', 'valu', 'adjust', '23', 'identifi', 'gene', 'consist', 'overexpress', 'amplifi', 'region', 'recurr', 'amplif', 'mann-', 'whitney', 'test', 'perform', 'adjust', 'ben-', 'jamini', 'hochberg', 'multipl', 'valu', 'adjust', 'adjust', 'valu', '\\\\0.05', 'consid', 'signific', 'unsupervis', 'analysi', 'perform', 'hierarch', 'cluster', '48', 'primari', 'tumour', 'use', '848', 'uniqu', 'gene', 'identifi', '1,300', 'intrins', 'gene', 'describ', 'hu', 'et', 'al', '24', 'gene', 'median', 'centr', 'averag', 'linkag', 'cluster', 'perform', 'use', 'pearson', \"'s\", 'correl', 'similar', 'metric', 'cluster', '25', 'result', 'subse-', 'quentli', 'visualis', 'use', 'java', 'treeview', 'http', '//jtreeview', 'sourceforge.net/', 'sam', 'softwar', '26', 'employ', 'determin', 'gene', 'significantli', 'differenti', 'express', 'basal-lik', 'her2', 'lumin', 'tumour', 'adopt', 'fals', 'discoveri', 'rate', '1,000', 'permut', 'ingenu', 'pathway', 'analysi', 'determin', 'pathway', 'network', 'significantli', 'regul', 'gene', 'express', 'data', 'basal-lik', 'her2', 'lumin', 'giii', 'tumour', 'perform', 'pathway', 'analysi', 'use', 'ingenu', 'pathway', 'analysi', 'ipa', 'program', 'http', '//www', 'ingenuity.com', 'hugo', 'gene', 'identifi', 'map', 'network', 'avail', 'ingenu', 'databas', 'rank', 'score', 'score', 'indic', 'likelihood', 'gene', 'network', 'found', 'togeth', 'due', 'random', 'chanc', 'use', '99', 'confid', 'level', 'score', 'c3', 'consid', 'signifi-', 'cant', 'ipa', 'perform', 'essenti', 'previous', 'describ', '16', '27', 'gene-set', 'describ', 'fig', '1.', 'exact', 'hypergeometr', 'probabl', 'analysi', 'gene', 'list', 'enrich', 'analysi', 'gene', 'list', 'enrich', 'perform', 'use', 'hypergeometr', 'probabl', 'analysi', 'wherebi', 'number', 'gene', 'common', 'two', 'group', 'identifi', 'represent', 'factor', 'number', 'overlap', 'gene', 'divid', 'expect', 'number', 'overlap', 'gene', 'drawn', 'two', 'independ', 'group', 'calcul', 'represent', 'factor', 'indic', 'overlap', 'expect', 'two', 'independ', 'group', 'wherea', 'represen-', 'tation', 'factor', '\\\\1', 'indic', 'less', 'overlap', 'expect', 'probabl', 'find', 'overlap', 'number', 'gene', 'calcul', 'use', 'hypergeometr', 'probabl', 'formula', '/c', 'http', '//', 'elegans.uky.edu/ma/progs/representation.stats.html', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '577', '123', 'valid', 'acgh', 'microarray-bas', 'express', 'profil', 'case', 'review', 'two', 'pathologist', 'smp-r', 'jsr-f', 'repres', 'area', '48', 'case', 'includ', 'tissu', 'microarray', 'contain', 'two', '1-mm', 'rep-', 'licat', 'core', 'case', 'case', 'use', 'valid', 'select', 'amplif', 'found', 'acgh', 'immunohistochemistri', 'cyclin', 'd1', 'perform', 'sp4', 'rabbit', 'monoclon', 'antibodi', 'neomark', 'suffolk', 'uk', '1:50', 'dilut', 'previous', 'describ', '28', 'score', 'use', 'allr', 'score', 'system', '28', 'immunohistochem', 'analysi', 'perform', 'two', 'author', 'smp-r', 'jsr-f', 'observ', 'blind', 'result', 'acgh', 'gene', 'express', 'cish', 'analys', 'chromogen', 'situ', 'hybridis', 'perform', 'use', 'zymed/invitrogen', 'spotlight', 'cish', 'probe', 'her2', 'ccnd1', 'egfr', '21', '28', '29', 'in-hous', 'gener', 'ppm1d', 'probe', '30', 'hybridis', 'wash', 'perform', 'previous', 'describ', '21', '28-30', 'gene', 'amplif', 'defin', 'presenc', 'signal', 'larg', 'gene', 'cluster', 'combin', 'signal', 'cluster', '50', 'morpholog', 'unequivoc', 'neoplast', 'cell', 'previous', 'describ', '21', '28-30', 'cish', 'analysi', 'perform', 'two', 'author', 'smp-r', 'fcg', 'observ', 'blind', 'result', 'acgh', 'gene', 'express', 'immunohistochem', 'analys', 'fig', 'work', 'flow', 'diagram', 'integr', 'acgh', 'gene', 'express', 'data', 'analysi', 'cohort', '48', 'grade', 'iii', 'invas', 'ductal', 'carcinoma', 'special', 'type', 'analysi', '48', 'grade', 'iii', 'invas', 'ductal', 'carcinoma', 'special', 'type', 'assign', 'molecular', 'subtyp', 'basal-lik', '13', 'her2', '13', 'lumin', '22', 'separ', '1nielsen', 'et', 'al', '18', 'mwu', 'mann-whitney', 'test', 'adjp', 'adjust', 'valu', '578', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'quantit', 'real-tim', 'revers', 'transcriptas', 'pcr', 'rna', '48', 'microdissect', 'case', 'avail', 'quan-', 'tit', 'real-tim', 'revers', 'transcriptas', 'pcr', 'qrt-pcr', 'analysi', 'first', 'strand', 'synthesi', 'perform', 'previous', 'describ', 'qrt-pcr', 'perform', 'taqman', 'chemistri', 'abi', 'prism', '7900ht', 'appli', 'biosystem', 'use', 'standard', 'curv', 'method', '21', '30', 'assay', 'ppm1d', 'tfrc', 'purchas', 'appli', 'biosys-', 'tem', 'ppm1d', 'express', 'level', 'normalis', 'gene', 'tfrc', 'assay', 'demand', 'id', 'hs00174609_m1-tfrc', 'hs00186230_m1-ppm1d', '30', 'result', 'breast', 'cancer', 'molecular', 'subtyp', 'classif', 'determin', 'basal-lik', 'lumin', 'her2', 'molecular', 'subtyp', 'class', 'subject', '48', 'giii', 'idc-nst', 'val-', 'idat', 'immunohistochem', 'surrog', 'molecular', 'subtyp', 'breast', 'cancer', 'describ', 'nielsen', 'et', 'al', '18', 'taxonomi', 'employ', 'given', 'use', 'classifi', 'archiv', 'tumour', 'sampl', 'accur', 'identifi', 'clinic', 'relev', 'subgroup', 'her2-posi-', 'tive/amplifi', 'tumour', 'i.e', 'amen', 'therapi', 'anti-', 'her2', 'agent', 'use', 'hu', 'et', 'al', \"'s\", '24', 'parker', 'et', 'al', \"'s\", '31', 'microarray-bas', 'classif', 'proport', 'er-posit', 'her2', 'amplifi', 'cancer', 'classifi', 'lumin', 'phenotyp', 'base', 'express', 'her2', 'er', 'ck', '5/6', 'egfr', '13', '22', '13', 'tumour', 'classifi', 'her2', 'lumin', 'basal-lik', 'phenotyp', 'respect', 'supple-', 'mentari', 'tabl', 'expect', 'good', 'agreement', 'classif', 'tumour', 'basal-lik', 'her2', 'lumin', 'use', 'nielsen', 'et', 'al', \"'s\", '18', 'immunohistochem', 'surrog', 'express', 'profil', 'use', 'hu', 'et', 'al', \"'s\", 'centroid', '24', 'observ', 'kappa', 'score', '0.753', '910-11', 'supplementari', 'tabl', 'expect', '31', 'case', 'harbour', 'her2', 'amplif', 'consist', 'classifi', 'her2', 'phenotyp', 'nielsen', 'et', 'al', '18', 'fell', 'either', 'her2', 'lumin', 'subgroup', 'use', 'hu', 'et', 'al', \"'s\", 'centroid', '24', 'identif', 'gene', 'whose', 'express', 'associ', 'gene', 'copi', 'number', 'defin', 'gene', 'whose', 'express', 'significantli', 'corre-', 'late', 'gene', 'copi', 'number', 'extract', 'dna', 'rna', 'microdissect', 'sampl', 'perform', 'acgh', 'anal-', 'ysi', 'use', 'bac', 'array', 'platform', 'actual', 'resolut', '50', 'kb', 'genome-wid', 'gene', 'express', 'profil', 'use', 'illumina', 'wg6', 'v2', 'platform', 'next', 'overlaid', 'acgh', 'gene', 'express', 'data', '48', 'giii', 'idc-nst', '12,699', 'significantli', 'regul', 'gene', '5,931', '46.70', 'display', 'mrna', 'express', 'level', 'significantli', 'cor-', 'relat', 'copi', 'number', 'pearson', \"'s\", 'correl', 'adjust', '0.05', 'supplementari', 'tabl', 'gene-set', 'fig', '1a', 'list', 'includ', 'gene', 'previous', 'found', 'increas', 'express', 'level', 'determin', 'gain', 'copi', 'number', 'oraov1', 'ash2l', 'fadd', 'rps6', 'kb1', 'nme1', 'eif4ebp2', 'grb7', 'ints2', 'perld1', 'myst3', 'pig', 'ints4', 'tlk2', 'ppm1d', 'prosc', 'stard3', 'erbb2', 'cttn', '3-6', '11', 'identifi', 'potenti', 'amplicon', 'driver', 'determin', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'recurr', 'amplifi', '811', 'pearson', \"'s\", 'correl', 'adjust', '0.05', 'amplifi', 'tumour', 'sup-', 'plementari', 'tabl', 'gene-set', 'gene', 'significantli', 'overexpress', 'amplifi', '469', 'mann-whitney', 'test', 'adjust', '0.05', 'supplementari', 'tabl', 'gene-set', 'surprisingli', 'list', 'gene', 'includ', 'known', 'driver', 'amplicon', '1q', '8p', '8q', '11q13', '17q12-q21', '17q22-q25', '20q13', '3-5', '11', '32-36', 'unbias', 'approach', 'accur', 'identifi', 'known', 'amplicon', 'driver', 'includ', 'her2', 'stard3', 'grb7', '37', '17q12', 'oraov1', '38', '11q13', 'ppm1d', '30', '17q23.2', 'gene', 'whose', 'express', 'activ', 'shown', 'requir', 'prolifer', 'and/or', 'surviv', 'cancer', 'cell', 'harbour', 'amplif', 'addit', 'analysi', 'also', 'reveal', 'addit', 'gene', 'chromosom', '17q', 'pertain', 'smallest', 'region', 'amplif', '17q12-', 'q21', '17q22-q25', 'significantli', 'overexpress', 'amplifi', 'e.g', 'mrpl45', 'fbxl20', 'med1', 'orm', 'dl3', 'fig', '2a', 'suggest', 'report', 'complex', 'pattern', 'genet', 'aberr', 'chromosom', '17', '39', 'may', 'biolog', 'impact', 'i.e', 'overexpress', 'specif', 'gene', 'mere', 'genom', \"'nois\", 'similar', 'find', 'observ', 'genom', 'region', 'adjac', 'amplicon', '11q13.2-q13.3', '17q21-q23', 'fig', '2b', 'hand', '8q23.2-24.3', 'amplicon', 'display', 'significantli', 'complex', 'pattern', 'genet', 'aberr', 'without', 'well-defin', 'core', 'unbias', 'analysi', 'dem-', 'onstrat', 'utp23', 'rad21', 'med30', 'mrpl13', 'mtbp', 'c8orf76', 'zhx1', 'c8orf32', 'fbxo32', 'ndufb9', 'kiaa0196', 'nsmce2', 'ddef1', 'efr3a', 'phf20l1', 'st3gal1', 'trappc9', 'fig', '2d', 'overexpress', 'amplifi', 'constitut', 'potenti', 'driver', 'amplicon', 'note', 'myc', 'map', 'region', 'express', 'level', 'correl', 'copi', 'number', 'consis-', 'tentli', 'overexpress', 'amplifi', 'previous', 'describ', '40', 'taken', 'togeth', 'result', 'previou', 'function', 'correl', 'studi', '11', '13', '37', '38', 'analysi', 'seem', 'identifi', 'accur', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'potenti', 'amplicon', 'driver', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '579', '123', 'molecular', 'pathway', 'regul', 'gene', 'copi', 'number', 'differ', 'basal-lik', 'her2', 'lumin', 'cancer', 'determin', '5,931', 'gene', 'whose', 'express', 'level', 'significantli', 'correl', 'gene', 'copi', 'number', 'associ', 'molecular', 'subtyp', 'giii', 'breast', 'can-', 'cer', 'i.e', 'basal-lik', 'her2', 'lumin', 'multiclass', 'signifi-', 'canc', 'analysi', 'microarray', 'sam', 'perform', '26', 'use', '1,000', 'permut', 'accept', 'fals', 'discoveri', 'rate', 'sam', 'analysi', 'reveal', '1,897', 'gene', 'gene-', 'set', 'fig', '1a', 'differenti', 'express', 'giii', 'idc-nst', 'basal-lik', 'her2', 'lumin', 'molecular', 'sub-', 'type', 'supplementari', 'tabl', 'expect', 'gene', 'per-', 'tain', 'her2', 'amplicon', 'perld1', 'c17orf37', 'grb7', 'stard3', 'her2', 'erbb2', 'significantli', 'overex-', 'press', 'cancer', 'her2', 'phenotyp', 'top', 'rank', 'gene', 'sam', 'analysi', 'ingenu', 'pathway', 'analysi', 'ipa', '1,897', 'tran-', 'script', 'reveal', 'network', 'pathway', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'differenti', 'express', 'molecular', 'subtyp', 'includ', 'network', 'relat', 'dna', 'replic', 'repair', 'cell', 'cycl', 'fig', 'supplementari', 'fig', 'sup-', 'plementari', 'tabl', 'instanc', 'network', \"'gene\", 'express', 'dna', 'replic', 'recombin', 'repair', 'cell', 'cycl', 'cancer', 'score', '40', 'fig', 'encompass', 'myc', 'key', 'gene', 'mdc1', 'mediat', 'dna', 'damag', 'checkpoint', 'msh3', 'mlh1', 'mlh3', 'play', 'role', 'mismatch', 'repair', 'rad50', 'involv', 'dna', 'double-strand', 'break', 'repair', 'basal-lik', 'cancer', 'dis-', 'play', 'recurr', 'copi', 'number', 'gains/amplif', 'upregul', 'myc', 'mdc1', 'loss', 'down-', 'regul', 'rad50', 'gene', 'relat', 'mismatch', 'repair', 'i.e', 'msh3', 'mlh1', 'mlh3', 'compar', 'her2', 'lumin', 'cancer', 'fig', 'network', \"'gene\", 'express', 'dna', 'replic', 'recombin', 'repair', 'cell', 'cycl', 'score', '40', 'supplementari', 'fig', 'centr', 'around', 'key', 'gene', 'involv', 'cell', 'cycl', 'ccnd1', 'e2f3', 'ccna2', 'ccndbp1', 'basal-lik', 'tumour', 'show', 'recurr', 'delet', 'downregul', 'ccnd1', 'fig', 'match', 'heatmap', 'gene', 'express', 'acgh', 'within', 'region', 'recurr', 'amplif', '17q12-q21.1', '11q13.2-q13.3', '17q21-q23', '8q23.3-q24.3', 'amplicon', 'gene', 'within', 'amplifi', 'region', 'recov', 'median', 'acgh', 'valu', 'state', 'assign', 'sampl', 'separ', 'harbour', 'amplifica-', 'tion', 'within', 'region', 'express', 'acgh', 'valu', 'depict', 'two', 'match', 'heatmap', 'acgh', 'state', 'left', 'express', 'valu', 'right', 'gene', 'order', 'accord', 'chromosom', 'posit', 'bar', 'plot', 'show', 'result', 'mann-whitney', 'test', 'express', 'continu', 'variabl', 'gene', 'amplif', 'group', 'variabl', 'bar', 'red', 'show', 'adjust', 'valu', 'less', '0.05.', 'acgh', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'bright', 'red', 'gene', 'amplif', 'gene', 'express', 'green', 'downregul', 'red', 'upregul', 'mwu', 'mann-whitney', 'test', 'adjp', 'adjust', 'valu', '580', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'copi', 'number', 'gain', 'upregul', 'e2f3', 'her2', 'wherea', 'lumin', 'cancer', 'harbour', 'recurr', 'amplifica-', 'tion', 'overexpress', 'ccnd1', 'less', 'frequent', 'gain', 'e2f3', 'supplementari', 'fig', 'agreement', 'ipa', 'also', 'identifi', 'canon', 'pathway', \"'cell\", 'cycl', 'g1/', 'checkpoint', 'regu-', 'lation', \"'role\", 'brca1', 'dna', 'damag', 'respons', 'significantli', 'enrich', 'gene', 'whose', 'express', 'cor-', 'relat', 'copi', 'number', 'significantli', 'differenti', 'express', 'breast', 'cancer', 'molecular', 'subtyp', '0.001', '0.0069', 'respect', 'supplementari', 'tabl', 'analysi', 'reveal', 'express', 'level', 'distinct', 'compon', 'g1/', 'checkpoint', 'regul', 'pathway', 'differenti', 'express', 'basal-lik', 'lumin', 'her2', 'cancer', 'due', 'specif', 'pattern', 'copi', 'number', 'aberr', 'fig', 'recurr', 'gains/amplif', 'upregul', 'cdk6', 'p16ink4a', 'e2f', 'ccne1', 'myc', 'observ', 'basal-lik', 'breast', 'cancer', 'wherea', 'ccne1', 'ccnd1', 'recurr', 'gained/amplifi', 'upregul', 'her2', 'tumour', 'ccnd1', 'lumin', 'tumour', 'skp1', 'skp2', 'recurr', 'delet', 'downregul', 'basal-lik', 'breast', 'cancer', 'wherea', 'recurr', 'gain', 'overexpress', 'skp2', 'found', 'her2', 'cancer', 'recurr', 'gain', 'overexpress', 'gene', 'observ', 'lumin', 'cancer', 'interestingli', 'p16', 'cdkn2a', 'gene', 'report', 'overexpress', 'major', 'basal-lik', 'breast', 'cancer', '41', 'recurr', 'gain', 'overex-', 'press', 'tumour', 'wherea', 'her2', 'lumin', 'can-', 'cer', 'harbour', 'recurr', 'delet', 'downregul', 'gene', 'gene', 'involv', 'brca1', 'dna', 'damag', 'control', 'fig', 'network', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'significantli', 'differenti', 'express', 'basal-lik', 'her2', 'lumin', 'giii', 'invas', 'ductal', 'carcinoma', 'special', 'type', 'defin', 'ingenu', 'pathway', 'analysi', 'softwar', \"'gene\", 'express', 'dna', 'replic', 'recombin', 'repair', 'cell', 'cycl', 'cancer', 'network', 'score', '40', 'basal-lik', 'ii', 'her2', 'iii', 'lumin', 'tumour', 'green', 'downregul', 'red', 'upregul', 'match', 'heatmap', 'acgh', 'categor', 'data', 'gene', 'differenti', 'express', 'basal-lik', 'ii', 'her2', 'iii', 'lumin', 'tumour', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'bright', 'red', 'gene', 'amplif', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '581', '123', 'dna', 'repair', 'mechan', 'includ', 'rb', 'mlh1', 'rad50', 'recurr', 'delet', 'downregul', 'e2f3', 'mre11a', 'fanc', 'recurr', 'copi', 'number', 'gain', 'upregul', 'basal-lik', 'cancer', 'compar', 'her2', 'lumin', 'tumour', 'supplementari', 'fig', 'although', 'brca1', 'reach', 'signific', 'multi-', 'class', 'sam', 'analysi', 'express', 'associ', 'copi', 'number', 'pearson', \"'s\", 'correl', 'adjust', '0.0433', 'lower', 'express', 'associ', 'basal-lik', 'tumour', '0.0283', 't-test', 'homoscedast', 'adjust', '0.01499', 'mann-whitney', 'test', 'addit', \"'protein\", 'ubiquitina-', 'tion', '0.001', \"'igf-1\", '0.0018', 'amongst', 'top', 'canon', 'pathway', 'significantli', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'supple-', 'mentari', 'tabl', 'result', 'suggest', 'transcript', 'regul', 'molecular', 'pathway', 'may', 'determin', 'pattern', 'genet', 'aberr', 'breast', 'cancer', 'distinct', 'pathway', 'network', 'driven', 'specif', 'pattern', 'copi', 'number', 'aberr', 'basal-lik', 'her2', 'lumin', 'tumour', 'corrobor', 'hypothes', 'overlaid', 'acgh', 'express', 'data', 'subtyp', 'separ', 'pearson', \"'s\", 'correl', 'adjust', '0.05', 'supplementari', 'tabl', 'result', 'list', '730', 'gene-set', 'e1', '933', 'gene-set', 'f1', '2,137', 'gene-set', 'g1', 'see', 'fig', '1b', 'gene', 'whose', 'express', 'associ', 'copi', 'number', '13', 'basal-lik', '13', 'her2', '22', 'lumin', 'cancer', 'respect', 'signific', 'overlap', 'defin', 'exact', 'hypergeometr', 'probabl', 'analysi', 'found', 'gene', 'identifi', 'copi', 'number', 'regul', 'approach', 'basal-lik', 'gene-set', 'vs.', 'gene-set', 'e1', 'rep-', 'resent', 'factor', '2.0', '10-25', 'her2', 'gene-set', 'vs.', 'gene-set', 'f1', 'represent', 'factor', '2.1', '10-37', 'lumin', 'gene-set', 'vs.', 'gene-set', 'g1', 'represent', 'factor', '1.9', '10-71', 'cancer', 'ingenu', 'pathway', 'analysi', 'gene-set', 'e1', 'f1', 'g1', 'pearson', \"'s\", 'correl', 'adjust', '0.05', 'reveal', 'fig', 'canon', 'pathway', \"'cell\", 'cycl', 'g1/', 'checkpoint', 'regul', 'significantli', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'differenti', 'express', 'basal-lik', 'her2', 'lumin', 'giii', 'tumour', '0.001', 'defin', 'ingenu', 'pathway', 'analysi', 'softwar', 'canon', 'pathway', 'basal-lik', 'ii', 'her2', 'iii', 'lumin', 'tumour', 'green', 'downregul', 'red', 'upregul', 'match', 'heatmap', 'acgh', 'data', 'basal-lik', 'ii', 'her2', 'iii', 'lumin', 'tumour', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'bright', 'red', 'gene', 'amplif', '582', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'network', 'pathway', 'independ', 'associ', 'molecular', 'subtyp', 'supplementari', 'tabl', 'analysi', 'reveal', 'heterogen', 'pattern', 'loss', 'gain', 'amplif', 'gene', 'whose', 'express', 'level', 'correl', 'copi', 'number', 'tumour', 'subtyp', 'lead', 'activ', 'distinct', 'network', 'pathway', 'basal-lik', 'cancer', 'gene-set', 'e1', \"'oestrogen\", 'receptor', 'signal', '0.00178', 'fig', \"'protein\", 'ubiquitin', '0.00181', 'canon', 'pathway', 'significantli', 'enrich', 'gene', 'whose', 'express', 'corre-', 'late', 'copi', 'number', 'supplementari', 'tabl', 'gene', 'pertain', \"'oestrogen\", 'receptor', 'signal', 'pathway', 'vast', 'major', 'lost', 'downregul', 'except', 'pgr', 'gain', 'amplifi', 'one', 'basal-lik', 'case', 'b11', 'express', 'high', 'mrna', 'protein', 'level', 'progesteron', 'receptor', 'pr', 'supplementari', 'tabl', 'strong', 'correl', 'observ', 'er', 'pr', 'statu', 'defin', 'immu-', 'nohistochemistri', 'esr1', 'pgr', 'mrna', 'express', 'level', 'defin', 'microarray-bas', 'express', 'analysi', 'mann-whitney', 'test', '5.5', '10-9', '1.2', '10-7', 'respect', 'supplementari', 'tabl', \"'oestrogen\", 'receptor', 'signal', 'canon', 'pathway', 'also', 'signifi-', 'cantli', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'her2', 'cancer', 'gene-set', 'f1', '0.01288', 'fig', 'howev', 'gene', 'copi', 'number', 'regu-', 'late', 'subtyp', 'larg', 'distinct', 'her2', 'tumour', 'raf1', 'erk1/2', 'recurr', 'lost', 'wherea', 'gain', 'multipl', 'compon', 'rna', 'polymeras', 'ii', 'complex', 'includ', 'polr2k', 'tata-bind', 'protein-', 'associ', 'factor', 'gene', 'i.e', 'taf4', 'taf15', 'observ', 'her2', 'cancer', \"'nucleotid\", 'excis', 'repair', 'pathway', 'significantli', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', '0.0214', 'recurr', 'gain', 'amplif', 'polr2k', 'loss', 'rpa2', 'fig', 'supplementari', 'tabl', 'result', 'suggest', 'sever', 'molecular', 'pathway', 'network', 'specif', 'associ', 'basal-', 'like', 'her2', 'lumin', 'cancer', 'driven', 'least', 'part', 'specif', 'genet', 'aberr', 'importantli', 'dna', 'copi', 'number', 'chang', 'appear', 'involv', 'downregu-', 'lation', 'oestrogen', 'receptor', 'signal', 'one', 'hallmark', 'featur', 'basal-lik', 'cancer', '42', '43', 'dys-', 'function', 'statu', 'gene', 'pathway', 'network', 'relat', 'variou', 'type', 'dna', 'repair', 'mechan', 'pathway', 'analysi', 'identifi', 'novel', 'potenti', 'therapeut', 'target', 'giii', 'idc-nst', 'basal-lik', 'her2', 'lumin', 'molecular', 'subtyp', 'previou', 'studi', 'demonstr', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'recurr', 'amplifi', 'gene', 'consist', 'overexpress', 'amplifi', 'like', 'amplicon', 'driver', 'may', 'con-', 'stitut', 'therapeut', 'target', '30', '36', '44', 'identifi', 'potenti', 'therapeut', 'target', 'specif', 'subgroup', 'basal-lik', 'her2', 'lumin', 'cancer', 'defin', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'molecular', 'subtyp', 'individu', 'pearson', \"'s\", 'correl', 'adjust', '0.05', 'recurr', 'amplifi', 'group', 'gene', 'overexpress', 'amplifi', 'group', 'individu', 'mann-whitney', 'adjust', '0.05', 'fig', '1b', 'first', 'analysi', 'result', 'list', '32', '157', '265', 'gene', 'basal-lik', 'her2', 'lumin', 'tumour', 'respect', 'gene-set', 'e2', 'f2', 'g2', 'respect', 'wherea', 'second', 'result', '16', '33', '159', 'gene', 'basal-lik', 'her2', 'lumin', 'tumour', 'respect', 'gene-', 'set', 'e3', 'f3', 'g3', 'respect', 'supplementari', 'tabl', '10', 'signific', 'overlap', 'defin', 'exact', 'hypergeo-', 'metric', 'probabl', 'analysi', 'found', 'gene', 'identifi', 'recurr', 'amplifi', 'gene', 'overexpress', 'amplifi', 'basal-lik', 'gene-set', 'e2', 'vs.', 'gene-set', 'e3', 'represent', 'factor', '372.0', '10-40', 'her2', 'gene-set', 'f2', 'vs.', 'gene-set', 'f3', 'represent', 'factor', '68.8', '10-50', 'lumin', 'gene-set', 'g2', 'vs.', 'gene-set', 'g3', 'represent', 'factor', '44.3', '10-252', 'tumour', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'recurr', 'amplifi', 'basal-lik', 'cancer', 'gene-set', 'e2', 'map', '1q22-q24', '1q44', '8q24.1', '10p15', '19q12-q13', 'ipa', 'reveal', 'network', \"'gene\", 'express', 'cancer', 'gastrointestin', 'diseas', 'score', '46', \"'gene\", 'express', 'cancer', 'cell', 'death', 'score', '17', 'significantli', 'enrich', 'gene', 'supplementari', 'tabl', '11', 'supplementari', 'fig', 'interestingli', 'differ', 'compon', 'net-', 'work', 'amplifi', 'individu', 'case', 'given', 'limit', 'sampl', 'size', 'gene', 'map', '19q12-q13', 'sig-', 'nificantli', 'overexpress', 'amplifi', 'basal-lik', 'can-', 'cer', 'multipl', 'comparison', 'adjust', 'gene-set', 'e3', 'fig', '1b', 'gene', 'includ', 'c19orf2', 'med30', 'supt', '5h', 'sertad1', 'sertad3', 'part', 'network', \"'cancer\", 'cell', 'death', 'endocrin', 'system', 'disord', 'score', '40', 'supplementari', 'fig', 'notabl', 'p21-activ', 'kinas', 'pak4', 'present', \"'gene\", 'express', 'cancer', 'cell', 'death', \"'cancer\", 'cell', 'death', 'endocrin', 'system', 'disord', 'network', 'recent', 'emerg', 'potenti', 'therapeut', 'target', '45', 'her2', 'cancer', 'network', \"'lipid\", 'metabol', 'molecular', 'transport', 'nucleic', 'acid', 'metabol', 'score', '37', 'significantli', 'enrich', 'recurr', 'amplifi', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'gene-set', 'f2', 'network', \"'amino\", 'acid', 'metabol', 'post-', 'translat', 'modif', 'small', 'molecul', 'biochemistri', 'score', '29', 'fig', 'supplementari', 'tabl', '11', '12', 'sig-', 'nificantli', 'enrich', 'gene', 'overexpress', 'amplifi', 'gene-set', 'f3', 'fig', '1b', 'network', 'compos', 'gene', 'map', 'her2', 'amplicon', '17q12-', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '583', '123', 'q21', 'also', 'distinct', 'amplicon', 'includ', '8p12', '8q22-', 'q24', '11q13-q14', '17p11', '17q23', 'amplifi', 'multipl', 'combin', 'tumour', 'fig', 'supple-', 'mentari', 'tabl', '10', 'interestingli', 'ppm1d', 'neg', 'regul', 'p53', 'pathway', 'potenti', 'therapeut', 'target', 'subgroup', 'patient', 'her2', 'amplifi', 'breast', 'cancer', 'present', 'network', 'lumin', 'tumour', \"'erk/mapk\", 'signal', '0.00631', '0.00978', 'respect', \"'pi3k/akt\", 'signal', '0.01445', '0.01047', 'respect', 'canon', 'pathway', 'significantli', 'enrich', 'recur-', 'rentli', 'amplifi', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'gene-set', 'g2', 'gene', 'overexpress', 'amplifi', 'gene-set', 'g3', 'e.g.', 'pak1', 'prkar1a', 'rps6kb1', 'ppp2r5a', 'supplementari', 'tabl', 'fig', 'supplementari', 'fig', 'protein', 'kinas', 'two', 'canon-', 'ical', 'pathway', 'may', 'constitut', 'target', 'therapeut', 'inter-', 'vention', 'pi3k', 'and/or', 'mek', 'inhibitor', '13', '46', '47', 'lumin', 'breast', 'cancer', 'also', 'significantli', 'enrich', 'gene', 'network', \"'cell\", 'cycl', 'reproduct', 'system', 'develop', 'function', 'haematolog', 'system', 'develop', 'function', 'score', '40', \"'reproduct\", 'system', 'diseas', 'cellular', 'assembl', 'organis', 'cellular', 'movement', 'score', '33', 'supplementari', 'tabl', '11', '12', 'fig', 'supplementari', 'fig', 'note', 'ppm1d', 'part', 'cell', 'cycl', 'network', 'provid', 'addit', 'evid', 'previou', 'find', 'ppm1d', 'may', 'also', 'potenti', 'therapeut', 'target', 'subgroup', 'lumin', 'breast', 'cancer', 'harbour', 'amplif', 'gene', 'valid', 'acgh', 'microarray-bas', 'express', 'profil', 'analysi', 'two', 'tissu', 'microarray', 'contain', '48', 'tumour', 'studi', 'subject', 'cish', 'probe', 'ccnd1', 'egfr', 'ppm1d', 'her2', 'perfect', 'agreement', 'acgh', 'cish', 'result', 'egfr', 'ppm1d', 'her2', 'observ', 'supplementari', 'tabl', 'supplementari', 'fig', 'wherea', 'ccnd1', 'two', 'case', 'show', 'concord', 'result', 'unweight', 'kappa', 'score', '0.86297', '95', 'ci', '0.7099-1.0000', 'good', 'correl', 'observ', 'micro-', 'array', 'qrt-pcr-defin', 'express', 'level', 'ppm1d', 'pearson', \"'s\", 'valu', '0.81421', '1.8', '10-11', 'agreement', 'result', 'acgh', 'express', 'array', 'analysi', 'cish', 'qrt-pcr-base', 'assess', 'ppm1d', 'copi', 'number', 'mrna', 'express', 'also', 'fig', 'canon', 'pathway', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', '13', 'basal-lik', '13', 'her2', 'giii', 'tumour', 'defin', 'ingenu', 'pathway', 'analysi', 'softwar', 'canon-', 'ical', 'pathway', \"'oestrogen\", 'receptor', 'signal', '0.0018', 'basal-lik', 'ii', \"'oestrogen\", 'receptor', 'signal', '0.013', 'her2', 'iii', \"'nucleotid\", 'excis', 'repair', 'pathway', '0.0214', 'her2', 'giii', 'tumour', 'note', 'express', 'level', 'gene', 'significantli', 'correl', 'gene', 'copi', 'number', 'match', 'heatmap', 'acgh', 'data', 'basal-lik', 'ii', 'her2', 'iii', 'her2', 'tumour', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'bright', 'red', 'gene', 'amplif', '584', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'demonstr', 'ppm1d', 'overexpress', 'ampli-', 'fie', 'mann-whitney', 'test', '5.0', '1.4', '10-5', 'immunohistochem', 'analysi', 'cyclin', 'd1', 'use', 'semiquantit', 'score', 'system', 'show', 'good', 'correl', 'ccnd1', 'mrna', 'level', 'defin', 'microarray', 'analysi', 'spearman', \"'s\", 'correl', 'coeffici', '0.68943', '10-6', 'despit', 'impact', 'er', 'express', 'cyclin', 'd1', 'express', 'level', '28', 'cish', 'immunohis-', 'tochemistri', 'base', 'assess', 'ccnd1', 'copi', 'number', 'cyclin', 'd1', 'protein', 'express', 'respect', 'confirm', 'ccnd1', 'overexpress', 'amplifi', 'mann-', 'whitney', 'test', '62.0', '0.00694', 'agreement', 'acgh', 'express', 'array', 'analysi', 'result', 'provid', 'direct', 'evid', 'accuraci', 'acgh', 'expres-', 'sion', 'profil', 'method', 'employ', 'studi', 'discuss', 'actual', 'impact', 'gene', 'copi', 'number', 'chang', 'transcriptom', 'profil', 'breast', 'cancer', 'molecular', 'subtyp', 'matter', 'content', 'whilst', 'her2', 'tumour', 'definit', 'biolog', 'driven', 'amplif', 'overexpress', 'erbb2', 'impact', 'genom', 'altera-', 'tion', 'transcriptom', 'phenotyp', 'basal-lik', 'lumin', 'tumour', 'less', 'clear', 'show', 'fig', 'top', 'rank', 'network', '13', 'her2', 'tumour', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'recurr', 'amplifi', 'pearson', \"'s\", 'correl', 'gene', 'overexpress', 'amplifi', 'mann-whitney', 'test', 'defin', 'ingenu', 'pathway', 'analysi', 'top', 'rank', 'network', 'her2', 'tumour', \"'lipid\", 'metabol', 'molecular', 'transport', 'nucleic', 'acid', 'metabol', 'pearson', \"'s\", 'correl', 'score', '37', 'ii', \"'amino\", 'acid', 'metabol', 'post-transl', 'modif', 'small', 'molecul', 'biochemistri', 'mann-whitney', 'test', 'score', '29', 'green', 'downregul', 'red', 'upregul', 'match', 'heatmap', 'acgh', 'data', 'her2', 'tumour', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'red', 'gene', 'amplif', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '585', '123', 'distinct', 'gene', 'network', 'canon', 'pathway', 'enri-', 'ched', 'gene', 'whose', 'express', 'level', 'correl', 'copi', 'number', 'basal-lik', 'her2', 'lumin', 'tumour', 'provid-', 'ing', 'evid', 'differ', 'molecular', 'subtyp', 'may', 'determin', 'least', 'part', 'pat-', 'tern', 'genet', 'aberr', 'meta-analys', 'microarray', 'express', 'data', 'demonstr', 'main', 'separ', 'molecular', 'subtyp', 'driven', 'er', 'her2', 'prolifer', 'prolifer', 'major', 'determin', 'breast', 'cancer', 'clin-', 'ical', 'behaviour', '48', '49', 'analysi', 'demonstr', 'network', 'canon', 'pathway', 'directli', 'relat', 'er', 'signal', 'fig', 'her2', 'signal', 'fig', 'prolifer-', 'ation', 'fig', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'note', 'although', 'er', 'pathway', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'esr1', 'mrna', 'level', 'dna', 'copi', 'number', 'fail', 'show', 'statist', 'signific', 'correl', 'pearson', \"'s\", 'correl', 'adjust', '0.5667', 'studi', 'esr1', 'amplifi', 'singl', 'er-posit', 'case', '1.8', 'concur', 'previou', 'report', 'suggest', 'small', 'minor', 'er-posit', 'cancer', 'harbour', 'esr1', 'gene', 'amplif', 'er', 'express', 'significantli', 'associ', 'esr1', 'copi', 'number', '50-53', 'group', 'other', 'demonstr', 'tumour', 'aris', 'patient', 'harbour', 'brca1', 'germlin', 'mutat', 'predominantli', 'display', 'basal-lik', 'phenotyp', 'subgroup', 'sporad', 'basal-lik', 'breast', 'cancer', 'harbour', 'dysfunct', 'brca1', 'pathway', '54-57', 'given', 'rariti', 'brca1', 'somat', 'mutat', 'sever', 'mechan', 'brca1', 'pathway', 'dysfunct', 'sporad', 'breast', 'cancer', 'propos', '43', '56', '57', 'includ', 'brca1', 'gene', 'promot', 'methyl', 'overexpress', 'id4', 'neg', 'regu-', 'lator', 'brca1', '57', 'studi', 'provid', 'evid', 'brca1', 'pathway', 'may', 'dysfunct', 'sporad', 'fig', 'top', 'rank', 'network', 'canon', 'pathway', 'identifi', 'ingenu', 'pathway', 'analysi', '22', 'lumin', 'tumour', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'recurr', 'amplifi', 'pearson', \"'s\", 'correl', 'top', 'rank', 'network', 'canon', 'pathway', 'lumin', 'tumour', \"'cell\", 'cycl', 'reproduct', 'system', 'develop', 'function', 'haematolog', 'system', 'develop', 'function', 'score', '40', 'ii', 'canon', 'pathway', \"'erk/mapk\", 'signal-', 'ling', '0.0063', 'iii', 'canon', 'pathway', \"'pi3k/akt\", 'signal', '0.0145', 'green', 'downregul', 'red', 'upregul', 'match', 'heatmap', 'acgh', 'data', 'her2', 'tumour', 'green', 'copi', 'number', 'loss', 'black', 'copi', 'number', 'chang', 'dark', 'red', 'copi', 'number', 'gain', 'red', 'gene', 'amplif', '586', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'basal-lik', 'cancer', 'due', 'gene', 'copi', 'number', 'aberr', 'affect', 'gene', 'pertain', 'pathway', \"'brca1\", 'dna', 'damag', 'respons', 'pathway', 'enrich', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'differenti', 'express', 'molecular', 'subtyp', 'supplementari', 'fig', 'brca1', 'express', 'found', 'associ', 'copi', 'number', 'downregu-', 'late', 'basal-lik', 'tumour', 'suggest', 'role', 'homolo-', 'gou', 'recombin', 'dna', 'repair', 'defect', 'subgroup', 'sporad', 'breast', 'cancer', 'addit', 'provid', 'evid', 'suggest', 'overexpress', 'brca1', 'neg', 'regu-', 'lator', 'id4', 'basal-lik', 'cancer', 'driven', 'copi', 'number', 'gain', 'express', 'level', 'gene', 'correl', 'copi', 'number', 'whole', 'cohort', 'pearson', \"'s\", 'correla-', 'tion', 'adjust', '0.01669', 'supplementari', 'tabl', 'id4', 'express', 'significantli', 'higher', 'level', 'basal-', 'like', 'breast', 'cancer', 'supplementari', 'tabl', 'id4', 'copi', 'number', 'gain', 'significantli', 'frequent', 'basal-lik', 'non-basal-lik', 'tumour', '9/13', 'basal-lik', 'cancer', '10/35', 'non-basal-lik', 'cancer', 'fisher', \"'s\", 'exact', 'test', '0.0187', 'taken', 'togeth', 'result', 'support', 'content', 'brca1', 'pathway', 'dysfunct', 'sporad', 'basal-lik', 'breast', 'cancer', 'may', 'driven', 'genet', 'aber-', 'ration', 'given', 'report', 'sensit', 'tumour', 'brca1', 'loss', 'function', 'parp', 'inhibitor', 'find', 'provid', 'credenc', 'molecular', 'basi', 'potenti', 'efficaci', 'sporad', 'basal-lik', 'breast', 'cancer', '58', 'compar', 'list', 'copi', 'number', 'regul', 'gene', 'publish', \"'intrins\", 'gene', 'list', 'observ', '61/207', '29', 'intrins', 'gene', 'describ', 'perou', 'et', 'al', 'present', 'dataset', '166/365', '45.5', 'intrins', 'gene', 'sorli', 'et', 'al', '59', '435/929', '46.8', 'uniqu', 'iden-', 'tifier', \"'intrinsic/unc\", 'gene', 'list', 'hu', 'et', 'al', '24', '684/1,488', '46', 'intrins', 'gene', 'parker', 'et', 'al', '31', 'significantli', 'associ', 'gene', 'copi', 'number', 'sub-', 'stantial', 'enrich', 'publish', \"'intrins\", 'gene', 'list', 'gene', 'whose', 'express', 'correl', 'copi', 'number', 'emphasis', 'potenti', 'import', 'genet', 'alter', 'molecular', 'classif', 'breast', 'cancer', 'furthermor', 'suggest', 'differenti', 'express', \"'intrins\", 'gene', 'distinct', 'molecular', 'subtyp', 'may', 'stem', 'cell', 'origin', 'also', 'result', 'genet', 'instabl', 'subsequ', 'acquisit', 'stochast', 'genet', 'aberr', 'identifi', 'possibl', 'therapeut', 'target', 'use', 'integr', 'approach', 'distinguish', 'recurr', 'amplifi', 'gene', 'whose', 'express', 'associ', 'copi', 'number', 'also', 'gene', 'consist', 'overexpress', 'amplifi', 'genome-wid', 'unbias', 'approach', 'identi-', 'fie', 'gene', 'function', 'valid', 'potenti', 'therapeut', 'target', 'specif', 'breast', 'cancer', 'molecular', 'subtyp', 'e.g', 'her2', 'ppm1d', 'pak1', 'ppapdc1b', '--', 'supplementari', 'tabl', '10', '13', '44', 'canon', 'pathway', 'exploit', 'preclin', 'therapeut', 'model', 'breast', 'cancer', 'e.g', 'pi3k/akt', 'signal', 'lumin', 'cancer', '--', 'fig', 'supplementari', 'fig', '47', 'well', 'comprehens', 'list', 'gene', 'may', 'constitut', 'amplicon', 'driver', 'novel', 'therapeut', 'target', 'molecular', 'subtype-specif', 'analys', 'gene', 'consis-', 'tentli', 'significantli', 'overexpress', 'amplifi', 'high-', 'light', 'network', 'canon', 'pathway', 'enrich', 'potenti', 'therapeut', 'target', 'network', 'identifi', 'basal-lik', 'tumour', 'found', 'enrich', 'multipl', 'gene', 'involv', 'transcript', 'regul', 'suggest', 'preclin', 'studi', 'test', 'transcript', 'regu-', 'lator', 'potenti', 'target', 'may', 'warrant', 'p21-', 'activ', 'kinas', 'pak4', 'play', 'role', 'integrin-', 'mediat', 'breast', 'cancer', 'cell', 'motil', '60', 'significantli', 'overexpress', 'amplifi', 'basal-lik', 'cancer', 'novel', 'strategi', 'target', 'pak4', 'famili', 'member', 'cancer', 'investig', '45', 'ingenu', 'pathway', 'analysi', 'her2', 'cancer', 'substanti', 'previou', 'find', 'use', 'ppm1d-target', 'therapi', 'subgroup', '35', 'reveal', 'pi3k', 'erk/mapk', 'canon', 'pathway', 'significantli', 'enrich', 'gene', 'amplifi', 'overexpress', 'lumin', 'tumour', 'e.g', 'pak1', 'ppp2r5a', 'prkar1a', 'interestingli', 'pak1', 'overexpress', 'lumin', 'breast', 'cancer', 'cell', 'line', 'recent', 'shown', 'significantli', 'sensit', 'mek', 'inhibit', '13', 'moreov', 'pi3k', 'catalyt', 'subunit', 'inhibitor', 'bez235', 'report', 'caus', 'select', 'cell', 'growth', 'inhibit', 'er-posit', 'breast', 'cancer', 'cell', 'line', '47', 'may', 'constitut', 'interest', 'therapeut', 'approach', 'subgroup', 'patient', 'lumin', 'breast', 'cancer', 'har-', 'bour', 'amplif', 'gene', 'conclus', 'high', 'resolut', 'integr', 'genom', 'analysi', 'basal-lik', 'her2', 'lumin', 'giii', 'breast', 'can-', 'cer', 'use', 'microdissect', 'tissu', 'identifi', 'pathway', 'network', 'specif', 'tumour', 'type', 'regul', 'chang', 'dna', 'level', 'identifi', 'key', 'gene', 'govern', 'distinct', 'pathway', 'result', 'provid', 'strong', 'circumstanti', 'evid', 'suggest', 'divers', 'breast', 'cancer', 'molecular', 'subtyp', 'may', 'stem', 'least', 'part', 'distinct', 'pattern', 'genet', 'aberr', 'found', 'cancer', 'furthermor', 'identifi', 'canon', 'pathway', 'enrich', 'gene', 'recurr', 'ampli-', 'fie', 'overexpress', 'amplifi', 'basal-lik', 'her2', 'lumin', 'cancer', 'function', 'genom', 'studi', 'test', 'biolog', 'import', 'pathway', 'thera-', 'peutic', 'implic', 'inhibit', 'pathway', 'appropri', 'preclin', 'model', 'warrant', 'acknowledg', 'studi', 'fund', 'part', 'breakthrough', 'breast', 'cancer', 'b.', 'weigelt', 'support', 'cancer', 'research', 'uk', 'fellowship', 'palacio', 'retic', 'rd06/0020/0013', 'pi', '08/', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '587', '123', '6080971', 'instituto', 'de', 'salud', 'carlo', 'iii', 'p07-cvi-03100', 'consejeria', 'de', 'innovacion', 'junta', 'de', 'andalucia', 'refer', '1.', 'natrajan', 'lambro', 'mb', 'rodriguez-pinilla', 'sm', 'et', 'al', '2009', 'tile', 'path', 'genom', 'profil', 'grade', 'invas', 'ductal', 'breast', 'cancer', 'clin', 'cancer', 're', '15:2711-2722', '2.', 'bergamaschi', 'kim', 'yh', 'wang', 'et', 'al', '2006', 'distinct', 'pattern', 'dna', 'copi', 'number', 'alter', 'associ', 'differ', 'clinicopatholog', 'featur', 'gene-express', 'subtyp', 'breast', 'cancer', 'gene', 'chromosom', 'cancer', '45:1033-1040', '3.', 'chin', 'devri', 'fridlyand', 'et', 'al', '2006', 'genom', 'tran-', 'scription', 'aberr', 'link', 'breast', 'cancer', 'pathophysiolog', 'cancer', 'cell', '10:529-541', '4.', 'chin', 'sf', 'teschendorff', 'ae', 'marioni', 'jc', 'et', 'al', '2007', 'high-reso-', 'lution', 'acgh', 'express', 'profil', 'identifi', 'novel', 'genom', 'subtyp', 'er', 'neg', 'breast', 'cancer', 'genom', 'biol', 'r215', '5.', 'hu', 'stern', 'hm', 'ge', 'et', 'al', '2009', 'genet', 'alter', 'oncogen', 'pathway', 'associ', 'breast', 'cancer', 'subtyp', 'mol', 'cancer', 're', '7:511-522', '6.', 'mackay', 'tamber', 'fenwick', 'et', 'al', '2009', 'high-resolut', 'integr', 'analysi', 'genet', 'express', 'profil', 'breast', 'cancer', 'cell', 'line', 'breast', 'cancer', 're', 'treat', 'doi:10.1007/s10549-', '008-0296-7', '7.', 'neve', 'rm', 'chin', 'fridlyand', 'et', 'al', '2006', 'collect', 'breast', 'cancer', 'cell', 'line', 'studi', 'function', 'distinct', 'cancer', 'subtyp', 'cancer', 'cell', '10:515-527', '8.', 'perou', 'cm', 'sorli', 'eisen', 'mb', 'et', 'al', '2000', 'molecular', 'portrait', 'human', 'breast', 'tumour', 'natur', '406:747-752', '9.', 'sorli', 'perou', 'cm', 'tibshirani', 'et', 'al', '2001', 'gene', 'express', 'pattern', 'breast', 'carcinoma', 'distinguish', 'tumor', 'subclass', 'clinic', 'implic', 'proc', 'natl', 'acad', 'sci', 'usa', '98:10869-10874', '10.', 'hick', 'krasnitz', 'lakshmi', 'et', 'al', '2006', 'novel', 'pattern', 'genom', 'rearrang', 'associ', 'surviv', 'breast', 'cancer', 'genom', 're', '16:1465-1479', '11.', 'adelaid', 'finetti', 'bekhouch', 'et', 'al', '2007', 'integr', 'pro-', 'file', 'basal', 'lumin', 'breast', 'cancer', 'cancer', 're', '67:11565-', '11575', '12.', 'natrajan', 'lambro', 'mb', 'geyer', 'fc', 'et', 'al', '2009', 'loss', '16q', 'high', 'grade', 'breast', 'cancer', 'associ', 'estrogen', 'receptor', 'statu', 'evid', 'progress', 'tumor', 'lumin', 'phe-', 'notyp', 'gene', 'chromosom', 'cancer', '48:351-365', '13.', 'heiser', 'lm', 'wang', 'nj', 'talcott', 'cl', 'et', 'al', '2009', 'integr', 'analysi', 'breast', 'cancer', 'cell', 'line', 'reveal', 'uniqu', 'signal', 'pathway', 'genom', 'biol', '10', 'r31', '14.', 'marchio', 'iravani', 'natrajan', 'et', 'al', '2008', 'genom', 'immunophenotyp', 'character', 'pure', 'micropapillari', 'carcinoma', 'breast', 'pathol', '215:398-410', '15.', 'simpson', 'pt', 'reis-filho', 'js', 'lambro', 'mb', 'et', 'al', '2008', 'molecular', 'profil', 'pleomorph', 'lobular', 'carcinoma', 'breast', 'evid', 'common', 'molecular', 'genet', 'pathway', 'classic', 'lobular', 'carcinoma', 'pathol', '215:231-244', '16.', 'weigelt', 'horl', 'hm', 'kreik', 'et', 'al', '2008', 'refin', 'breast', 'cancer', 'classif', 'molecular', 'character', 'his-', 'tolog', 'special', 'type', 'pathol', '216:141-150', '17.', 'wolff', 'ac', 'hammond', 'schwartz', 'jn', 'et', 'al', '2007', 'american', 'societi', 'clinic', 'oncology/colleg', 'american', 'pathologist', 'guidelin', 'recommend', 'human', 'epiderm', 'growth', 'factor', 'receptor', 'test', 'breast', 'cancer', 'clin', 'oncol', '25:118-145', '18.', 'nielsen', 'hsu', 'fd', 'jensen', 'et', 'al', '2004', 'immunohistochem', 'clinic', 'character', 'basal-lik', 'subtyp', 'invas', 'breast', 'carcinoma', 'clin', 'cancer', 're', '10:5367-5374', '19.', 'marchio', 'natrajan', 'shiu', 'kk', 'et', 'al', '2008', 'genom', 'profil', 'her2-amplifi', 'breast', 'cancer', 'influenc', 'er', 'statu', 'pathol', '216:399-407', '20.', 'huang', 'gusnanto', \"o'sullivan\", 'et', 'al', '2007', 'robust', 'smooth', 'segment', 'approach', 'array', 'cgh', 'data', 'analysi', 'bioinfor-', 'matic', '23:2463-2469', '21.', 'arriola', 'marchio', 'tan', 'ds', 'et', 'al', '2008', 'genom', 'analysi', 'her2/top2a', 'amplicon', 'breast', 'cancer', 'breast', 'cancer', 'cell', 'line', 'lab', 'invest', '88:491-503', '22.', 'reis-filho', 'js', 'simpson', 'pt', 'jone', 'et', 'al', '2005', 'pleomorph', 'lobular', 'carcinoma', 'breast', 'role', 'comprehens', 'molecular', 'patholog', 'character', 'entiti', 'pathol', '207:1-13', '23.', 'benjamini', 'hochberg', '1995', 'control', 'fals', 'discoveri', 'rate', 'practic', 'power', 'approach', 'multipl', 'test', 'roy', 'stat', 'soc', '57:289-300', '24.', 'hu', 'fan', 'oh', 'ds', 'et', 'al', '2006', 'molecular', 'portrait', 'breast', 'tumor', 'conserv', 'across', 'microarray', 'platform', 'bmc', 'genom-', 'ic', '7:96', '25.', 'eisen', 'mb', 'spellman', 'pt', 'brown', 'po', 'et', 'al', '1998', 'cluster', 'analysi', 'display', 'genome-wid', 'express', 'pattern', 'proc', 'natl', 'acad', 'sci', 'usa', '95:14863-14868', '26.', 'tusher', 'vg', 'tibshirani', 'chu', '2001', 'signific', 'analysi', 'microarray', 'appli', 'ioniz', 'radiat', 'respons', 'proc', 'natl', 'acad', 'sci', 'usa', '98:5116-5121', '27.', 'weigelt', 'kreik', 'reis-filho', 'js', '2008', 'metaplast', 'breast', 'car-', 'cinoma', 'basal-lik', 'breast', 'cancer', 'genom', 'profil', 'analysi', 'breast', 'cancer', 're', 'treat', 'doi:10.1007/s10549-008-0197-9', '28.', 'reis-filho', 'js', 'savag', 'lambro', 'mb', 'et', 'al', '2006', 'cyclin', 'd1', 'protein', 'overexpress', 'ccnd1', 'amplif', 'breast', 'car-', 'cinoma', 'immunohistochem', 'chromogen', 'situ', 'hybridis', 'analysi', 'mod', 'pathol', '19:999-1009', '29.', 'reis-filho', 'js', 'pinheiro', 'lambro', 'mb', 'et', 'al', '2006', 'egfr', 'amplif', 'lack', 'activ', 'mutat', 'metaplast', 'breast', 'carcinoma', 'pathol', '209:445-453', '30.', 'tan', 'ds', 'lambro', 'mb', 'rayter', 'et', 'al', '2009', 'ppm1d', 'potenti', 'therapeut', 'target', 'ovarian', 'clear', 'cell', 'carcinoma', 'clin', 'cancer', 're', '15:2269-2280', '31.', 'parker', 'js', 'mullin', 'cheang', 'mc', 'et', 'al', '2009', 'supervis', 'risk', 'predictor', 'breast', 'cancer', 'base', 'intrins', 'subtyp', 'clin', 'oncol', '27:1160-1167', '32.', 'cheng', 'kw', 'lahad', 'jp', 'kuo', 'wl', 'et', 'al', '2004', 'rab25', 'small', 'gtpase', 'determin', 'aggress', 'ovarian', 'breast', 'cancer', 'nat', 'med', '10:1251-1256', '33.', 'fidd', 'rj', 'campbel', 'dh', 'jane', 'pw', 'et', 'al', '1998', 'analysi', 'grb7', 'recruit', 'heregulin-activ', 'erbb', 'receptor', 'reveal', 'novel', 'target', 'select', 'erbb3', 'biol', 'chem', '273:7717-7724', '34.', 'hata', 'furukawa', 'sunamura', 'et', 'al', '2005', 'rna', 'interfer', 'target', 'aurora', 'kinas', 'suppress', 'tumor', 'growth', 'enhanc', 'taxan', 'chemosensit', 'human', 'pancreat', 'cancer', 'cell', 'cancer', 're', '65:2899-2905', '35.', 'rayter', 'elliott', 'traver', 'et', 'al', '2008', 'chemic', 'inhibitor', 'ppm1d', 'select', 'kill', 'cell', 'overexpress', 'ppm1d', 'oncogen', '27:1036-1044', '36.', 'reis-filho', 'js', 'simpson', 'pt', 'turner', 'nc', 'et', 'al', '2006', 'fgfr1', 'emerg', 'potenti', 'therapeut', 'target', 'lobular', 'breast', 'car-', 'cinoma', 'clin', 'cancer', 're', '12:6652-6662', '37.', 'kao', 'pollack', 'jr', '2006', 'rna', 'interference-bas', 'function', 'dissect', '17q12', 'amplicon', 'breast', 'cancer', 'reveal', 'con-', 'tribut', 'coamplifi', 'gene', 'gene', 'chromosom', 'cancer', '45:761-769', '38.', 'jiang', 'zeng', 'yang', 'et', 'al', '2008', 'oral', 'cancer', 'overexpress', 'oraov1', 'regul', 'cell', 'growth', 'tumor', 'angio-', 'genesi', 'oral', 'squamou', 'cell', 'carcinoma', 'int', 'cancer', '123:1779-', '1786', '39.', 'orsetti', 'nugoli', 'cervera', 'et', 'al', '2004', 'genom', 'express', 'profil', 'chromosom', '17', 'breast', 'cancer', 'reveal', '588', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '123', 'complex', 'pattern', 'alter', 'novel', 'candid', 'gene', 'can-', 'cer', 're', '64:6453-6460', '40.', 'guan', 'kuo', 'wl', 'stilwel', 'jl', 'et', 'al', '2007', 'amplif', 'pvt1', 'contribut', 'pathophysiolog', 'ovarian', 'breast', 'cancer', 'clin', 'cancer', 're', '13:5745-5755', '41.', 'subhawong', 'ap', 'subhawong', 'nassar', 'et', 'al', '2009', 'basal-', 'like', 'breast', 'carcinoma', 'demonstr', 'rb-/p16', 'immu-', 'nophenotyp', 'hpv-relat', 'poorli', 'differenti', 'squamou', 'cell', 'carcinoma', 'resembl', 'morpholog', 'surg', 'pathol', '33:163-175', '42.', 'reis-filho', 'js', 'tutt', '2008', 'tripl', 'neg', 'tumour', 'critic', 'review', 'histopatholog', '52:108-118', '43.', 'rakha', 'ea', 'reis-filho', 'js', 'elli', 'io', '2008', 'basal-lik', 'breast', 'can-', 'cer', 'critic', 'review', 'clin', 'oncol', '26:2568-2581', '44.', 'bernard-pierrot', 'gruel', 'stranski', 'et', 'al', '2008', 'character-', 'izat', 'recurr', '8p11-12', 'amplicon', 'identifi', 'ppapdc1b', 'phosphatas', 'protein', 'new', 'therapeut', 'target', 'breast', 'can-', 'cer', 'cancer', 're', '68:7165-7175', '45.', 'eswaran', 'soundararajan', 'knapp', '2009', 'target', 'group', 'ii', 'pak', 'cancer', 'metastasi', 'cancer', 'metastasi', 'rev', '28:209-217', '46.', 'ihl', 'nt', 'powi', '2009', 'take', 'pik', 'phosphatidylinositol', '3-', 'kinas', 'inhibitor', 'race', 'clinic', 'toward', 'cancer', 'therapi', 'mol', 'cancer', 'ther', '8:1-9', '47.', 'crowder', 'rj', 'phommali', 'tao', 'et', 'al', '2009', 'pik3ca', 'pik3cb', 'inhibit', 'produc', 'synthet', 'lethal', 'combin', 'estrogen', 'depriv', 'estrogen', 'receptor-posit', 'breast', 'cancer', 'cancer', 're', '69:3955-3962', '48.', 'desmedt', 'haibe-kain', 'wirapati', 'et', 'al', '2008', 'biolog', 'process', 'associ', 'breast', 'cancer', 'clinic', 'outcom', 'depend', 'molecular', 'subtyp', 'clin', 'cancer', 're', '14:5158-5165', '49.', 'wirapati', 'sotiri', 'kunkel', 'et', 'al', '2008', 'meta-analysi', 'gene', 'express', 'profil', 'breast', 'cancer', 'toward', 'unifi', 'understand', 'breast', 'cancer', 'subtyp', 'prognosi', 'signatur', 'breast', 'cancer', 're', '10', 'r65', '50.', 'reis-filho', 'js', 'druri', 'lambro', 'mb', 'et', 'al', '2008', 'esr1', 'gene', 'amplif', 'breast', 'cancer', 'common', 'phenomenon', 'nat', 'genet', '40:809-810', 'author', 'repli', '10-2', '51.', 'brown', 'la', 'hoog', 'chin', 'sf', 'et', 'al', '2008', 'esr1', 'gene', 'amplifica-', 'tion', 'breast', 'cancer', 'common', 'phenomenon', 'nat', 'genet', '40:806-807', 'author', 'repli', '10-2', '52.', 'horl', 'hm', 'bergamaschi', 'nordgard', 'sh', 'et', 'al', '2008', 'esr1', 'gene', 'amplif', 'breast', 'cancer', 'common', 'phenomenon', 'nat', 'genet', '40:807-808', 'author', 'repli', '10-2', '53.', 'vincent-salomon', 'gruel', 'lucchesi', 'et', 'al', '2007', 'identifi-', 'cation', 'typic', 'medullari', 'breast', 'carcinoma', 'genom', 'sub-', 'group', 'basal-lik', 'carcinoma', 'heterogen', 'new', 'molecular', 'entiti', 'breast', 'cancer', 're', 'r24', '54.', 'lakhani', 'sr', 'reis-filho', 'js', 'fulford', 'et', 'al', '2005', 'predict', 'brca1', 'statu', 'patient', 'breast', 'cancer', 'use', 'estrogen', 'receptor', 'basal', 'phenotyp', 'clin', 'cancer', 're', '11:5175-5180', '55.', 'foulk', 'wd', 'stefansson', 'im', 'chappui', 'po', 'et', 'al', '2003', 'germlin', 'brca1', 'mutat', 'basal', 'epitheli', 'phenotyp', 'breast', 'cancer', 'natl', 'cancer', 'inst', '95:1482-1485', '56.', 'turner', 'nc', 'reis-filho', 'js', '2006', 'basal-lik', 'breast', 'cancer', 'brca1', 'phenotyp', 'oncogen', '25:5846-5853', '57.', 'turner', 'nc', 'reis-filho', 'js', 'russel', 'et', 'al', '2007', 'brca1', 'dysfunct', 'sporad', 'basal-lik', 'breast', 'cancer', 'oncogen', '26', '2126-2132', '58.', 'ashworth', '2008', 'synthet', 'lethal', 'therapeut', 'approach', 'poli', 'adp', 'ribos', 'polymeras', 'inhibitor', 'treatment', 'cancer', 'defici', 'dna', 'double-strand', 'break', 'repair', 'clin', 'oncol', '26:3785-3790', '59.', 'sorli', 'tibshirani', 'parker', 'et', 'al', '2003', 'repeat', 'observ', 'breast', 'tumor', 'subtyp', 'independ', 'gene', 'express', 'data', 'set', 'proc', 'natl', 'acad', 'sci', 'usa', '100:8418-8423', '60.', 'zhang', 'li', 'viklund', 'ek', 'et', 'al', '2002', 'p21-activ', 'kinas', 'interact', 'integrin', 'alpha', 'beta', 'regul', 'alpha', 'beta', '5-', 'mediat', 'cell', 'migrat', 'cell', 'biol', '158:1287-1297', 'breast', 'cancer', 're', 'treat', '2010', '121:575-589', '589', '123']\n"
     ]
    }
   ],
   "source": [
    "print(texts[2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Train\n",
    "\n",
    "References:\n",
    "\n",
    "https://radimrehurek.com/gensim/tut1.html\n",
    "\n",
    "https://machinelearningmastery.com/develop-word-embeddings-python-gensim/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: You are using pip version 19.1.1, however version 19.2.3 is available.\r\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!pip3 install --user --upgrade --quiet gensim\n",
    "import gensim\n",
    "import gensim.models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = gensim.models.Word2Vec(texts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Word2Vec(vocab=6968, size=100, alpha=0.025)\n",
      "[-2.4755862e-01 -7.8153157e-01 -8.0560011e-01 -4.8409903e-01\n",
      "  3.3647668e-01  2.1645081e-01 -3.5262248e-01 -9.0181446e-01\n",
      " -3.0526063e-01 -6.3625085e-01  2.7524340e-01 -2.7332041e-02\n",
      "  2.6516172e-01 -1.0393782e+00  8.1709069e-01 -3.4102071e-02\n",
      " -1.1518853e+00  4.8040858e-01  6.9009745e-01 -9.9844307e-01\n",
      "  1.4020872e+00 -8.2177454e-01  3.6775216e-01  9.1776472e-01\n",
      "  1.8031593e-02 -8.1322372e-02  1.0435814e+00 -1.0838606e+00\n",
      " -3.1519359e-01  1.1972399e+00 -1.1905935e+00 -3.7958217e-01\n",
      "  1.4032669e+00 -1.2239828e+00 -6.3940579e-01 -5.2143693e-01\n",
      "  1.1923801e-01 -1.7005415e+00  1.8628597e-01  4.1503182e-01\n",
      " -8.2240582e-01  2.2537463e+00  1.1062886e+00  9.3109196e-01\n",
      "  1.1846621e+00  7.1248150e-01  3.6509210e-01 -4.0508384e-01\n",
      " -6.2892437e-01 -1.0977871e+00  6.3394874e-01 -2.2931777e-01\n",
      " -5.3467047e-01 -5.2831000e-01  1.3826777e-01  4.5151806e-01\n",
      "  7.9939818e-01 -3.3114481e-01  1.7573163e-01 -8.7270774e-02\n",
      "  4.5848361e-01 -5.2088195e-01  4.6835256e-01 -2.8171641e-01\n",
      " -2.3621620e-01  5.8776456e-01 -3.2315168e-01  9.6579432e-01\n",
      "  9.0720350e-01  4.1582045e-01 -7.7664062e-02  4.8509982e-01\n",
      " -7.7719772e-01 -1.8562576e-01 -2.7694589e-01  6.2182289e-01\n",
      "  1.5218930e-01  4.3937829e-01 -2.2978990e-01  4.2170042e-01\n",
      "  4.4733027e-01 -1.8678181e-01  7.1311116e-01  1.3124338e+00\n",
      " -4.6814916e-01 -6.1381957e-03 -2.1534414e+00 -1.6971055e+00\n",
      "  2.8098735e-03 -1.2137053e+00  1.8610638e-01  2.0975596e-03\n",
      " -1.2259905e+00 -8.7314747e-02 -4.4723931e-01 -3.8411310e-01\n",
      " -1.4490870e-01 -5.7762957e-01 -8.7783337e-02 -6.2653828e-01]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: DeprecationWarning: Call to deprecated `__getitem__` (Method will be removed in 4.0.0, use self.wv.__getitem__() instead).\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "print(model)\n",
    "print(model['brca1'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Visualize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sklearn.decomposition\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `__getitem__` (Method will be removed in 4.0.0, use self.wv.__getitem__() instead).\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "X = model[model.wv.vocab]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: DeprecationWarning: Call to deprecated `__getitem__` (Method will be removed in 4.0.0, use self.wv.__getitem__() instead).\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "# fit a 2d PCA model to the vectors\n",
    "X = model[model.wv.vocab]\n",
    "pca = sklearn.decomposition.PCA(n_components=2)\n",
    "result = pca.fit_transform(X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA2kAAANOCAYAAACLIUQoAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjAsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+17YcXAAAgAElEQVR4nOzdfbBeZX3v/88i2YEIScNTqiaU4NRfCOQ5G0GTWpXYpLVFHifVYE9Lq7+D1WlLhxKgf3BmnDNUwNJYGIpiDzDh/IQeEo7IMT+ZBAVspDsPNIKkkNOkuFHYURITSSAJ6/wB7iMSBc2dvb9kv14zmcm97nVf13Vn9h+8Wetau2nbNgAAANRwyGAvAAAAgP9LpAEAABQi0gAAAAoRaQAAAIWINAAAgEKGD8akxxxzTDthwoTBmBoAAGDQrV69ekvbtsfu671BibQJEyakp6dnMKYGAAAYdE3TbP5Z77ndEQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYi0Yvbs2TPYSwAAAAaRSNuHW265JVOnTs20adPykY98JF/60pdy6qmnZsaMGZk7d26efvrpJMkVV1yRCy64IO95z3vytre9LYsXL/6ZYyRJX19fzjnnnJxyyik55ZRT8uCDD/aP85GPfCSzZ8/uPxcAABiahg/2Aqp55JFH8qlPfSrf+MY3cswxx+QHP/hBmqbJqlWr0jRNPv/5z+fTn/50rrnmmiTJY489lpUrV2b79u2ZOHFiLrzwwvzbv/3bq8ZIkj/7sz/LX/zFX2TOnDn5j//4j8ybNy/f/va3kySPPvpoHnjggYwcOXLQvjsAADD4RFqSZWt7c9XyDXlq6840j34lM39jfo455pgkyVFHHZX169dnwYIF+e53v5sXXnghJ5xwQv9nP/CBD+TQQw/NoYcemrFjx+bpp5/OihUrct55571ijCS599578+ijj/Z/9oc//GF27NiRJDnjjDMEGgAA4HbHZWt7c+md69O7dWfaJFt37s59G57JsrW9/ed88pOfzCc+8YmsX78+//AP/5Bdu3b1v3fooYf2/33YsGE/d0/Ziy++mFWrVmXdunVZt25dent7c8QRRyRJDj/88M5/OQAA4A1nyEfaVcs3ZOfuvf2vD/u1qdn26P35r3c+lCT5wQ9+kG3btmXcuHFJkptvvvk1x3zf+96XO+64I9///vf7x0iS3/qt38pnP/vZ/vPWrVvXse8BAAAcHIZ8pD21decrXo849vj8yjsXZN0Nf55p06bloosuyhVXXJHzzjsvs2bN6r+F8ec5+eSTc/nll+c3f/M3+8dIksWLF6enpydTp07NSSedlBtuuOGAfCcAAOCNq2nbdsAn7e7ubnt6egZ83n2ZfeWK9P5UqCXJuDEj8+Ci9w3CigAAgINd0zSr27bt3td7Q/5K2sXzJmZk17BXHBvZNSwXz5s4SCsCAACGsiH/dMczZ7y01+zHT3d865iRuXjexP7jAAAAA2nIR1ryUqiJMgAAoIIhf7sjAABAJSINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKGS/I61pmsOapnmoaZqHm6Z5pGma/9KJhQEAAAxFwzswxvNJ3te27Y6mabqSPNA0zf9q23ZVB8YGAAAYUvY70tq2bZPsePll18t/2v0dFwAAYCjqyJ60pmmGNU2zLskzSb7atu0393HOx5qm6Wmapqevr68T0wIAABx0OhJpbdvubdt2epLxSd7RNM3kfZxzY9u23W3bdh977LGdmBYAAOCg09GnO7ZtuzXJyiTzOzkuAADAUNGJpzse2zTNmJf/PjLJ+5M8tr/jAgAADEWdeLrjW5Lc3DTNsLwUfbe3bXt3B8YFAAAYcjrxdMd/TTKjA2sBAAAY8jq6Jw0AAID9I9IAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUst+R1jTNcU3TrGya5tGmaR5pmubPOrEwAACAoWh4B8bYk+Qv27Zd0zTNqCSrm6b5atu2j3ZgbAAAgCFlv6+ktW373bZt17z89+1Jvp1k3P6OCwAAMBR1dE9a0zQTksxI8s19vPexpml6mqbp6evr6+S0AAAAB42ORVrTNEck+R9J/rxt2x/+9Ptt297Ytm1327bdxx57bKemBQAAOKh0JNKapunKS4G2pG3bOzsxJgAAwFDUiac7NkluSvLttm0/s/9LAgAAGLo6cSVtdpKPJHlf0zTrXv7zOx0YFwAAYMjZ70fwt237QJKmA2sBAAAY8jr6dEcAAAD2j0gDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQSEcirWmaLzRN80zTNN/qxHgAAABDVaeupP23JPM7NBYAAMCQ1ZFIa9v260l+0ImxAAAAhrIB25PWNM3HmqbpaZqmp6+vb6CmBQAAeEMZsEhr2/bGtm2727btPvbYYwdqWgAAgDcUT3cEAAAoRKQBAAAU0qlH8P/3JP+cZGLTNN9pmuaPOzEuAADAUDO8E4O0bfuhTowDAAAw1LndEQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAA5amzZtyuTJkzs65pNPPpn3vve9Oemkk3LyySfn7/7u7zo6/vCOjgYAAPAGs3fv3gwbNux1nz98+PBcc801mTlzZrZv355Zs2bl/e9/f0466aSOrMeVNAAA4KC2Z8+eLFy4MJMmTcq5556b5557LhMmTMgll1ySmTNn5o477sgTTzyRuXPnZtq0aZk5c2Y2btyYHTt25PTTT8/MmTMzZcqU3HXXXUmSt7zlLZk5c2aSZNSoUZk0aVJ6e3s7tl5X0gAAgIPahg0bctNNN2X27Nm54IILcv311ydJjj766KxZsyZJcuqpp2bRokU566yzsmvXrrz44osZMWJEli5dmtGjR2fLli057bTTcsYZZ6Rpmv6xN23alLVr1+bUU0/t2HpdSQMAAA5qxx13XGbPnp0kOf/88/PAAw8kSRYsWJAk2b59e3p7e3PWWWclSQ477LC86U1vStu2ueyyyzJ16tTMnTs3vb29efrpp/vH3bFjR84555xce+21GT16dMfW60oaAABw0Fm2tjdXLd+QzZs3pW/781m2tjdnzhiXJP1Xwg4//PCfO8aSJUvS19eX1atXp6urKxMmTMiuXbuSJLt3784555yThQsX5uyzz+7o2l1JAwAADirL1vbm0jvXp3frziTJC1ufyZ8vvj3L1vbmtttuy5w5c15x/qhRozJ+/PgsW7YsSfL888/nueeey7Zt2zJ27Nh0dXVl5cqV2bx5c5Kkbdv88R//cSZNmpSLLrqo4+sXaQAAwEHlquUbsnP33v7Xw48any0P/c98eN678uyzz+bCCy981WduvfXWLF68OFOnTs273vWufO9738vChQvT09OTKVOm5JZbbsmJJ56YJHnwwQdz6623ZsWKFZk+fXqmT5+ee+65p2Prb9q27dhgr1d3d3fb09Mz4PMCAAAHvxMWfTn7qpwmyb9f+YGBXs4+NU2zum3b7n2950oaAABwUHnrmJG/0PFqRBoAAHBQuXjexIzseuUvpx7ZNSwXz5s4SCv6xXi6IwAAcFD58VMcr1q+IU9t3Zm3jhmZi+dN7D9enUgDAAAOOmfOGPeGibKf5nZHAACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINACADtu6dWuuv/761zzvPe95TyZOnJjp06dn+vTpeeaZZ5IkN9xwQ6ZMmZLp06dnzpw5efTRRw/0koFCRBoAQIe93khLkiVLlmTdunVZt25dxo4dmyT58Ic/nPXr12fdunX5q7/6q1x00UUHcrlAMcMHewEAAAebRYsWZePGjZk+fXq6uroycuTIjBo1Kk888UTe+9735vrrr88hh/zs/1c+evTo/r//6Ec/StM0A7FsoAiRBgDQYVdeeWW+9a1vZd26dbnvvvsyf/78PProozn++OMzf/783HnnnTn33HOTJH/0R3+UYcOG5Zxzzslf//Vf9wfZddddl8985jN54YUXsmLFisH8OsAAc7sjAMAB9o53vCNve9vbMmzYsHzoQx/KAw88kOSlWx3Xr1+f+++/P/fff39uvfXW/s/86Z/+aTZu3Ji/+Zu/yac+9anBWjowCEQaAECHLFvbm9lXrsicv1mR/73lR1m2tjdJXnW74o9fjxs3LkkyatSofPjDH85DDz30qjF///d/P8uWLTvAKwcqEWkAAB2wbG1vLr1zfXq37kwzYmRe2PmjXHrn+jzweF8eeuih/Pu//3tefPHFfPGLX8ycOXOyZ8+ebNmyJUmye/fu3H333Zk8eXKS5PHHH+8f98tf/nLe/va3D8p3AgaHPWkAAB1w1fIN2bl7b5Jk2MjROXTcSdl4w/+bKw8dmVNOOSWf+MQn+h8cctZZZ2Xnzp2ZN29edu/enb1792bu3Ln56Ec/miT5+7//+9x7773p6urKkUcemZtvvnkwvxowwJq2bQd80u7u7ranp2fA5wUAOFBOWPTl7Ou/qnb9x7+m+4cP5u677x7wNQF1NU2zum3b7n2953ZHAIAOeOuYkfs8fswRhw7wSoA3OpEGANABF8+bmJFdw15xbGTXsHzqwgWuogG/EHvSAAA64MwZLz2p8arlG/LU1p1565iRuXjexP7jAK+XSAMA6JAzZ4wTZcB+c7sjAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABTSkUhrmmZ+0zQbmqZ5ommaRZ0YEwAAYCja70hrmmZYkuuS/HaSk5J8qGmak/Z3XAAAgKGoE1fS3pHkibZt/3fbti8k+f+SfLAD4wIAAAw5nYi0cUme/InX33n52Cs0TfOxpml6mqbp6evr68C0AAAAB58Be3BI27Y3tm3b3bZt97HHHjtQ0wIAALyhdCLSepMc9xOvx798DAAAgF9QJyLtX5K8vWmaE5qmGZHk95P8zw6MCwAAMOQM398B2rbd0zTNJ5IsTzIsyRfatn1kv1cGAAAwBO13pCVJ27b3JLmnE2MBAAAMZQP24BAAAABem0gDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAAygTZs2ZfLkyR0f94ILLsjYsWMPyNgMLJEGAADF7N279xf+zB/+4R/mK1/5ygFYDQNNpAEAwADbs2dPFi5cmEmTJuXcc8/Nc889lwkTJuSSSy7JzJkzc8cdd+SJJ57I3LlzM23atMycOTMbN27Mjh07cvrpp2fmzJmZMmVK7rrrrv4x3/3ud+eoo44axG9Fpwwf7AUAAMBQs2HDhtx0002ZPXt2Lrjgglx//fVJkqOPPjpr1qxJkpx66qlZtGhRzjrrrOzatSsvvvhiRowYkaVLl2b06NHZsmVLTjvttJxxxhlpmmYwvw4dJtIAAGAALFvbm6uWb8jmzZsy4lfGpu9NE5Ik559/fhYvXpwkWbBgQZJk+/bt6e3tzVlnnZUkOeyww5Iku3fvzmWXXZavf/3rOeSQQ9Lb25unn346b37zmwf+C3HAiDQAADjAlq3tzaV3rs/O3S/tNdvbtrn0zvVJktFJ/5Wwww8//OeOs2TJkvT19WX16tXp6urKhAkTsmvXrgO6dgaePWkAAHCAXbV8Q3+gJcneH/Zl66Zv5arlG3Lbbbdlzpw5rzh/1KhRGT9+fJYtW5Ykef755/Pcc89l27ZtGTt2bLq6urJy5cps3rx5QL8HA0OkAQDAAfbU1p2veD38qPHZvubL+Zer/lOeffbZXHjhha/6zK233prFixdn6tSpede73pXvfe97WbhwYXp6ejJlypTccsstOfHEE/vP/9CHPpR3vvOd2bBhQ8aPH5+bbrrpgH8vDoymbdsBn7S7u7vt6ekZ8HkBAGAwzL5yRXp/KtSSZNyYkXlw0fsGYUUMtqZpVrdt272v91xJAwCAA+zieRMzsmvYK46N7BqWi+dNHKQVUZkHhwAAwAF25oxxSV7am/bU1p1565iRuXjexP7j8JNEGgAADIAzZ4wTZbwubncEAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKCQ/Yq0pmnOa5rmkaZpXmyaprtTiwIAABiq9vdK2reSnJ3k6x1YCwAAwJA3fH8+3Lbtt5OkaZrOrAYAAGCIG7A9aU3TfKxpmp6maXr6+voGaloAAIA3lNe8ktY0zb1J3ryPty5v2/au1ztR27Y3JrkxSbq7u9vXvUIAAIAh5DUjrW3buQOxEAAAADyCHwAAoJT9fQT/WU3TfCfJO5N8uWma5Z1ZFgAAwNC0v093XJpkaYfWAgAAMOS53REAAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAANZnoyYAAAsySURBVEAhIg0AAKCQ/Yq0pmmuaprmsaZp/rVpmqVN04zp1MIAAACGov29kvbVJJPbtp2a5N+SXLr/SwIAABi69ivS2rb9/9u23fPyy1VJxu//kgAAAIauTu5JuyDJ/+rgeAAAAEPO8Nc6oWmae5O8eR9vXd627V0vn3N5kj1JlvyccT6W5GNJ8mu/9mu/1GIBAAAOdq8ZaW3bzv157zdN84dJfjfJ6W3btj9nnBuT3Jgk3d3dP/M8AACAoWx/n+44P8lfJTmjbdvnOrMk6KxNmzZl8uTJAzbXbbfdNiBzAQBwcNrfPWl/n2RUkq82TbOuaZobOrAmGHB79+7tyDgiDQCA/bW/T3f89bZtj2vbdvrLf/5zpxYGnbRnz54sXLgwkyZNyrnnnpvnnnsuEyZMyCWXXJKZM2fmjjvuyMaNGzN//vzMmjUrv/Ebv5HHHnssSfKlL30pp556ambMmJG5c+fm6aefTpJ87Wtfy/Tp0zN9+vTMmDEj27dvz6JFi3L//fdn+vTp+du//dvB/MoAALxBveaeNDgYbNiwITfddFNmz56dCy64INdff32S5Oijj86aNWuSJKeffnpuuOGGvP3tb883v/nNfPzjH8+KFSsyZ86crFq1Kk3T5POf/3w+/elP55prrsnVV1+d6667LrNnz86OHTty2GGH5corr8zVV1+du+++ezC/LgAAb2AijTJ+53d+J7fddlvGjBnTkfGWre3NVcs3ZPPmTRnxK2PT96YJSZLzzz8/ixcvTpIsWLAgSbJjx4584xvfyHnnndf/+eeffz5J8p3vfCcLFizId7/73bzwwgs54YQTkiSzZ8/ORRddlIULF+bss8/O+PF+TSAAAPtPpDHo2rZN27a55557OjbmsrW9ufTO9dm5+6W9ZnvbNpfeuT5JMjpJ0zRJksMPPzxJ8uKLL2bMmDFZt27dq8b65Cc/mYsuuihnnHFG7rvvvlxxxRVJkkWLFuUDH/hA7rnnnsyePTvLly/v2PoBABi6OvnLrBlCPvOZz2Ty5MmZPHlyrr322ixatCjXXXdd//tXXHFFrr766uzYsSOnn356Zs6cmSlTpuSuu+5K8tIDNiZOnJg/+IM/yOTJk/Pkk09mwoQJ2bJlS5LkzDPPzKxZs3LyySfnxhtv7B/3iCOOyOWXX55p06bltNNO698f9tOuWr6hP9CSZO8P+7J107dy1fINue222zJnzpxXnD969OiccMIJueOOO5K8FI4PP/xwkmTbtm0ZN25ckuTmm2/u/8zGjRszZcqUXHLJJTnllFPy2GOPZdSoUdm+ffsv/e8KAAAijV/Y6tWr84//+I/55je/mVWrVuVzn/tcFixYkNtvv73/nNtvvz0LFizIYYcdlqVLl2bNmjVZuXJl/vIv/zI//nV6jz/+eD7+8Y/nkUceyfHHH/+KOb7whS9k9erV6enpyeLFi/P9738/SfKjH/0op512Wh5++OG8+93vzuc+97l9rvGprTtf8Xr4UeOzfc2X8y9X/ac8++yzufDCC1/1mSVLluSmm27KtGnTcvLJJ/cH5RVXXJHzzjsvs2bNyjHHHNN//rXXXpvJkydn6tSp6erqym//9m9n6tSpGTZsWKZNm+bBIQAA/FLc7sjr8uP9XU9t3Zk8ck9Oeefp/bcKnn322bn//vvzzDPP5KmnnkpfX1+OPPLIHHfccdm9e3cuu+yyfP3rX88hhxyS3t7e/qtfxx9/fE477bR9zrd48eIsXbo0SfLkk0/m8ccfz9FHH50RI0bkd3/3d5Mks2bNyle/+tV9fv6tY0am9+VQG/4rv5pxH33pt0OMGzMy/2PR+5K8dDXvJ51wwgn5yle+8qqxPvjBD+aDH/zgq45/9rOf3efcK1as2OdxAAB4PVxJ4zX9eH9X79adaZNs27k7K779TJat7X3Feeedd17+6Z/+KV/84hf7H8ixZMmS9PX1ZfXq1Vm3bl1+9Vd/Nbt27Uryf/eD/bT77rsv9957b/75n/85Dz/8cGbMmNH/ma6urv79ZMOGDcuePXv2OcbF8yZmZNewVxwb2TUsF8+b+Ev/OwAAwEBwJY3X9NP7uw4df3K+f8+1ufJLD+f9/8+YLF26NLfeemtGjBiRj370o9myZUu+9rWvJXlpP9fYsWPT1dWVlStXZvPmza8537Zt23LkkUfmTW96Ux577LGsWrXqF17zmTPG9a/9qa0789YxI3PxvIn9xwEAoCqRxmv66f1dh77513PE5NOz5rMX5tTbR+VP/uRPMmPGjCTJ9u3bM27cuLzlLW9JkixcuDC/93u/lylTpqS7uzsnnnjia843f/783HDDDZk0aVImTpz4M2+JfC1nzhgnygAAeMNpfvwQh4HU3d3d9vT0DPi8/HJmX7mif3/XTxo3ZmQefHl/FwAA8Po1TbO6bdvufb1nTxqvyf4uAAAYOG535DXZ3wUAAANHpPG62N8FAAADw+2OAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFiDQAAIBCRBoAAEAhIg0AAKAQkQYAAFCISAMAAChEpAEAABQi0gAAAAoRaQAAAIWINAAAgEJEGgAAQCEiDQAAoBCRBgAAUIhIAwAAKESkAQAAFCLSAAAAChFpAAAAhYg0AACAQkQaAABAISINAACgEJEGAABQiEgDAAAoRKQBAAAUItIAAAAKEWkAAACFNG3bDvykTdOXZPOAT3zwOCbJlsFeBEOSnz0Gg587BoOfOwaDn7uh5fi2bY/d1xuDEmn8n/bu73XPOY7j+PPVtnKAHOxA7fstTr+h1FpqB2TS8M2OiZJTq622hP0JCgeU5ERZSSGlxJRTwmxqjJbEFiEHnGl5O7ju1Q42JXa/r/u6no+j+7ru++B18O7uel2f68d/k+TTqtrZnUPz4+ypg3OnDs6dOjh3usDLHSVJkiRpRCxpkiRJkjQilrTV9FJ3AM2Ws6cOzp06OHfq4NwJ8J40SZIkSRoVV9IkSZIkaUQsaZIkSZI0Ipa0FZfkUJJKsr07i6YvydNJTif5IslbSa7rzqTpSrI3yddJziR5ojuP5iHJepIPk3yZ5FSSA92ZNB9JtiT5PMk73VnUy5K2wpKsA3cD33dn0WwcA26qqluAb4Anm/NoopJsAV4A7gE2gAeSbPSm0kycBw5V1QZwG/CYs6clOgB81R1C/Sxpq+1Z4HHAp79oKarq/ao6v9j8CFjrzKNJ2wWcqapvq+pP4DVgX3MmzUBV/VhVxxef/2A4YN7Rm0pzkGQNuA94uTuL+lnSVlSSfcC5qjrZnUWz9SjwbncITdYO4IeLts/igbKWLMkNwK3Ax71JNBPPMZx8/6s7iPpt7Q6gy0vyAXD9Jb46AjzFcKmj9L/6p7mrqrcXvznCcEnQ0WVmk6RlSXI18AZwsKp+786jaUuyCfxcVZ8luaM7j/pZ0kasqu661P4kNwM3AieTwHDJ2fEku6rqpyVG1ARdbu4uSPIIsAnsKV+0qCvnHLB+0fbaYp90xSXZxlDQjlbVm915NAu7gfuT3AtcBVyb5NWqeqg5l5r4MusJSPIdsLOqfu3OomlLshd4Bri9qn7pzqPpSrKV4eE0exjK2SfAg1V1qjWYJi/D2c9XgN+q6mB3Hs3PYiXtcFVtdmdRH+9Jk/RvPA9cAxxLciLJi92BNE2LB9TsB95jeHDD6xY0Lclu4GHgzsX/3InF6oYkLY0raZIkSZI0Iq6kSZIkSdKIWNIkSZIkaUQsaZIkSZI0IpY0SZIkSRoRS5okSZIkjYglTZIkSZJGxJImSZIkSSPyN+H2z0IJSg5hAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x1080 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "words = list(model.wv.vocab)\n",
    "targets = [\"brca1\", \"brca2\", \"tp53\", \"breast\", \"cancer\", \"ovarian\"]\n",
    "indexes = [words.index(w) for w in targets]\n",
    "\n",
    "pca = sklearn.decomposition.PCA(n_components=2)\n",
    "result = pca.fit_transform(X[indexes])\n",
    "\n",
    "# create a scatter plot of the projection\n",
    "plt.figure(figsize=(15,15))\n",
    "pyplot.scatter(result[:, 0], result[:, 1])\n",
    "words = list(model.wv.vocab)\n",
    "\n",
    "for i, word in enumerate(targets):\n",
    "    pyplot.annotate(word, xy=(result[i, 0], result[i, 1]))\n",
    "pyplot.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
